---
title: "Effect of Paracetamol on Neuronal Differentiation"
Authors: Ankush Sharma with contributions from Martin Falck
output:
  date: "`r Sys.Date()`"
  html_document:
    df_print: paged
---

#==========================================================================================================

#Description: This script analyzes the integrative single cell ATAC seq analysis  for D20 P100 and  D20 P200  and D20 ctrl sample of human embryonic stem cells differentiated for 20 days (more details in Samara et al ,2022 StarProtocol and Iscience)  and the analysis and resuts will appear in manuscript of Spildrejorde et al. 

#==========================================================================================================

```{r setup Import packages, message=FALSE, warning=FALSE,include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
invisible(utils::memory.limit(64000))

reqPkg = c("markdown","Seurat","SingleCellExperiment","SummarizedExperiment","EnsDb.Hsapiens.v86","BSgenome.Hsapiens.UCSC.hg38","AnnotationHub","clustree","cowplot","ensembldb","ggthemes","ggplot2","ggplotify","gridGraphics","Matrix","org.Hs.eg.db","Matrix.utils","patchwork","reticulate","reshape2","RCurl","scales","scater","chromVAR","tidyverse","edgeR","plotly","robustbase","knitr","ggpubr","clustree","dplyr","kableExtra", "VISION","JASPAR2018","TFBSTools","Signac")

lapply(reqPkg, library, character.only = TRUE)
# Seed (set for reproducibility)
set.seed(1234)

# Increase threshold before R starts showing scientific notation
options(scipen = 10L)


```



```{r importing files }

# Settings  Up variable for file name 
currentjob <- "NeuroDiffParacet"

# Path and name
dirname1<- "~/INPUT_FILES/aggr_d0_ctrl/outs/filtered_feature_bc_matrix/"

# scRNAseq - Initialize the Seurat object with the raw (non-normalized data).
NeuroDiffParacet.data <- Read10X(data.dir = dirname1)
D0 <- CreateSeuratObject(counts = NeuroDiffParacet.data, 
                           min.cells = 3, 
                           min.features = 200, 
                           project = "D0",
                           assay = "RNA",
                           names.delim = "-")
###
dirname2 <- "~/INPUT_FILES/aggr_d7_ctrl/outs/filtered_feature_bc_matrix/"
NeuroDiffParacet.data <- Read10X(data.dir = dirname2)
D7 <- CreateSeuratObject(counts = NeuroDiffParacet.data, 
                           min.cells = 3, 
                           min.features = 200, 
                           project = "D7",
                           assay = "RNA",
                           names.delim = "-")

####
dirname3<- "~/INPUT_FILES/aggr_d13_ctrl/outs/filtered_feature_bc_matrix/"
NeuroDiffParacet.data <- Read10X(data.dir = dirname3)
D13 <- CreateSeuratObject(counts = NeuroDiffParacet.data, 
                           min.cells = 3, 
                           min.features = 200, 
                           project = "D13",
                           assay = "RNA",
                           names.delim = "-")
#####
dirname4<- "~/INPUT_FILES/aggr_d20_ctrl/outs/filtered_feature_bc_matrix/"

NeuroDiffParacet.data <- Read10X(data.dir = dirname4)
D20 <- CreateSeuratObject(counts = NeuroDiffParacet.data, 
                           min.cells = 3, 
                           min.features = 200, 
                           project = "D20",
                           assay = "RNA",
                           names.delim = "-")

# Path and name
dirname1<- "~/INPUT_FILES/aggr_d7_p100/outs/filtered_feature_bc_matrix/"


# scRNAseq - Initialize the Seurat object with the raw (non-normalized data).
NeuroDiffParacet.data <- Read10X(data.dir = dirname1)
D7.P100 <- CreateSeuratObject(counts = NeuroDiffParacet.data, 
                           min.cells = 3, 
                           min.features = 200, 
                           project = "D7.P100",
                           assay = "RNA",
                           names.delim = "-")
###
dirname2 <- "~/INPUT_FILES/aggr_d7_p200/outs/filtered_feature_bc_matrix/"
NeuroDiffParacet.data <- Read10X(data.dir = dirname2)
D7.P200 <- CreateSeuratObject(counts = NeuroDiffParacet.data, 
                           min.cells = 3, 
                           min.features = 200, 
                           project = "D7.P200",
                           assay = "RNA",
                           names.delim = "-")

####
dirname3 <- "~/INPUT_FILES/aggr_d20_p100/outs/filtered_feature_bc_matrix/"
NeuroDiffParacet.data <- Read10X(data.dir = dirname3)
D20.P100 <- CreateSeuratObject(counts = NeuroDiffParacet.data, 
                           min.cells = 3, 
                           min.features = 200, 
                           project = "D20.P100",
                           assay = "RNA",
                           names.delim = "-")
#####
dirname4<- "~/INPUT_FILES/aggr_d20_p200/outs/filtered_feature_bc_matrix/"

NeuroDiffParacet.data <- Read10X(data.dir = dirname4)
D20.P200 <- CreateSeuratObject(counts = NeuroDiffParacet.data, 
                           min.cells = 3, 
                           min.features = 200, 
                           project = "D20.P200",
                           assay = "RNA",
                           names.delim = "-")




NeuroDiffParacet.merged <- merge(D0, y = c(D7, D13, D20, D7.P100, D7.P200, D20.P100, D20.P200), add.cell.ids = c("D0", "D7","D13","D20","D7.P100","D7.P200","D20.P100","D20.P200"), project = "NeuroDiffParacetamol")


head(colnames(NeuroDiffParacet.merged))

table(NeuroDiffParacet.merged$orig.ident)


```


```{r Removing objects, include=FALSE}

NeuroDiffParacet <- NeuroDiffParacet.merged
dir.create("plots")
dir.create("plots/qc")
dir.create("plots/RDSFiles")
dir.create("plots/results")
dir.create("RMDFiles")

```

```{r QC metrics -violin plots1}
# scRNAseq
counts_per_cell <- Matrix::colSums(NeuroDiffParacet)
counts_per_gene <- Matrix::rowSums(NeuroDiffParacet)
genes_per_cell <- Matrix::colSums(NeuroDiffParacet)

# Plots
# Plots

# Counts per cell
p1 <- qplot(log10(counts_per_cell+1), geom = "histogram", main="Counts per cell",
        xlab="log10 Counts per cell",
        fill=I('#dbd5c8'),
        col=I("white"))+theme_pubr()



# Genes per cell

p2 <- qplot(log10(genes_per_cell+1), geom = "histogram", main="Genes per cell",
        xlab="log10 Genes per cell",
        fill=I('#dbd5c8'),
        col=I("white"))+theme_pubr()

# Counts per cell vs Genes per cell
p3 <- qplot(counts_per_cell, genes_per_cell, log = "xy", 
            colour=I('#dbd5c8'), 
            main="Counts vs Genes per cell",
            ylab = "Genes",
            xlab="Counts")+theme_pubr()

# Histogram of Counts per gene in log10 scale
p4 <- qplot(log10(counts_per_gene+1), geom = "histogram", main="Counts per gene",
             xlab="log10 Counts per gene",
        fill=I('#dbd5c8'),
        col=I("white"))+theme_pubr()


(p1 & p2) / (p3 & p4)
dev.copy(pdf,"../output/qc/QCmetrics_panel.pdf")+theme_pubr(base_family = "Arial")
dev.off()

```


```{r QC metrics -violin plots2}

# The [[ operator can add columns to object metadata. This is a great place to stash QC stats
NeuroDiffParacet[["percent.mt"]] <- PercentageFeatureSet(NeuroDiffParacet, pattern = "^MT-")

# Show QC metrics for the first 5 cells
head(NeuroDiffParacet@meta.data, 5)

# Visualize QC metrics as a violin plot

p1 <- VlnPlot(object = NeuroDiffParacet, features = c("nCount_RNA","nFeature_RNA","percent.mt"), ncol = 3, pt.size = 0.2)
p1
dev.copy(pdf,paste0("../output/qc/QCmetrics-",currentjob,".pdf"))
dev.off()

```

```{r feature scatter plots}
# FeatureScatter is typically used to visualize feature-feature relationships, but can be used
# for anything calculated by the object, i.e. columns in object metadata, PC scores etc.

plot2 <- FeatureScatter(NeuroDiffParacet, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle("Counts vs Mitochondrial percentage") +  theme(plot.title = element_text(size = 12, face = "bold")) + NoLegend() + theme(text = element_text(size = 10))
plot3 <- FeatureScatter(NeuroDiffParacet, feature1 = "nCount_RNA", feature2 = "nFeature_RNA") + ggtitle("Counts vs Features") +  theme(plot.title = element_text(size = 12, face = "bold")) + NoLegend() + theme(text = element_text(size = 10))
plot2 + plot3
dev.copy(pdf,paste0("../output/qc/Featurescatter-",currentjob,".pdf"))
dev.off()

```
#Regression, but no filtering yet (QC metrics evaluated again later in script)!
```{r Percentage feature set, message=FALSE, warning=FALSE}
# Add mitochondria gene % information
NeuroDiffParacet <- PercentageFeatureSet(NeuroDiffParacet, pattern = "^MT-", col.name = "percent.mt")

# Plot and save QC metrics for dataset (unfiltered)
VlnPlot(NeuroDiffParacet, features = c("nFeature_RNA", "nCount_RNA", "percent.mt"), ncol = 3, combine = TRUE)
dev.copy(pdf,paste0("../output/qc/", currentjob,"_1.nFeat-nCount-perMT.pdf"), width=8, height=10, paper='special')
dev.off()
```


```{r CC Scoring, message=FALSE, warning=FALSE}
# Import cell cycle (cc) genes stored in Seurat then do cc scoring on dataset
# As an alternative to completely removing cell-cycle signal this workflow regress out the difference between the G2M and S-phase scores. Seurat recommends this for differentiating cells and it was best in tests.
s.genes <- cc.genes.updated.2019$s.genes
g2m.genes <- cc.genes.updated.2019$g2m.genes
NeuroDiffParacet <- CellCycleScoring(NeuroDiffParacet, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
NeuroDiffParacet$CC.Difference <- NeuroDiffParacet$S.Score - NeuroDiffParacet$G2M.Score

# Visualize the distribution of cell cycle markers across
RidgePlot(NeuroDiffParacet, features = c("PCNA", "TOP2A", "MCM6", "MKI67"), ncol = 2)
dev.copy(pdf,paste0("../output/qc/", currentjob,"_2.CC-markers-distribution.pdf"), height=5, width=8)
dev.off()
```


```{r SCTransform normalization, message=FALSE, warning=FALSE, include=FALSE}
# Normalization and scaling with SCTransform (see Seurat webpage for information)
NeuroDiffParacet <- SCTransform(NeuroDiffParacet, vars.to.regress = c("percent.mt","CC.Difference"))

# Save the normalized object for other analysis and backup
#saveRDS(NeuroDiffParacet, file = paste0("../output/RDSFiles/", currentjob,"_B.SCTransformed.DS-object.rds"))

```

#Principal component analysis (PCA)
```{r PCA, message=FALSE, warning=FALSE}
# Run PCA
NeuroDiffParacet <- RunPCA(NeuroDiffParacet, verbose = FALSE)
```

#Clustree branchpoint analysis
```{r Clustree, message=FALSE, warning=FALSE}
# This code can be used to evaluate what resolution can be used when running clustering algorithm (louvain)
# It is only an approximation of how many clusters are reasonable. 

# Run the FindClusters with a vector of different values then run clustree and see where datset has stabile branching then use that resolution for further analysis. 
NeuroDiffParacet <- FindNeighbors(NeuroDiffParacet, dims = 1:30, verbose = FALSE)
NeuroDiffParacet <- FindClusters(NeuroDiffParacet, resolution = c(
  0.1,0.15,0.2,0.25,0.3,0.35,0.4,0.45,0.5,0.55,0.6,0.7,0.8,1),
  verbose = FALSE)
NeuroDiffParacet <- RunUMAP(NeuroDiffParacet, dims = 1:30, verbose = FALSE)

# Clustree plots
clustree(NeuroDiffParacet, prefix="SCT_snn_res.", use_core_edges=TRUE, label=T)
dev.copy(pdf,paste0("../output/qc/", currentjob,"_3.Clustree.pdf"), height=5, width=8)
dev.off()
clustree_overlay(NeuroDiffParacet, x_value = "umap1", y_value = "umap2", red_dim = "umap", prefix="SCT_snn_res.")
dev.copy(pdf,paste0("../output/qc/", currentjob,"_3.2.ClustreeOverlay.pdf"), height=5, width=8)
dev.off()

# Once decided on a resolution, rerun FindClusters with the picked value (in this case 0.5)
NeuroDiffParacet <- FindClusters(NeuroDiffParacet, resolution = 0.5, verbose = FALSE)
```


#Clustering and dimensions reduction (UMAP)
```{r, message=FALSE, warning=FALSE}
# Pick appropriate dims from Elbowplot (when it flattens enough). 
# Use Jackstraw for more accuracy if needed
# Tweak FindNeigbors in Clustree chunk if you change dims!

ElbowPlot(NeuroDiffParacet, ndims = 50)
dev.copy(pdf,paste0("../output/qc/", currentjob,"_4.Elbowplot.pdf"), height=5, width=8)
dev.off()

# Standard Seurat3, remember the resolution from Clustree above!
NeuroDiffParacet <- RunUMAP(NeuroDiffParacet, dims = 1:30, verbose = FALSE)
NeuroDiffParacet <- FindNeighbors(NeuroDiffParacet, dims = 1:30, verbose = FALSE)
NeuroDiffParacet <- FindClusters(NeuroDiffParacet, resolution = 0.5, verbose = FALSE)
DimPlot(NeuroDiffParacet, label = TRUE, label.size = 5)
dev.copy(pdf,paste0("../output/qc/", currentjob,"_5.UMAP.un-labelled-clusters.pdf"), height=5, width=8)
dev.off()

```

#Find cluster markers
```{r Find cluster markers, message=FALSE, warning=FALSE}
# Run normalization and scaledata on "RNA" slot as recommended by Seurat/Satijahlab
DefaultAssay(NeuroDiffParacet) <- "RNA"
NeuroDiffParacet <- NormalizeData(NeuroDiffParacet, assay="RNA")
NeuroDiffParacet <- FindVariableFeatures(NeuroDiffParacet, selection.method = "vst", nfeatures = 2000, assay="RNA")
top10 <- head(VariableFeatures(NeuroDiffParacet), 10)
NeuroDiffParacet <- ScaleData(NeuroDiffParacet, vars.to.regress = c("percent.mt","CC.Difference"))

NeuroDiffParacet.markers <- FindAllMarkers(NeuroDiffParacet, only.pos = F, min.pct = 0.1, logfc.threshold = 0.2)
top500 <- NeuroDiffParacet.markers %>% group_by(cluster) %>% top_n(n = 500, wt = avg_log2FC)

# Write markers to file
write.table(top500, paste0("../output/outputfiles/", currentjob,"_Unfilt.Top500-cluster-markers.csv"))

```

#Heatmap Top10 genes per Cluster
```{r Heatmap: Top10 genes per Cluster1, message=FALSE, warning=FALSE}
# setting slim.col.label to TRUE will print just the cluster IDS instead of
# every cell name
top10 <- NeuroDiffParacet.markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)
DoHeatmap(NeuroDiffParacet, features = top10$gene, label = FALSE) 
pdf(file = paste0("../output/plots/", currentjob,"_6.heatmap.Top10-cluster-markers.pdf"), width=12, height=14, paper='special')
DoHeatmap(NeuroDiffParacet, features = top10$gene, label = FALSE) 
dev.off()

```



```{r UMAP PLOTS, message=FALSE, warning=FALSE}
# Assign new IDs based on scCatch annotation results - Using Seurat Markers
#new.cluster.ids_2 <- c("ESC1","ESC2","ESC3","ESC4","NSC1","NSC2","NPC1","ESC5","Neuron", "ESC6","NPC2","NPC3","ESC7","ESC8","Astrocyte1","MicroglialCells","Astrocyte2")
#names(new.cluster.ids_2) <- levels(NeuroDiffParacet)
#NeuroDiffParacet <- RenameIdents(NeuroDiffParacet, new.cluster.ids_2)

# Change back to "SCT" assay for dims/clusters
DefaultAssay(NeuroDiffParacet) <- "SCT"

# Make UMAP with labels from scCatch
plot9 <- DimPlot(NeuroDiffParacet, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
HoverLocator(plot = plot9, information = FetchData(NeuroDiffParacet, vars = c("ident", "PC_1", "nFeature_RNA")))
pdf(file = paste0("../output/plots/", currentjob,"_7.UMAP-scCatch.unFilt.pdf"), width=8, height=5, paper='special')
DimPlot(NeuroDiffParacet, reduction = "umap", label = TRUE, pt.size = 0.5)
dev.off()

# So some labels do not look great considering what the sample i made up from. We should check QC metrics on clusters! E.g. for day 20 some are labelled ESCs!
DimPlot(NeuroDiffParacet, label = TRUE, label.size = 5, reduction = 'umap', repel=TRUE, group.by = 'orig.ident')


#FOR ASSIGNING COLORS
#DimPlot(NeuroDiffParacet, label = TRUE, label.size = 5, reduction = 'umap', group.by = 'orig.ident', cols = c('d1-ASC' = '#ffd700', 'd0_ASC' = '#8b9dc3', 'd3-ASC' = '#ff7f50', 'd_2-ASC' = 'black', 'd1-LS1' = '#ffd700', 'd0_LS1' = '#008080', 'd3-LS1' = '#f30000', 'd_2-LS1' = '#e0e6b0'))
        
dev.copy(pdf,paste0("../output/plots/", currentjob,"_5.UMAP.by.origin.pdf"), height=5, width=8)
dev.off()

```
#Save RDS for end of pipe file, first run (no filtering)
```{r Save end-RDS nr 1 of 2, message=FALSE, warning=FALSE}
# Save final RDS and countfile (files 1 of 2, unfiltered)
saveRDS(NeuroDiffParacet, file = paste0("../output/RDSFiles/", currentjob,"_FirstIteration.rds"))
sct.counts <- NeuroDiffParacet@assays$SCT@counts
write.csv(sct.counts, paste0("../output/outputFiles/", currentjob,"_FirstIteration_countMatrix-unFilt.csv"))
```

#Time to re-evaluate some strange clusters!
```{r Unfiltered dataset, warning=FALSE}
# Check the processed dataset for re-evaluation on counts, features and percent mt across cell types
# We found this to be a good diagnostic to tweak QC, filtering using isOutlier() from scater as we can look at both biology and QC metrics across clusters and annotation and THEN we can do filtering to not loose important cells.
# Add the active ident (assigned cell types by scCatch) to metadata as 'cell_type'
currentjob <- "NeuroDiffParacetFinal"

names(NeuroDiffParacet@meta.data)
NeuroDiffParacet <- AddMetaData(NeuroDiffParacet, metadata =NeuroDiffParacet@active.ident ,col.name = 'cell_type')

# Print table with n cells and fraction per cluster/cell-type
table(Idents(NeuroDiffParacet))
prop.table(table(Idents(NeuroDiffParacet)))

# VlnPlot counts, features and % mit grouped by cell-type
VlnPlot(NeuroDiffParacet, c("nCount_RNA"), pt.size = 0.1, ncol = 1, group.by = "cell_type") + NoLegend()
dev.copy(pdf,paste0(currentjob,"_8.ReEval.nCount-cell-type.pdf"), width=8, height=6, paper='special')
dev.off()
VlnPlot(NeuroDiffParacet, c("nFeature_RNA"), pt.size = 0.1, ncol = 1, group.by = "cell_type") + NoLegend()
dev.copy(pdf,paste0(currentjob,"_8.ReEval.nFeature-cell-type.pdf"), width=8, height=6, paper='special')
dev.off()
VlnPlot(NeuroDiffParacet, c("percent.mt"), pt.size = 0.1, ncol = 1, group.by = "cell_type") + NoLegend()
dev.copy(pdf,paste0(currentjob,"_8.ReEval.percent.mt-cell-type.pdf"), width=8, height=6, paper='special')
dev.off()

# Scatterplot comparing counts to % mit
FeatureScatter(NeuroDiffParacet, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle(label = "Scatter: count vs % mt | Unfiltered data")
dev.copy(pdf,paste0("../output/plots/", currentjob,"_8.ReEval.scatter.count.vs.mt.pdf"), width=8, height=5, paper='special')
dev.off()
FeatureScatter(NeuroDiffParacet, feature1 = "nCount_RNA", feature2 = "percent.mt") + ggtitle(label = "Scatter: count vs % mt | Unfiltered data")+ scale_x_continuous(name="nCount_RNA", limits=c(0, 10000))
dev.copy(pdf,paste0("../output/plots/", currentjob,"_8.ReEval.scatter.count.vs.mt.ZOOM.pdf"), width=8, height=5, paper='special')
dev.off() 

# E.g d20: It looks like cells with counts under 2500 and percent mt above ~ 12 are bad!
# We found two clusters that were likely wrongly named/identified and also contain extreme amounts of % mt reads and very low counts and features. These clusters/cell-names were "Mesenchymal stem cell" and "Astrocyte" 
 
# Check the particular offender(s) by subsetting the data
#unfil_Mesenchymal <- subset(NeuroDiffParacet, idents = c("ESC4"))
#VlnPlot(unfil_Mesenchymal, c("nCount_RNA","nFeature_RNA","percent.mt"), pt.size = 0.1, ncol = 3, ) + NoLegend()
#dev.copy(pdf,paste0(currentjob,"_9.ReEval.mesenchymal.pdf"), width=8, height=10, paper='special')
#dev.off()

# This gives a reason to go back and provide an as-good as possible filter to avoid this cluster from forming and this will be the adapted threshold to aim for. Doing analysis with unfiltered data first is thus good as we check the results first without removing any signal that could be biology then we go back and adjust after.
```
#Go back and filter on findings from above!
```{r Scater analysis, echo=TRUE, warning=FALSE}
# Work from a newly named qc dataset
NeuroDiffParacet.qc <- NeuroDiffParacet

# Change back to the raw counts
DefaultAssay(NeuroDiffParacet.qc) <- "RNA"
DefaultAssay(NeuroDiffParacet.qc) #Confirm

# Rerun PercentageFeatureSet
NeuroDiffParacet.qc <- PercentageFeatureSet(NeuroDiffParacet.qc, pattern = "^MT-", col.name = "percent.mt")

# Scater detection of outliers, isOutlier()
# nmads= will be tweaked here to set a proper filter level so to not loose important cells. 
# Some cells might look weird but have important gene expression and actually not be junk cells! We start from the default of nmads=3.
discard.mito <- isOutlier(NeuroDiffParacet.qc$percent.mt, type="higher", nmads=3)


# Synnary gives a certain amount of cells that will be removed.
summary(discard.mito)

# Plot counts vs mit % and add horizontal line 
plot(NeuroDiffParacet.qc$nCount_RNA, NeuroDiffParacet.qc$percent.mt, log="x",
    xlab="Total count", ylab='Mitochondrial %')
abline(h=attr(discard.mito, "thresholds")["higher"], col="red")
dev.copy(pdf,paste0("../output/plots/", currentjob,"_10.FILT.qc_countsVSmitpercent.pdf"), width=8, height=6, paper='special')
dev.off()

lost <- calculateAverage(NeuroDiffParacet.qc@assays$RNA@counts[,!discard.mito])
kept <- calculateAverage(NeuroDiffParacet.qc@assays$RNA@counts[,discard.mito])

# If discarded pool is enriched for a certain cell type, we should observe increased expression of the corresponding marker genes.
logged <- cpm(cbind(lost, kept), log=TRUE, prior.count=2)
logFC <- logged[,1] - logged[,2]
abundance <- rowMeans(logged)
is.mito <- grep("^MT-",rownames(NeuroDiffParacet.qc@assays$RNA),ignore.case = TRUE)
plot(abundance, logFC, xlab="Average count", ylab="Log-FC (lost/kept)", pch=16)
points(abundance[is.mito], logFC[is.mito], col="dodgerblue", pch=16)
dev.copy(pdf,paste0("../output/plots/", currentjob,"_10.FILT.qc_logFC-abundance-in-Discarded-Cells.pdf"), width=8, height=6, paper='special')
dev.off()

# Top 200 gene set of positive log-fold changes, remove unwanted cells and create .qcFilt
# Analyze this list to know what nmads= threshold is suitable for keeping cells/features that are interesting
coefs <- predFC(cbind(lost, kept), design=cbind(1, c(1, 0)))[,2]
info <- data.frame(logFC=coefs, Lost=lost, Kept=kept, 
    row.names=rownames(NeuroDiffParacet.qc))
top200_info <- head(info[order(info$logFC, decreasing=TRUE),], 200)

```
#Scater outlier filtering on determined nmads=
```{r SCATER outlier filter on determined nmads, echo=TRUE}
# From the top200 table above, we picked e.g. nmads=5 for our d20 sample
# Using that nmads value and rerunning previous code to gather diagnostic plots for comparison
discard.mito <- isOutlier(NeuroDiffParacet.qc$percent.mt, type="higher", nmads=5)
summary(discard.mito)
plot(NeuroDiffParacet.qc$nCount_RNA, NeuroDiffParacet.qc$percent.mt, log="x",
    xlab="Total count", ylab='Mitochondrial %')
abline(h=attr(discard.mito, "thresholds")["higher"], col="red")
dev.copy(pdf,paste0("../output/plots/", currentjob,"_10.2.FILT.mads5.qc_countsVSmitpercent.pdf"), width=8, height=6, paper='special')
dev.off()
lost <- calculateAverage(NeuroDiffParacet.qc@assays$RNA@counts[,!discard.mito])
kept <- calculateAverage(NeuroDiffParacet.qc@assays$RNA@counts[,discard.mito])
logged <- cpm(cbind(lost, kept), log=TRUE, prior.count=2)
logFC <- logged[,1] - logged[,2]
abundance <- rowMeans(logged)
is.mito <- grep("^MT-",rownames(NeuroDiffParacet.qc@assays$RNA),ignore.case = TRUE)
plot(abundance, logFC, xlab="Average count", ylab="Log-FC (lost/kept)", pch=16)
points(abundance[is.mito], logFC[is.mito], col="dodgerblue", pch=16)
dev.copy(pdf,paste0("../output/plots/", currentjob,"_10.2.FILT.mads5.qc_logFC-abundance-in-Discarded-Cells.pdf"), width=8, height=6, paper='special')
dev.off()
coefs <- predFC(cbind(lost, kept), design=cbind(1, c(1, 0)))[,2]
info.nmads5 <- data.frame(logFC=coefs, Lost=lost, Kept=kept, 
    row.names=rownames(NeuroDiffParacet.qc))
top200_nmads5.info <- head(info.nmads5[order(info.nmads5$logFC, decreasing=TRUE),], 200)
top200_nmads5.info
# Save the top200 list for documentation
write.csv(top200_nmads5.info, paste0("../output/outputfiles/", currentjob,"_F.top200.nmads5.logFC-in-DiscardedPool.csv"))

```

#Rerun pipeline without nmads filtered outliers
```{r Continue with filtered dataset using all steps in first run!, message=FALSE, warning=FALSE, include=FALSE}
# Remove outliers from dataset
# Remove above 50k reads and under 2000
NeuroDiffParacet.qcFilt <- NeuroDiffParacet.qc[,!discard.mito]
NeuroDiffParacet.qcFilt <- subset(NeuroDiffParacet.qcFilt, nCount_RNA >= 2000 & nCount_RNA <= 50000)

# Rerun pipeline with .qcFilt dataset
# CC-regression, SCTransform, PCA, etc
s.genes <- cc.genes.updated.2019$s.genes
g2m.genes <- cc.genes.updated.2019$g2m.genes
NeuroDiffParacet.qcFilt <- CellCycleScoring(NeuroDiffParacet.qcFilt, s.features = s.genes, g2m.features = g2m.genes, set.ident = TRUE)
NeuroDiffParacet.qcFilt$CC.Difference <- NeuroDiffParacet.qcFilt$S.Score - NeuroDiffParacet.qcFilt$G2M.Score
NeuroDiffParacet.qcFilt <- SCTransform(NeuroDiffParacet.qcFilt, vars.to.regress = c("percent.mt","CC.Difference"))
NeuroDiffParacet.qcFilt <- RunPCA(NeuroDiffParacet.qcFilt)
NeuroDiffParacet.qcFilt <- FindNeighbors(NeuroDiffParacet.qcFilt,dims = 1:30, verbose = FALSE)
#NeuroDiffParacet.qcFilt <- RunTSNE(NeuroDiffParacet.qcFilt, dims = 1:30)
NeuroDiffParacet.qcFilt <- RunUMAP(NeuroDiffParacet.qcFilt, dims = 1:30)
```

#Clustree II, branchpoint analysis again!
```{r Clustree II, echo=TRUE, warning=FALSE}
# Run the FindClusters with a vector of different values analysis. 
NeuroDiffParacet.qcFilt <- FindNeighbors(NeuroDiffParacet.qcFilt, dims = 1:30, verbose = FALSE)
NeuroDiffParacet.qcFilt <- FindClusters(NeuroDiffParacet.qcFilt, resolution = c(
  0.1,0.15,0.2,0.25,0.3,0.35,0.4,0.45,0.5,0.55,0.6,0.7,0.8,0.9,1),
  verbose = FALSE)
NeuroDiffParacet.qcFilt <- RunUMAP(NeuroDiffParacet.qcFilt, dims = 1:30, verbose = FALSE)

# Clustree II plots
clustree(NeuroDiffParacet.qcFilt, prefix="SCT_snn_res.", use_core_edges=TRUE, label=T)
dev.copy(pdf,paste0("../output/plots/", currentjob,"_11.FILT.Clustree.pdf"), height=5, width=8)
dev.off()
clustree_overlay(NeuroDiffParacet.qcFilt, x_value = "umap1", y_value = "umap2", red_dim = "umap", prefix="SCT_snn_res.")
dev.copy(pdf,paste0("../output/plots/", currentjob,"_11.2.FILT.ClustreeOverlay.pdf"), height=5, width=8)
dev.off()

# Decided on e.g. 0.20 for the d20 sample
DefaultAssay(NeuroDiffParacet.qcFilt) <- "SCT"
NeuroDiffParacet.qcFilt <- FindClusters(NeuroDiffParacet.qcFilt, resolution = 0.5, verbose = FALSE)
#NeuroDiffParacet.qcFilt <- RunTSNE(NeuroDiffParacet.qcFilt, dims = 1:30)
NeuroDiffParacet.qcFilt <- RunUMAP(NeuroDiffParacet.qcFilt, dims = 1:30)

```

#MARKERS filtered
```{r markers filtered}
# Also check if only.pos, min.pct, and logfc.threshold can be changed to speed up with a better cell pool

# Run default normalization and scaleDATA on the "RNA" slot as recommended by Seurat/Satijalab
DefaultAssay(NeuroDiffParacet.qcFilt) <- "RNA"
NeuroDiffParacet.qcFilt <- NormalizeData(NeuroDiffParacet.qcFilt, assay="RNA")
NeuroDiffParacet.qcFilt <- FindVariableFeatures(NeuroDiffParacet.qcFilt, selection.method = "vst", nfeatures = 2000, assay="RNA")
top10 <- head(VariableFeatures(NeuroDiffParacet.qcFilt), 10)
NeuroDiffParacet.qcFilt <- ScaleData(NeuroDiffParacet.qcFilt, vars.to.regress = c("percent.mt","CC.Difference"))

#scCatch
DefaultAssay(NeuroDiffParacet.qcFilt) <- "RNA"
NeuroDiffParacet.qcFilt.markers <- FindAllMarkers(
  NeuroDiffParacet.qcFilt,
  only.pos = F,
  min.pct = 0.1,
  logfc.threshold = 0.25)

top500.NeuroDiffParacet.qcFilt <- NeuroDiffParacet.qcFilt.markers %>%
  group_by(cluster) %>% top_n(n = 500, wt = avg_log2FC)
top30.NeuroDiffParacet.qcFilt <- NeuroDiffParacet.qcFilt.markers %>%
  group_by(cluster) %>% top_n(n = 30, wt = avg_log2FC)

plotUMAP1 <- DimPlot(NeuroDiffParacet.qcFilt, reduction = "umap", label = TRUE, pt.size = 0.5,cols = c('0' = '#D37295', '1' = '#FF9888', '2' = '#8175AA', '3' = '#DAB6AF','4' = '#FFC966','5' = '#C3CE3D', '6' = '#8CC2CA', '7' = '#59A14F', '8' = '#CCCCCC', '9' = '#008080', '10' = '#C0D6E4','11' = '#FFD700','12' = '#A3ACB9','13' = '#F28E2B','14' = '#B60A1C','15' = '#8A494D','16' = '#C8D0D9'))

plotUMAP1
dev.copy(pdf,paste0("../output/plots/", currentjob,"_5.UMAP_default_labeled.pdf"), height=5, width=8)
dev.off()
dev.set(dev.next())
write.table(top500.NeuroDiffParacet.qcFilt, paste0("../output/outputfiles/", currentjob,"_filt.Top500-cluster-markers.csv"))
write.table(top30.NeuroDiffParacet.qcFilt, paste0("../output/outputfiles/", currentjob,"_filt.Top30-cluster-markers.csv"))

```
```{r UMAP_BY_FILT by origin }

#UMAP_BY_FILT by origin
library(crosstalk)

#####DIFFRENTIATION COLOR BY ORIG;IDENT cols =c('D0' = '#FF9888', 'D7' = '#B60A1C','D13' = '#C3CE3D','D20' = '#C0D6E4'))
DimPlot(NeuroDiffParacet.qcFilt, label = TRUE, label.size = 4, reduction = 'umap', group.by = 'orig.ident', repel = TRUE, cols =c('D0' = '#FF9888', 'D7' = '#B60A1C','D13' = '#C3CE3D','D20' = '#C0D6E4', 'D7.P100' = '#FFD700', 'D7.P200' = '#8CC2CA','D20.P100' = '#D37295', 'D20.P200' = '#8175AA'))
dev.copy(pdf,paste0("../output/plots/", currentjob,"_5.UMAP_by_origin.pdf"), height=5, width=8)
dev.off()
dev.set(dev.next())


##ADDD ctrl in Project where we add input data if we require d0.ctrl
DimPlot(NeuroDiffParacet.qcFilt, label = TRUE, label.size = 4, reduction = 'umap', group.by = 'orig.ident', repel = TRUE, cols =c('D0.Ctrl' = '#FF9888', 'D7.Ctrl' = '#B60A1C','D13.Ctrl' = '#C3CE3D','D20.Ctrl' = '#C0D6E4', 'D7.P100' = '#FFD700', 'D7.P200' = '#8CC2CA','D20.P100' = '#D37295', 'D20.P200' = '#8175AA'))
dev.copy(pdf,paste0("../output/plots/", currentjob,"_5.UMAP_by_origin.pdf"), height=5, width=8)
dev.off()
dev.set(dev.next())

#plotUMAP2 <-  DimPlot(NeuroDiffParacet.qcFilt, label = TRUE, label.size = 5, reduction = 'umap', group.by = 'orig.ident', repel = TRUE)
#HoverLocator(plot = plotUMAP2, information = FetchData(NeuroDiffParacet.qcFilt, vars = c("ident", "PC_1", "nFeature_RNA","percent.mt")))


```

```{r scCatch Annotation using seurat}
rnaobjects.qcFilt$original_seurat_clusters<-rnaobjects.qcFilt$seurat_clusters 
DimPlot(rnaobjects.qcFilt, reduction = "umap", label = TRUE,group.by = "original_seurat_clusters", pt.size = 0.5,cols = c('0' = '#D37295', '1' = '#FF9888', '2' = '#8175AA', '3' = '#DAB6AF','4' = '#FFC966','5' = '#C3CE3D', '6' = '#8CC2CA', '7' = '#59A14F', '8' = '#CCCCCC', '9' = '#008080', '10' = '#C0D6E4','11' = '#FFD700','12' = '#A3ACB9','13' = '#F28E2B','14' = '#B60A1C','15' = '#8A494D'))##+theme(text=element_text(size=16,  family="Arial"))
dev.copy(pdf,paste0("../output/plots/", currentjob,"_13.default_cluster_NUMBERS_FILT.UMAP_final.pdf"), width=8, height=6, paper='special')
dev.off()

# Assign new IDs based on scCatch annotation results - Using Seurat Markers
#renumbered clusters from 0to13 
#fil.cluster.ids <- c("4","5","7","0","8","1","12","9","2","6","11","10","13","3")
f

# fil.cluster.ids <- c("4","5","2","7","1","9","12","8","11","3","6","13","10","15","14")
# Idents(rnaobjects.qcFilt) <- "seurat_clusters"
# names(fil.cluster.ids) <- levels(rnaobjects.qcFilt)
# rnaobjects.qcFilt <- RenameIdents(rnaobjects.qcFilt, fil.cluster.ids)
# DimPlot(rnaobjects.qcFilt, reduction = "umap", label = TRUE,group.by = "ident", pt.size = 0.5,cols = c('5' = '#FF9888', '1' = '#C3CE3D', '7' = '#8175AA', '2' = '#DAB6AF', '10' = '#FFD700', '3' = '#023858', '4' = '#D37295','9' = '#FFC966', '6' ='#008080','11' = '#59A14F', '8' = '#B15928','12' ='#8CC2CA',  '13' = '#C0D6E4',  '14' = '#F28E2B',  '15' = '#A3ACB9'))#+theme(text=element_text(size=12, family="Arial"))

# #fil.cluster.ids <- c("R4","R5","R2","R7","R1","R9","R12","R8","R11","R3","R6","R13","R10","R15","R14")
# Idents(rnaobjects.qcFilt) <- "seurat_clusters"
# names(fil.cluster.ids) <- levels(rnaobjects.qcFilt)
# rnaobjects.qcFilt <- RenameIdents(rnaobjects.qcFilt, fil.cluster.ids)
# 
# 
# DimPlot(rnaobjects.qcFilt, reduction = "umap", label = TRUE,group.by = "ident", pt.size = 0.5,cols = c('R5' = '#FF9888', 'R1' = '#C3CE3D', 'R7' = '#8175AA', 'R2' = '#DAB6AF', 'R10' = '#FFD700', 'R3' = '#023858', 'R4' = '#D37295','R9' = '#FFC966', 'R6' ='#008080','R11' = '#59A14F', 'R8' = '#B15928','R12' ='#8CC2CA',  'R13' = '#C0D6E4',  'R14' = '#F28E2B',  'R15' = '#A3ACB9'))#+theme(text=element_text(size=12, family="Arial"))
# 
# 
# dev.copy(pdf,paste0("../output/plots/", currentjob,"_13.clusternames_FILT.UMAP_final.pdf"), width=8, height=6, paper='special')
# dev.off()
rnaobjects.qcFilt$name_number_seurat_clusters<-rnaobjects.qcFilt$seurat_clusters
# Add in meta-data
rnaobjects.qcFilt <- AddMetaData(rnaobjects.qcFilt, metadata =rnaobjects.qcFilt@active.ident, col.name = 'cell_type')




```






```{r markers filtered}
FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("OTP"), reduction = "umap")
dev.copy(pdf,paste0("../output/plots/", currentjob,"featureplots_1.pdf"), height=5, width=8)
dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("ISL1"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_2.pdf"), height=5, width=8)
# dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("SIX3","SIX6"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_3.pdf"), height=5, width=8)
# dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("DLX1","DLX2"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_4.pdf"), height=5, width=8)
# dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("DLX5","DLX6"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_5.pdf"), height=5, width=8)
# dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("NKX2-1","LHX2"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_7.pdf"), height=5, width=8)
# dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("ASCL1","GNRH1"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_8.pdf"), height=5, width=8)
# dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("SP9","ARX"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_9.pdf"), height=5, width=8)
# dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("FANCI","LHX2"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_10.pdf"), height=5, width=8)
# dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("DCX","NES","VIM"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_11.pdf"), height=5, width=8)
# dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("TUBB3"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_12.pdf"), height=5, width=8)
# dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("SOX2","DHCR24","SOX21"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_13.pdf"), height=5, width=8)
# dev.off()
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("PODXL","DHCR24","VCAN"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_14.pdf"), height=5, width=8)
# dev.off()
# 
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("FOXD3-AS1","TUBB4B","PRDX1"), reduction = "umap")
# 
# dev.copy(pdf,paste0(currentjob,"featureplots_15.pdf"), height=5, width=8)
# dev.off()
# 
# FeaturePlot(object = NeuroDiffParacet.qcFilt, features = c("SMC4","CD24","FOXG1","IGFBP5"), reduction = "umap")
# dev.copy(pdf,paste0(currentjob,"featureplots_15.pdf"), height=5, width=8)
# dev.off()
```

#Heatmap Top10 genes per Cluster

```{r Heatmap: Top10 genes per Cluster2, message=FALSE, warning=FALSE}

# setting slim.col.label to TRUE will print just the cluster IDS instead of
# every cell name
top10 <- NeuroDiffParacet.qcFilt.markers %>% group_by(cluster) %>% top_n(n = 10, wt = avg_log2FC)
DoHeatmap(NeuroDiffParacet.qcFilt, features = top10$gene, label = FALSE) 
pdf(file = paste0("../output/plots/", currentjob,"_6_FILT.heatmap.Top10-cluster-markers.pdf"), width=12, height=14, paper='special')
DoHeatmap(NeuroDiffParacet.qcFilt, features = top10$gene, label = FALSE) 
dev.off()

```





```{r scCatch Annotation using seurat}
# Assign new IDs based on scCatch annotation results - Using Seurat Markers
#fil.cluster.ids <- c("ESC1","ESC2","Astrocyte","ESC3","NSC1","ESC4","Neuron","NPC1","ESC5","Neural_stemscells","Neuron","MicroglialCells","Astrocyte1","ESC6")
#Idents(NeuroDiffParacet.qcFilt) <- "seurat_clusters"
#names(fil.cluster.ids) <- levels(NeuroDiffParacet.qcFilt)
#NeuroDiffParacet.qcFilt <- RenameIdents(NeuroDiffParacet.qcFilt, fil.cluster.ids)

# Make UMAP with labels from scCatch
# plot9 <- DimPlot(NeuroDiffParacet.qcFilt, reduction = "umap", label = TRUE, pt.size = 0.5) + NoLegend()
# HoverLocator(plot = plot9, information = FetchData(NeuroDiffParacet.qcFilt, vars = c("ident", "PC_1", "nFeature_RNA")))
# #pdf(file = paste0(currentjob,"_12.FILT.UMAP-scCatch-II.pdf"), width=8, height=5, paper='special')
# DimPlot(NeuroDiffParacet.qcFilt, reduction = "umap", label = TRUE, pt.size = 0.5,cols = c('0' = '#D37295', '1' = '#FF9888', '2' = '#8175AA', '3' = '#DAB6AF','4' = '#FFC966','5' = '#C3CE3D', '6' = '#8CC2CA', '7' = '#59A14F', '8' = '#CCCCCC', '9' = '#008080', '10' = '#C0D6E4','11' = '#FFD700','12' = '#A3ACB9','13' = '#F28E2B','14' = '#B60A1C','15' = '#8A494D','16' = '#C8D0D9'))
# dev.copy(pdf,paste0(currentjob,"_12.FILT.UMAP-scCatch-II.pdf"), width=8, height=6, paper='special')
# dev.off()
# 
# DimPlot(NeuroDiffParacet.qcFilt, reduction = "umap", label = TRUE, pt.size = 0.5,cols = c('0' = '#D37295', '1' = '#FF9888', '2' = '#8175AA', '3' = '#DAB6AF','4' = '#FFC966','5' = '#C3CE3D', '6' = '#8CC2CA', '7' = '#59A14F', '8' = '#CCCCCC', '9' = '#008080', '10' = '#C0D6E4','11' = '#FFD700','12' = '#A3ACB9','13' = '#F28E2B','14' = '#B60A1C','15' = '#8A494D','16' = '#C8D0D9'))

# Add in meta-data
#NeuroDiffParacet.qcFilt <- AddMetaData(NeuroDiffParacet.qcFilt, metadata =NeuroDiffParacet.qcFilt@active.ident, col.name = 'cell_type')



```

```{r QCmetric - Violin plot}

# Re-visit QC metrics per cell-type
table(Idents(NeuroDiffParacet.qcFilt))
prop.table(table(Idents(NeuroDiffParacet.qcFilt)))
# FILT VlnPlot counts, features and % mit grouped by cell-type
VlnPlot(NeuroDiffParacet.qcFilt, c("nCount_RNA"), pt.size = 0.1, ncol = 1, group.by = "cell_type") + NoLegend()
dev.copy(pdf,paste0("../output/qc/", currentjob,"_13.FILT.nCount-cell-type.pdf"), width=8, height=6, paper='special')
dev.off()
VlnPlot(NeuroDiffParacet.qcFilt, c("nFeature_RNA"), pt.size = 0.1, ncol = 1, group.by = "cell_type") + NoLegend()
dev.copy(pdf,paste0("../output/qc/", currentjob,"_13.FILT.nFeature-cell-type.pdf"), width=8, height=6, paper='special')
dev.off()
VlnPlot(NeuroDiffParacet.qcFilt, c("percent.mt"), pt.size = 0.1, ncol = 1, group.by = "cell_type") + NoLegend()
dev.copy(pdf,paste0("../output/qc/", currentjob,"_13.FILT.percent.mt-cell-type.pdf"), width=8, height=6, paper='special')
dev.off()

# FILT scatter feature vs % mit
FeatureScatter(NeuroDiffParacet.qcFilt, "nCount_RNA", "nFeature_RNA") + ggtitle(label = "Scatter: feature vs % mt | Filtered data")
dev.copy(pdf,paste0("../output/qc/", currentjob,"_13.FILT.scatter.feat.vs.mt.pdf"), width=8, height=5, paper='special')
dev.off()

# Calculate the average expression of all cells within a cluster
cluster.averages.FILT <- AverageExpression(NeuroDiffParacet.qcFilt, assays = "RNA")
cluster.averages.FILT
head(cluster.averages.FILT[["RNA"]][, 1:3])

# Need to replace spaces " " with "_" so that ggplot does not fail
cluster.averages.FILT.original <- levels(NeuroDiffParacet.qcFilt)
Idents(NeuroDiffParacet.qcFilt) <- gsub(pattern = " ", replacement = "_", x = Idents(NeuroDiffParacet.qcFilt))
cluster.averages.FILT.original <- gsub(pattern = " ", replacement = "_", x = cluster.averages.FILT.original)
levels(NeuroDiffParacet.qcFilt) <- cluster.averages.FILT.original
cluster.averages.FILT <- AverageExpression(NeuroDiffParacet.qcFilt, assays = "RNA", return.seurat = TRUE)
cluster.averages.FILT

# Scatterplot for clusters
CellScatter(cluster.averages.FILT, cell1 = "1", cell2 = "2")

# Heatmap Cluster Averages
DoHeatmap(NeuroDiffParacet.qcFilt, features = unlist(TopFeatures(NeuroDiffParacet.qcFilt[["pca"]], balanced = TRUE)), size = 3, draw.lines = FALSE)
dev.copy(pdf,paste0(currentjob,"_14.FILT.heatmap-TopFeat-cluster-averages.pdf"), width=8, height=10, paper='special')
dev.off()
dev.set(dev.next())
# Check
Neural_stem_cell <- subset(NeuroDiffParacet.qcFilt, idents = "1")
VlnPlot(Neural_stem_cell, c("nCount_RNA","nFeature_RNA","percent.mt"), pt.size = 0.1, ncol = 3, ) + NoLegend()
dev.copy(pdf,paste0("../output/qc/", currentjob,"_15.FILT.specialCellTest.pdf"), width=8, height=10, paper='special')
dev.off()
dev.set(dev.next())
# Output top variable features and analyze in e.g. GO
top20_Neural_stem_cell <- head(VariableFeatures(Neural_stem_cell), 20)
# Output list for copy paste into browser
# List of genes showed many involved in neurogenesis in shinyGO
cat(paste(noquote(top20_Neural_stem_cell),collapse=','), '\n')

```



```{r Cell-cycle genes!, echo=TRUE}
# Print and save cells in cell-cycle
# kable is pretty neat!
table(NeuroDiffParacet.qcFilt@meta.data$Phase) %>% 
  kable(caption = "Number of Cells in each Cell Cycle Stage", "markdown",col.names = c("Stage", "Count"), align = "c") %>% kable_styling() 

table(NeuroDiffParacet.qcFilt@meta.data$seurat_clusters) %>% 
  kable(caption = "Number of Cells in each Seurat clusters", "markdown",col.names = c("Stage", "Count"), align = "c") %>% kable_styling() 

table(NeuroDiffParacet.qcFilt@meta.data$seurat_clusters) %>% 
  kable(caption = "Number of Cells in each Seurat clusters", "html",col.names = c("Stage", "Count"), align = "c") %>% kable_styling()  %>% save_kable(file = paste0("../output/outputfiles/", currentjob,"_H.FILT.Cells-in-seurat_clusters.html"), self_contained = T)

table(NeuroDiffParacet.qcFilt@meta.data$Phase) %>% 
  kable(caption = "Number of Cells in each Cell Cycle Stage", "html",col.names = c("Stage", "Count"), align = "c") %>% kable_styling() %>% save_kable(file = paste0("../output/outputfiles/", currentjob,"_H.FILT.Cells-in-CellCycle.html"), self_contained = T)

```

```{r Quantification of Data , echo=TRUE}
#Quantification of gene expressed in how many cells?
#Number of cells in each cluster expressing genes
PrctCellExpringGene <- function(object, genes, group.by = "all"){
    if(group.by == "all"){
        prct = unlist(lapply(genes,calc_helper, object=object))
        result = data.frame(Markers = genes, Cell_proportion = prct)
        return(result)
    }

    else{        
        list = SplitObject(object, group.by)
        factors = names(list)

        results = lapply(list, PrctCellExpringGene, genes=genes)
        for(i in 1:length(factors)){
        results[[i]]$Feature = factors[i]
        }
        combined = do.call("rbind", results)
        return(combined)
    }
}

calc_helper <- function(object,genes){
    counts = object[['RNA']]@counts
    ncells = ncol(counts)
    if(genes %in% row.names(counts)){
    sum(counts[genes,]>0)/ncells
    }else{return(NA)}
}
NumCellsGene<-as.data.frame(PrctCellExpringGene(NeuroDiffParacet.qcFilt ,genes =c("SEMA4A","SEMA4D","LIN28A","TACC3","SYP","HCRT","WNT7B","EMX2","GNG8","LHX9","BCL11A","RAX","HESX1","HES5","POU5F1","FGF8","DLX5","SIX6","NKX2-1","NEUROG1","ASCL1","NEUROD1","GNRH1","ELAVL4","TH","DCX","HES6","NEFM","RELN","SPARCL1","FABP7","GNG11","FN1","SCG3","SCG2","ISL1", "NEUROD1"), group.by = "cell_type"))
write.csv(NumCellsGene,"_Number_of_cells_for_genes_in_Clusters.txt",sep="\t",quote = F, row.names = T)


cellNum =table(NeuroDiffParacet.qcFilt@meta.data$seurat_clusters, NeuroDiffParacet.qcFilt@meta.data$orig.ident)
cellNum
write.table(cellNum,"_cellsincluster_and_Cellsperday.txt",sep="\t")

cellNum =table(NeuroDiffParacet.qcFilt@meta.data$orig.ident, NeuroDiffParacet.qcFilt@meta.data$Phase)
cellNum
write.table(cellNum, "_cellsinphase_and_Cellsperday.txt",sep="\t")

cellNum =table(NeuroDiffParacet.qcFilt@meta.data$seurat_clusters, NeuroDiffParacet.qcFilt@meta.data$Phase)
cellNum
write.table(cellNum,"_cellsinphase_and_Cellspercelltype.txt",sep="\t")

```



```{r Save final 2/2 RDS file and count matrix!, message=FALSE, warning=FALSE, include=FALSE}
# RDS For cytotrace 
saveRDS(NeuroDiffParacet.qcFilt, paste0("../output/RDSFiles/", currentjob,"_for_cytotrace_NeuroDiffParacet.qcFilt.rds"))
#Save SCT count to matrix
write.csv(NeuroDiffParacet.qcFilt@assays$SCT@counts, paste0("../output/RDSFiles/", currentjob,"_endOfpipe_countMatrix-Filt.csv"))
```



```{r Cytotrace, echo=TRUE, message=FALSE, warning=FALSE}
#NeuroDiffParacet.qcFilt<-readRDS("../output/RDSFiles/NeuroDiffParacetFinal_for_cytotrace_NeuroDiffParacet.qcFilt.rds")

library(scRNAseq)
library(SingleCellExperiment)
library(reticulate)
library(CytoTRACE)
library(umap)
#BiocManager::install("SingleCellExperiment")
#BiocManager::install("scRNAseq")
Sys.setenv(RETICULATE_PYTHON="/usr/local/bin/python3.7")
#use_python(Sys.which("python"))
py_config()
reticulate::import("scanorama")
#sce_A <- readRDS("Merged_runs_endOfpipeFirstIteration.rds")
cytoT_A <- as.matrix(GetAssayData(NeuroDiffParacet.qcFilt, "counts"))
cytodf_list <- cytoT_A
results <- CytoTRACE(cytodf_list,enableFast = FALSE, ncores=6)


plotCytoTRACE(results,emb = cyto_UMAP, outputDir = "../output/plots/")
#DimPlot(results, reduction = 'umap')
#sce_A<- RunUMAP(results, dims = 1:30, verbose = FALSE)
cyto_UMAP <- NeuroDiffParacet.qcFilt@reductions[["umap"]]@cell.embeddings
plotCytoTRACE(
  cyto_obj = results,
  phenotype = NULL,
  gene = "HIST1H4C",
  colors = NULL,
  emb = cyto_UMAP,
  outputDir = "../output/plots/"
)


plotCytoGenes(results,
              numOfGenes = 10,
              outputDir = "../output/plots/"
)

```

```{r pseudotime trajectory SLingshot , echo=TRUE, message=FALSE, warning=FALSE}
#NeuroDiffParacet.qcFilt <- readRDS("Diff_Merged_Final_EndofPipe_Final_NeuroDiffParacet.qcFilt.rds")
#install.packages("rsample")
#install.packages("parsnip")
#install.packages("ranger")
library(slingshot)
library (viridis)
library(dplyr)
library(rsample)
library(parsnip)
library(ranger)
library(RColorBrewer)
#https://bustools.github.io/BUS_notebooks_R/slingshot.html
####
# Need to use DimPlot due to weird workflowr problem with PCAPlot that calls seu[[wflow.build]]
# and eats up memory. I suspect this is due to the sys.call() in 
# Seurat:::SpecificDimPlot. 
pdf(file = paste0("../output/plots/", currentjob,"_SLINGSHOT.PCAPLOT.SEURAT.pdf"), width=8, height=5, paper='special')

DimPlot(NeuroDiffParacet.qcFilt, reduction = "pca",
        group.by = "cell_type", pt.size = 0.5, label = TRUE, repel = TRUE,cols = c('0' = '#D37295', '1' = '#FF9888', '2' = '#8175AA', '3' = '#DAB6AF','4' = '#FFC966','5' = '#C3CE3D', '6' = '#8CC2CA', '7' = '#59A14F', '8' = '#CCCCCC', '9' = '#008080', '10' = '#C0D6E4','11' = '#FFD700','12' = '#A3ACB9','13' = '#F28E2B','14' = '#B60A1C','15' = '#8A494D','16' = '#C8D0D9')) 
dev.off()
dev.set(dev.next())


 DimPlot(NeuroDiffParacet.qcFilt, reduction = "umap",
        group.by = "seurat_clusters", pt.size = 0.5, label = TRUE, repel = TRUE,cols = c('0' = '#D37295', '1' = '#FF9888', '2' = '#8175AA', '3' = '#DAB6AF','4' = '#FFC966','5' = '#C3CE3D', '6' = '#8CC2CA', '7' = '#59A14F', '8' = '#CCCCCC', '9' = '#008080', '10' = '#C0D6E4','11' = '#FFD700','12' = '#A3ACB9','13' = '#F28E2B','14' = '#B60A1C','15' = '#8A494D','16' = '#C8D0D9'))
dev.copy(pdf,paste0("../output/plots/", currentjob,"_SLINGSHOT.UMAP.SEURAT.pdf"), width=14, height=10, paper='special')
dev.off()
names(NeuroDiffParacet.qcFilt@meta.data)

sds <- slingshot(Embeddings(NeuroDiffParacet.qcFilt, "umap"), clusterLabels = NeuroDiffParacet.qcFilt$seurat_clusters,lineages = list(),
                 start.clus = 9,end.clus = 4,  stretch = 2)

#' Assign a color to each cell based on some value
#' 
#' @param cell_vars Vector indicating the value of a variable associated with cells.
#' @param pal_fun Palette function that returns a vector of hex colors, whose
#' argument is the length of such a vector.
#' @param ... Extra arguments for pal_fun.
#' @return A vector of hex colors with one entry for each cell.
cell_pal <- function(cell_vars, pal_fun,...) {
  if (is.numeric(cell_vars)) {
    pal <- pal_fun(100, ...)
    return(pal[cut(cell_vars, breaks = 100)])
  } else {
    categories <- sort(unique(cell_vars))
    pal <- setNames(pal_fun(length(categories), ...), categories)
    return(pal[cell_vars])
  }
}

cell_colors <- cell_pal(NeuroDiffParacet.qcFilt$cell_type, brewer_pal("qual", "Set3"))
cell_colors_clust <- cell_pal(NeuroDiffParacet.qcFilt$seurat_clusters, hue_pal())

##What does the inferred trajectory look like compared to cell types

plot(reducedDim(sds), col = cell_colors, pch = 16, cex = 0.5, cols = c('0' = '#D37295', '1' = '#FF9888', '2' = '#8175AA', '3' = '#DAB6AF','4' = '#FFC966','5' = '#C3CE3D', '6' = '#8CC2CA', '7' = '#59A14F', '8' = '#CCCCCC', '9' = '#008080', '10' = '#C0D6E4','11' = '#FFD700','12' = '#A3ACB9','13' = '#F28E2B','14' = '#B60A1C','15' = '#8A494D','16' = '#C8D0D9'))
dev.copy(pdf,paste0("../output/plots/", currentjob,"_SLINGSHOT.UMAP.pdf"), width=14, height=10, paper='special')
dev.off()
plot(reducedDim(sds), col = cell_colors, pch = 16, cex = 0.5, cols = c('0' = '#D37295', '1' = '#FF9888', '2' = '#8175AA', '3' = '#DAB6AF','4' = '#FFC966','5' = '#C3CE3D', '6' = '#8CC2CA', '7' = '#59A14F', '8' = '#CCCCCC', '9' = '#008080', '10' = '#C0D6E4','11' = '#FFD700','12' = '#A3ACB9','13' = '#F28E2B','14' = '#B60A1C','15' = '#8A494D','16' = '#C8D0D9'))
lines(sds, lwd = 2, type = 'lineages', col = 'black')
dev.copy(pdf,paste0("../output/plots/", currentjob,"_SLINGSHOT.LINEAGES.SEURAT.pdf"), width=14, height=10, paper='special')
dev.off()
dev.set(dev.next())
plot(reducedDim(sds), col = cell_colors_clust, pch = 16, cex = 0.5, cols = c('0' = '#D37295', '1' = '#FF9888', '2' = '#8175AA', '3' = '#DAB6AF','4' = '#FFC966','5' = '#C3CE3D', '6' = '#8CC2CA', '7' = '#59A14F', '8' = '#CCCCCC', '9' = '#008080', '10' = '#C0D6E4','11' = '#FFD700','12' = '#A3ACB9','13' = '#F28E2B','14' = '#B60A1C','15' = '#8A494D','16' = '#C8D0D9'))
lines(sds, lwd = 2, type = 'lineages', col = 'grey')
dev.copy(pdf,paste0("../output/plots/", currentjob,"_SLINGSHOT.LINEAGES.SEURAT_cellcolorclusts.pdf"), width=14, height=10, paper='special')
dev.off()
dev.set(dev.next())
plot(reducedDim(sds), col = cell_colors, pch = 16, cex = 0.5,cols = c('0' = '#D37295', '1' = '#FF9888', '2' = '#8175AA', '3' = '#DAB6AF','4' = '#FFC966','5' = '#C3CE3D', '6' = '#8CC2CA', '7' = '#59A14F', '8' = '#CCCCCC', '9' = '#008080', '10' = '#C0D6E4','11' = '#FFD700','12' = '#A3ACB9','13' = '#F28E2B','14' = '#B60A1C','15' = '#8A494D','16' = '#C8D0D9'))
lines(sds, lwd = 2, col = 'darkblue')
dev.copy(pdf,paste0("../output/plots/", currentjob,"_SLINGSHOT.LINEAGES.SEURAT_TRAJECTORIES.pdf"), width=14, height=10, paper='special')
dev.off()
dev.set(dev.next())

nc <- 3
pt <- slingPseudotime(sds)
nms <- colnames(pt)
nr <- ceiling(length(nms)/nc)
pal <- viridis(100, end = 0.95)
par(mfrow = c(nr, nc))
for (i in nms) {
  colors <- pal[cut(pt[,i], breaks = 100)]
  plot(reducedDim(sds), col = colors, pch = 16, cex = 0.5, main = i)
  lines(sds, lwd = 2, col = 'grey', type = 'lineages')
}
dev.copy(pdf,paste0("../output/plots/", currentjob,"_SLINGSHOT.LINEAGES.SEURAT_pseudotime.curves.pdf"), width=14, height=10, paper='special')
dev.off()



##Differentially expressed genes

# Get top highly variable genes
top_hvg <- HVFInfo(object = NeuroDiffParacet.qcFilt, selection.method = 'sct')[1:10000, ] %>% 
  mutate(., bc = rownames(.)) %>% 
  arrange(desc(residual_variance)) %>% 
  top_n(1000, residual_variance) %>% 
  pull(bc)
# Prepare data for random forest
dat_use <- t(GetAssayData(NeuroDiffParacet.qcFilt, slot = "data")[top_hvg,])
dat_use_df <- cbind(slingPseudotime(sds)[,2], dat_use) # Do curve 2, so 2nd columnn
colnames(dat_use_df)[1] <- "pseudotime"
dat_use_df <- as.data.frame(dat_use_df[!is.na(dat_use_df[,1]),])
library(parsnip)
dat_split <- initial_split(dat_use_df)
dat_train <- training(dat_split)
dat_val <- testing(dat_split)
library(tidymodels)
library(janitor)
library(caret)

colnames(dat_train)

model <- rand_forest(mtry = 100, trees = 100, min_n = 15, mode = "regression") %>%
  set_engine("ranger", importance = "impurity", num.threads = 3) %>%
    fit(pseudotime ~., data = dat_train)


val_results <- dat_val %>% 
  mutate(estimate = predict(model, .[,-1]) %>% pull()) %>% 
  select(truth = pseudotime, estimate)
metrics(data = val_results, truth, estimate)

var_imp <- sort(model$fit$variable.importance, decreasing = TRUE)
top_genes <- names(var_imp)[1:6]
# Convert to gene symbol
#top_gene_name <- gns$gene_name[match(top_genes, gns$gene)]
top_gene_name <- dat_train$gene_name[match(top_genes, dat_train$gene)]
top_genes
par(mfrow = c(3, 2))
for (i in seq_along(top_genes)) {
  colors <- pal[cut(dat_use[,top_genes[i]], breaks = 100)]
  plot(reducedDim(sds), col = colors, 
       pch = 16, cex = 0.5, main = top_gene_name[i])
  lines(sds, lwd = 2, col = 'black', type = 'lineages',title ="top_genes")
}

sessionInfo()
```

```{r SAVE RDS_FINAL, echo=TRUE, message=FALSE, warning=FALSE}
write.csv(NeuroDiffParacet.qcFilt@assays$SCT@counts, paste0("../output/outputfiles/", currentjob,"_endOfpipe_countMatrix-Filt.csv"))
saveRDS(NeuroDiffParacet.qcFilt, paste0("../output/RDSFILES/", currentjob,"_EndofPipe_Final_NeuroDiffParacet.qcFilt.rds"))
```


You can retrieve the number of cells expressing a gene my_gene from a Seurat object my_object in this way:

```{r Seurat, echo=TRUE, message=FALSE, warning=FALSE}
#https://github.com/satijalab/seurat/issues/371
sum(GetAssayData(object = my_object, slot = "data")[my_gene,]>0)

#The percentage of cells expressing that gene would be:

sum(GetAssayData(object = my_object, slot = "data")[my_gene,]>0)/nrow(my_object@meta.data)

#The object my_object@meta.data can be used to eventually extract cells (rownames) with a specific Identity or belonging to a specific cluster if you performed that analysis before, etc. After you have obtained a subset of cells of interest (my_cells) you can obtain the same as above by passing them as colnames:

sum(GetAssayData(object = my_object, slot = "data")[my_gene,my_cells]>0)
sum(GetAssayData(object = my_object, slot = "data")[my_gene,my_cells]>0)/length(my_cells)
```



```{r singleR, echo=TRUE}
library(SingleR)
NeuroDiffParacet.anno <- NeuroDiffParacet.qcFilt

# Put normalized counts in an object for singleR
singler.rna <- GetAssayData(NeuroDiffParacet.qcFilt, assay = "RNA", slot = "data")

# Get some more memory 
rm(logged, info, info.nmads5, plot.data, plot9, NeuroDiffParacet, NeuroDiffParacet.data, NeuroDiffParacet.markers, NeuroDiffParacet.qc, NeuroDiffParacet3d)
gc()

# Datasets
hpca.se <- HumanPrimaryCellAtlasData()
bed.se <- BlueprintEncodeData()

# REF HumanPrimaryCellAtlas
common <- intersect(rownames(singler.rna), rownames(hpca.se))
hpca.se <- hpca.se[common,]
singler.rna <- singler.rna[common,]

pred.hpca <- SingleR(test = singler.rna, ref = hpca.se, 
    labels = hpca.se$label.fine, assay.type.ref = "logcounts")
table(pred.hpca$labels)

# Labels and plot
NeuroDiffParacet.anno <- NeuroDiffParacet.qcFilt
NeuroDiffParacet.anno[["SingleR.labels"]] <- pred.hpca$labels
DimPlot(
  NeuroDiffParacet.anno,
  pt.size = 0.5,
  reduction = "umap",
  group.by = "SingleR.labels"
)

# REF BlueprintEncodeData
singler.bed.rna <- GetAssayData(NeuroDiffParacet.qcFilt, assay = "RNA", slot = "data")
common <- intersect(rownames(singler.bed.rna), rownames(bed.se))
bed.se <- bed.se[common,]
singler.bed.rna <- singler.bed.rna[common,]
pred.bed <- SingleR(test = singler.bed.rna, ref = bed.se, 
    labels = bed.se$label.fine, assay.type.ref = "logcounts")
table(pred.bed$labels)
NeuroDiffParacet.anno[["SingleR.BED"]] <- pred.bed$labels

# Plots
at1 <- DimPlot(NeuroDiffParacet.anno,pt.size = 0.5,reduction = "umap",group.by = "SingleR.labels") + ggtitle(label = "Human Primary Cell Atlas")
at2 <- DimPlot(NeuroDiffParacet.anno,pt.size = 0.5,reduction = "umap",group.by = "SingleR.BED") + ggtitle(label = "Blueprint Encode Data")
at1 / at2
dev.copy(pdf,paste0("../output/plots/", currentjob,"_17.FILT.singleR-UMAP.pdf"), width=8, height=7, paper='special')
dev.off()
# Heatmaps
plotScoreHeatmap(pred.bed, max.labels = 15, main="Blueprint Encode Data")
dev.copy(pdf,paste0("../output/plots/", currentjob,"_16.FILT.singleR-HCA-labels-heatmap.pdf"), width=14, height=10, paper='special')
dev.off()
plotScoreHeatmap(pred.hpca, max.labels = 15, cellwidth=0.1,main="Human Primary cell Atlas")
dev.copy(pdf,paste0("../output/plots/", currentjob,"_16.2.FILT.singleR-HCA-labels-heatmap.pdf"), width=14, height=10, paper='special')
dev.off()

table(pred.bed$labels)
table(pred.hpca$labels)

```


```{r SAVE RDS_FINAL, echo=TRUE, message=FALSE, warning=FALSE}
saveRDS(NeuroDiffParacet.qcFilt, paste0("../output/RDSFILES/", currentjob,"_EndofPipe_Final_NeuroDiffParacet.qcFilt_WithPathways.rds"))

```


```{r Script citation info, echo=TRUE, message=FALSE, warning=FALSE}
# Ankush Sharma with contributions from Martin Falck
# If you use this script for your analysis, please acknowledge us and cite Seurat Version3 package
#Credits please cite manuscript if you use the script

```







