---
#==========================================================================================================
title: "Neuronal Differentiation Paracet scATACSeq Pipeline "
author: "Ankush Sharma"
date: "`r Sys.Date()`"
output: html_document
Description: This script analyzes the integrative single cell ATAC seq analysis  for D20 P100 and  D20 P200  and D20 ctrl sample of human embryonic stem cells differentiated for 20 days (more details in Samara et al ,2022 StarProtocol and Iscience)  and the analysis and resuts will appear in manuscript of Spildrejorde et al. 

#==========================================================================================================
---


<!-- ####################################   -->
<!--    #loading data and Quality Control   -->
<!-- ####################################   -->

```{r 1_library and work, include=FALSE}
#devtools::install_github("GreenleafLab/ArchR", ref="master", repos = BiocManager::repositories())
# Directory settings
currentjob <- "NeuroDiffParacet"
addArchRThreads(threads = 4) #Use thread =1 if running for first time , parallelizing sometimes throws error in some function. 
addArchRGenome("hg38")
library(BSgenome.Hsapiens.UCSC.hg38)
#genomeAnnotation <- createGenomeAnnotation(genome = BSgenome.Hsapiens.UCSC.hg38)
#genomeAnnotation
library(org.Hs.eg.db)


#library(renv)
#renv::init()
#renv::activate()
library(ArchR)
library(progress)
library(hdf5r)
library(HDF5Array)
library(TFBSTools)
library(hexbin)
library(Seurat)
set.seed(1)
library(TxDb.Hsapiens.UCSC.hg38.knownGene)
geneAnnotation <- createGeneAnnotation(TxDb = TxDb.Hsapiens.UCSC.hg38.knownGene, OrgDb = org.Hs.eg.db)
geneAnnotation <- createGeneAnnotation(
  TSS = geneAnnotation$TSS, 
  exons = geneAnnotation$exons, 
  genes = geneAnnotation$genes)

###color custom Palettes
ArchRPalettes$custom <-c('D20' = '#C0D6E4','D20.P100' = '#D37295', 'D20.P200' = '#8175AA')
sample_id_palette <- ArchRPalettes$custom


 ArchRPalettes$custom <-c( '1' = '#CCCCCC','2'= '#C3CE3D', '3'='#DAB6AF','4' = '#023858','5' = '#8175AA','6' ='#B15928','7' = '#FFC966','8' = '#FFD700','9' = '#59A14F','10' = '#8CC2CA','11' = '#C0D6E4','12' = '#F28E2B','13' = '#A3ACB9')
 Atac_cluster_palette_1 <- ArchRPalettes$custom
 
ArchRPalettes$custom <- c('C1' = '#2A7185','C2' = '#A64027', 'C3' = '#FBDF72', 'C4' = '#60824F', 'C5' = '#9CDFF0', 'C6' = '#CAB2D6', 'C7' = '#A5BE6A', 'C8' = '#A2A475',  'C9' = '#2F8AC4','C10' = '#C0D6E4','C11' = '#FFD700','C12' = '#A3ACB9','C13' = '#F28E2B','C14' = '#B60A1C','C15' = '#8A494D')
Atac_cluster_palette <- ArchRPalettes$custom

ArchRPalettes$custom <- c('P3' = '#8175AA', 'P6' = '#FFC966','P9' = '#8CC2CA','P5' = '#B15928', 'P8' = '#59A14F', 'P10' = '#C0D6E4', 'P4' = '#A3ECB9', 'P12' = '#F28E2B','P13' = '#B60A1C')
 unconstrained_palette<- ArchRPalettes$custom
 ArchRPalettes$custom <-c('P3' = '#8175AA','P9' = '#8CC2CA','P5' = '#B15928',  'P10' = '#C0D6E4', 'P13' = '#BEE9C0')
constrained_palette <- ArchRPalettes$custom

ArchRPalettes$custom <-c('D20' = '#C0D6E4','D20.P100' = '#D37295', 'D20.P200' = '#8175AA')
sample_id_palette <- ArchRPalettes$custom
```


```{r 2_Creating ARROW files,include=FALSE}

ArrowFiles <- createArrowFiles(
  inputFiles <-c("~/d20_ctrl/fragments.tsv.gz","~/d20_p100/fragments.tsv.gz","~/d20_p200/fragments.tsv.gz"),
  sampleNames = c("D20","D20.P100","D20.P200"),
  filterTSS = 3, #Dont set this too high because you can always increase later
  filterFrags = 1000, 
  addTileMat = TRUE,
  addGeneScoreMat = TRUE
)
ArrowFiles
```




#reloading projects

```{r Reloading Projects}
#OR load Projects IF entire code had been run once and user need to load the projects for further additional analysis

#paracetP1<-loadArchRProject("Save-paracetP1/")
#paacetP2<-loadArchRProject("Save-paracetP2/")
#paracetP3<-loadArchRProject("Save-paracetP3/")
#paracetP4<-loadArchRProject("Save-paracetP4/")
#paracetP5<-loadArchRProject("Save-paracetP5/")
```

```{r 3_doublet score,include=FALSE}
Sys.setenv('R_MAX_VSIZE'=1200000)
doubScores <- addDoubletScores(
    input = ArrowFiles,
    k = 10,             #Refers to how many cells near a "pseudo-doublet" to count.
    knnMethod = "UMAP", #Refers to the embedding to use for nearest neighbor search with doublet projection.
    LSIMethod = 1
)
doubScores
plotPDF(doubScores, name = "0_doubscores_k10_UMAP_LSI1.pdf", addDOC = TRUE)

```

```{r 4_Setting up project}

set.seed(1)
paracetP1 <- ArchRProject(
  ArrowFiles = ArrowFiles, 
  outputDirectory = "paracetP1",
  copyArrows = FALSE
  ) 
```

```{r 5_Setting up memory,include=FALSE}
getAvailableMatrices(paracetP1)
```


```{r 6_NAMES, include=FALSE}
##these steps are not neccesaary but dooesnot harm to run 
#access the cell names and sample names associated with each cell:
head(paracetP1$cellNames)
head(paracetP1$Sample)
#access the TSS Enrichment Scores for each cell
quantile(paracetP1$TSSEnrichment)
#subset the project numerically, for example taking the first 100 cells 
paracetP1[1:100, ]
#subset the project based on certain cell names
paracetP1[paracetP1$cellNames[1:100], ]
```

```{r plotting data1} 
#retreiving column by name , such as number of unique nuclear (i.e. non-mitochondrial) fragments per cell:
df <- getCellColData(paracetP1, select = "nFrags")
df

#performing operation of columns 
df <- getCellColData(paracetP1, select = c("log10(nFrags)", "nFrags - 1"))
df
 #standard scATAC-seq metrics for quality control of individual cells
df <- getCellColData(paracetP1, select = c("log10(nFrags)", "TSSEnrichment"))
df

#Number of unique nuclear fragments (log10) by the TSS enrichment score. This plot is key for identifying high quality cells. 
plot1 <- ggPoint(
    x = df[,1], 
    y = df[,2], 
    colorDensity = TRUE,
    continuousSet = "sambaNight",
    xlabel = "Log10 Unique Fragments",
    ylabel = "TSS Enrichment",
    xlim = c(log10(500), quantile(df[,1], probs = 0.99)),
    ylim = c(0, quantile(df[,2], probs = 0.99))
) + geom_hline(yintercept = 4, lty = "dashed") + geom_vline(xintercept = 3, lty = "dashed")

plot1
plotPDF(plot, name = "0_QC-Differentiation_TSS-vs-Frags.pdf", ArchRProj = paracetP1, addDOC = TRUE)

```

```{r Plotting Sample Statistics} 

plot1 <- plotGroups(
    ArchRProj = paracetP1, 
    groupBy = "Sample", 
    colorBy = "cellColData", 
    name = "TSSEnrichment",
    plotAs = "ridges",
    pal = sample_id_palette
   )
plot1


plot2 <- plotGroups(
    ArchRProj = paracetP1, 
    groupBy = "Sample", 
    colorBy = "cellColData", 
    name = "TSSEnrichment",
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    pal = sample_id_palette
   )
plot2


plot3 <- plotGroups(
    ArchRProj = paracetP1, 
    groupBy = "Sample", 
    colorBy = "cellColData", 
    name = "log10(nFrags)",
    plotAs = "ridges",
    pal = sample_id_palette
   )
plot3


plot4 <- plotGroups(
    ArchRProj = paracetP1, 
    groupBy = "Sample", 
    colorBy = "cellColData", 
    name = "log10(nFrags)",
    plotAs = "violin",
    alpha = 0.4,
    addBoxPlot = TRUE,
    pal = sample_id_palette
   )
plot4
plot1+plot2+plot3+plot4
plotPDF(plot1+plot2,plot3,plot4, name = "0_QC-Sample-Statistics.pdf", ArchRProj = paracetP1, addDOC = TRUE, width = 4, height = 4)

```


```{r Plotting Sample Fragment Size Distribution and TSS Enrichment Profiles} 

plot5 <- plotFragmentSizes(ArchRProj = paracetP1,pal = sample_id_palette)
plot5
plot6 <- plotTSSEnrichment(ArchRProj = paracetP1, pal = sample_id_palette)
plot6
plotPDF(plot5,plot6, name = "0_QC-Sample-FragSizes-TSSProfile.pdf", ArchRProj = paracetP1, addDOC = TRUE, width = 5, height = 5)
saveArchRProject(ArchRProj = paracetP1, outputDirectory = "Save-paracetP1", load = FALSE)
```
<!-- #################################### -->
<!--      End of Save Project 1           -->
<!--           Filter Doublet             -->
<!-- #################################### -->

```{r filter doublet function} 
#start fromhere now
###RUN THIS FUNCTION before running ARCHR FILTER step in order to avoid posix.ct numeric
filterDoublets <- function(ArchRProj = NULL, cutEnrich = 1, cutScore = -Inf, filterRatio = 1){

  fn <- unclass(lsf.str(envir = asNamespace("ArchR"), all = TRUE))
    for (i in seq_along(fn)) {
        tryCatch({
            eval(parse(text = paste0(fn[i], "<-ArchR:::", fn[i])))
        }, error = function(x) {
        })
    }

  .validInput(input = ArchRProj, name = "ArchRProj", valid = c("ArchRProj"))
  .validInput(input = cutEnrich, name = "cutEnrich", valid = c("numeric"))
  .validInput(input = cutScore, name = "cutScore", valid = c("numeric"))
  .validInput(input = filterRatio, name = "filterRatio", valid = c("numeric"))

  if(any(grepl("filterDoublets", names(ArchRProj@projectSummary)))){
    stop("Already ran filterDoublets on ArchRProject! Cannot be re-ran on an ArchRProject!")
  }

  df <- getCellColData(ArchRProj, c("Sample", "DoubletEnrichment", "DoubletScore"))
  splitDF <- split(seq_len(nrow(df)), as.character(df$Sample))

  cellsFilter <- lapply(splitDF, function(y){

    x <- df[y, ,drop = FALSE]

    n <- nrow(x)

    x <- x[order(x$DoubletEnrichment, decreasing = TRUE), ]
    
    if(!is.null(cutEnrich)){
      x <- x[which(x$DoubletEnrichment >= cutEnrich), ]
    } 
    
    if(!is.null(cutScore)){
      x <- x[which(x$DoubletScore >= cutScore), ]
    } 

    if(nrow(x) > 0){
      head(rownames(x), filterRatio * n * (n / 100000))
    }else{
      NULL
    }

  }) %>% unlist(use.names=FALSE)

  message("Filtering ", length(cellsFilter), " cells from ArchRProject!")
  tabRemove <- table(df[cellsFilter,]$Sample)
  tabAll <- table(df$Sample)
  samples <- unique(df$Sample)
  for(i in seq_along(samples)){
    if(!is.na(tabRemove[samples[i]])){
      message("\t", samples[i], " : ", tabRemove[samples[i]], " of ", tabAll[samples[i]], " (", round(100 * tabRemove[samples[i]] / tabAll[samples[i]], 1),"%)")
    }else{
      message("\t", samples[i], " : ", 0, " of ", tabAll[samples[i]], " (0%)")
    }
  }

  if(length(cellsFilter) > 0){
    
    ArchRProj@cellColData <- ArchRProj@cellColData[rownames(ArchRProj@cellColData) %ni% cellsFilter,,drop=FALSE]

  }
  
  ArchRProj

}
saveArchRProject(ArchRProj = paracetP1, outputDirectory = "Save-paracetP1", load = FALSE)
```


```{r filterDoublets} 
paracetP2 <- filterDoublets(paracetP1, filterRatio = 1)
#Differentiation_P0 <- filterDoublets(Differentiation_P0, filterRatio = 1.0)
#filter more cells from the ArchR Project, you would use a higher filterRatio. To see additional arguments that can be tweaked, try ?filterDoublets.
#<!-- #################################### -->
#<!--          #Dimensionality reduction   -->
#<!-- #################################### -->

paracetP2 <- addIterativeLSI(
    ArchRProj = paracetP2,
    useMatrix = "TileMatrix", 
    name = "IterativeLSI", 
    iterations = 10, 
    clusterParams = list( #See Seurat::FindClusters
      # 
      # resolution = c(0.1, 0.2, 0.3, 0.35, 0.40, 0.45, 0.5, 0.55, 0.6, 0.7, 0.8, 0.9, 1.0), 
        resolution = c(0.5),
        sampleCells = 10000,
        n.start = 10),
        LSIMethod = 2,
        varFeatures = 25000,
        dimsToUse = 1:30,
        force = TRUE,
   
    logFile = createLogFile("addIterativeLSI")
)


```

#Harmaony
```{r batch effect correction} 
library(harmony)
paracetP2 <- addHarmony(
    ArchRProj = paracetP2,
    reducedDims = "IterativeLSI",
    name = "Harmony",
    groupBy = "Sample"
)
```

#Clustering
```{r Clustering using seurat : resolution 0.6} 


paracetP2 <- addClusters(
    input = paracetP2,
    reducedDims = "IterativeLSI",
    method = "Seurat",
    metric = "pca",
    name = "Clusters",
    resolution = 0.7,
    force = TRUE
)
```

```{r final seurat cluster treshold 0.6} 
#HERE  RESOLUTION BASED ON ABOVE AAnalysis
paracetP2<- addClusters(
    input = paracetP2,
    reducedDims = "IterativeLSI",
    method = "Seurat",
    name = "Clusters",
    resolution = c(0.7),
    force = TRUE
)

```


```{r Confusion matrix} 
head(paracetP2$Clusters)
table(paracetP2$Clusters)
cM <- confusionMatrix(paste0(paracetP2$Clusters), paste0(paracetP2$Sample))
cM
```

```{r heatmap} 

library(pheatmap)

cM <- cM / Matrix::rowSums(cM)
p <- pheatmap::pheatmap(
    mat = as.matrix(cM), 
    color = paletteContinuous("whitePurple"), 
    border_color = "grey"
)
p
```

```{r UMAP of Interactive LSI} 
library(uwot)

paracetP2 <- addUMAP(
    ArchRProj = paracetP2, 
    reducedDims = "IterativeLSI", 
    name = "UMAP", 
    nNeighbors = 4, 
    minDist = 0.8, 
    metric = "cosine",
    force = TRUE
)

p1 <- plotEmbedding(ArchRProj = paracetP2, colorBy = "cellColData", name = "Sample", embedding = "UMAP",pal = sample_id_palette,rastr = FALSE,size=0.3)
p1

p2 <- plotEmbedding(ArchRProj = paracetP2, colorBy = "cellColData", name = "Clusters", embedding = "UMAP",pal = Atac_cluster_palette,rastr = FALSE,size=0.3)
p2
ggAlignPlots(p1, p2, type = "h")
plotPDF(p1,p2, name = "1_Plot-UMAP1-Sample-Clusters.pdf", ArchRProj = paracetP1, addDOC = TRUE, width = 11, height = 11,plotList = NULL)


```


```{r umap using COSINE harmony: DEFAULT } 
library(umap)

##cosine metric works well

paracetP2 <- addUMAP(
    ArchRProj = paracetP2, 
    reducedDims = "Harmony", 
    name = "UMAPHarmony", 
    nNeighbors = 8, 
    minDist = 0.2, 
    metric = "cosine",
    force = TRUE
)
p3 <- plotEmbedding(ArchRProj = paracetP2, colorBy = "cellColData", name = "Sample", embedding = "UMAPHarmony",pal = sample_id_palette, size=0.3,rastr=FALSE)
p3
p4 <- plotEmbedding(ArchRProj = paracetP2, colorBy = "cellColData", name = "Clusters", embedding = "UMAPHarmony",pal = Atac_cluster_palette,size=0.3,rastr=FALSE)
p4
ggAlignPlots(p3, p4, type = "h")
plotPDF(p1,p2,p3,p4, name = "1_Plot-UMAP2Harmony-Sample-Clusters_n4_dist0.8_.pdf", ArchRProj = paracetP2, addDOC = TRUE, width = 11, height = 11)

```

```{r marker ID for ATAC,include=FALSE} 

markersGS <- getMarkerFeatures(
   ArchRProj = paracetP2, 
   useMatrix = "GeneScoreMatrix", 
   groupBy = "Clusters",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  testMethod = "wilcoxon"
)

markerListGS <- getMarkers(markersGS, cutOff = "FDR <= 0.01 & Log2FC >= 0.1")
Cluster_marker_list_GS <- write.csv(markerListGS, "1_ATAC_cluster_marker_list_GS.csv", sep = "\t")
markerListGS
markersGS 
```

```{r marker list heatmap,include=FALSE}


markerGenes  <- c('SOX2','IRX2','RAX','LIX1','PAX6','HESX1','HES5','LRRC75A','FOXH1','CDH1','TDGF1','DNMT3B','EPCAM','PODXL','ID1','POU5F1','GAL','FOXG1','DPPA4','NR6A1','CD24','LRRC75A','ZIC2','FGF8','DLX6','DLX5','BMP4','GRIA1','SIX6','NKX2-1','NEUROG1','DLL1','ASCL1','NHLH1','NEUROD1','FEZF2','L1CAM','INA','GRIA2','GNRH1','INSM1','ELAVL4','ISL1','TH','DCX','TAGLN3','SCG2','ELAVL3','PCSK2','MAP6','HES6','DLL3','TLX3','NEUROD4','SYP','STMN2','POU2F2','TUBB3','SNAP25','NSG2','GAP43','NEFM','NEFL','RET','NTRK3','RELN','NTRK1','DLX2','DLX1','GAD1','ARX','SOX3','SYT4','SPARCL1','OTX2','LMO1','FABP7','DCT','VEPH1','PRRX1','CDK6','GNG11','HES1','TYMS','PCNA','CDK1','DLX6-AS1','TUBB4B','SELENOP','ID2','CRABP1','APOE','LDHA','FZD5','KRT18','FGF17','ASCL1','REST','GNRH1','GAD1','GAD2','ARX','CDKN1C','TAGLN','NANOG','KIF15','SIX3','PHC1','TUBB4A','NRXN1','CDH7','ROBO1','NCAM1','INA','NEFL','NEFM','TUBB3','TIBB4B','FN1','CDH2','NES','VIM','ACTG1','THY-1','EPHA1','CDH3','PAMBR1','RELN','SEMA4A','SEMA4D','TH')
ArchRPalettes$custom <- c('1' = '#B60A1C','2' = '#AECDDF')
customcolor_heatmaps <- ArchRPalettes$custom
heatmapGS <- markerHeatmap(
  seMarker = markersGS, 
  cutOff = "FDR <= 0.01 & Log2FC >= 0.1", 
  labelMarkers = markerGenes,
  transpose = TRUE,
  
)

ComplexHeatmap::draw(heatmapGS, heatmap_legend_side = "bot", annotation_legend_side = "bot")
plotPDF(heatmapGS, name = "1_RNAseq-markers_GeneScores-Marker_list-Heatmap-0.1-C", width = 8, height = 6, ArchRProj = paracetP2, addDOC = TRUE)


p2 <- lapply(p, function(x){
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
)
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

markerGenes_atac  <- c('LOC105369438','LOC101927668','SCGB3A2','L1TD1','LNCPRESS2','RYR2','ELL2','ZFP42','TRG-AS1','LINC00678','RFTN2','ASTN2','PVT1','GPC5','UST-AS1','LNCPRESS2','TPRG1','RAB17','HDAC2-AS2','TRG-AS1','C3','SH2D3A','CLDN9','KLK9','PPP1R14A','SMIM24','KLK8','ARHGAP45','SPINT2','DPRX','VENTX','KLK8','IRX4','ARHGAP45','LINC01962','FAM83H','KLK9','MMP23A','TMEM102','CDX2','MIR2355','PTPRN2','GSE1','LINC01315','CCDC33','MAP6','MSI2','INSM1','MIR7158','EPHA4','FZD8','RBM5-AS1','NSF','MIR770','LOC101927557','MIR2392','PRDM16','PRTG','MAP1B','NSFP1','MSI2','FAM204A','MIR99AHG','CDH2','PCDH9','PLEKHG1','ZFHX4','SLIT2','GLI3','MAP1B','GLIS3','FIGN','FIGN','AATF','LOC101927557','AKAP6','MIR770','MBIP','MSI2','PRTG','NPAS3','PRDM16','SOX6')
heatmapGS <- plotMarkerHeatmap(
  seMarker = markersGS, 
  cutOff = "FDR <= 0.01 & Log2FC >= 0.1", 
  labelMarkers = markerGenes_atac,
  transpose = TRUE,
  
)

ComplexHeatmap::draw(heatmapGS, heatmap_legend_side = "bot", annotation_legend_side = "bot")
plotPDF(heatmapGS, name = "1_ATACmarkers_GeneScores-Marker_list-Heatmap-0.1-C", width = 8, height = 6, ArchRProj = paracetP2, addDOC = TRUE)


p2 <- lapply(p, function(x){
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
)
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))


Transcription_Factors <- c('ARX','ASCL1','DLX1','DLX2','DLX5','DLX6','EMX2','FOXD3','FOXG1','FOXH1','HES1','HES5','HES6','HESX1','ID1','ID2','INSM1','IRX2','ISL1','MYC','NANOG','NEUROD1','NEUROD4','NEUROG1','NHLH1','NKX2-1','NR6A1','OTX2','PAX6','POU2F2','POU3F1','POU5F1','PRRX1','RAX','REST','SIX3','SIX6','SOX11','SOX21','SOX2','SOX3','TLX3','ZIC2','LHX2','NFIB','POU3F3','SOX5','ZIC4','ZEB1','NR2F2','ARX','ASCL1','DLX1','DLX2','DLX5','DLX6','EMX2','FOXD3','FOXG1','FOXH1','HES1','HES5','HES6','HESX1','ID1','ID2','INSM1','IRX2','ISL1','MYC','NANOG','NEUROD1','NEUROD4','NEUROG1','NHLH1','NKX2-1','NR6A1','OTX2','PAX6','POU2F2','POU3F1','POU5F1','PRRX1','RAX','REST','SIX3','SIX6','SOX11','SOX21','SOX2','SOX3','TLX3','ZIC2','LHX2','NFIB','POU3F3','SOX5','ZIC4','ZEB1','NR2F2')

heatmapGS <- plotMarkerHeatmap(
  seMarker = markersGS, 
  cutOff = "FDR <= 0.01 & Log2FC >= 0.1", 
  labelMarkers = Transcription_Factors,
  transpose = TRUE,
  
)

ComplexHeatmap::draw(heatmapGS, heatmap_legend_side = "bot", annotation_legend_side = "bot")
plotPDF(heatmapGS, name = "1_Transcription factors_markers_GeneScores-Marker_list-Heatmap-0.1-C", width = 8, height = 6, ArchRProj = paracetP2, addDOC = TRUE)


p2 <- lapply(p, function(x){
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
)
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

markerList_marker_genes <- getMarkers(markersGS, cutOff = "FDR <= 0.01 & Log2FC >= 0.1")
Cluster_marker_list_PM <- write.csv(markerListGS, "1_ATAC_genescores_plot_marker_genes_TopClusterMarkers_.csv", sep = "\t")

```
#markergenesPlotembeddinh
```{r vizualise marker genes on an embedding,include=FALSE} 
#make UMAP plots of marker genes

markerGenes_atac  <- c('LOC105369438','LOC101927668','MAP1B','GLIS3','FIGN','FIGN','AATF','LOC101927557','AKAP6','MIR770','MBIP','MSI2','PRTG','NPAS3','PRDM16','SOX6')
#markerGenes_atac_test  <- c('LOC105369438','LOC101927668','SCGB3A2','L1TD1','LNCPRESS2')
p <- plotEmbedding(
    ArchRProj = paracetP2,
    reducedDims = "IterativeLSI",
    colorBy = "GeneScoreMatrix", 
    name = markerGenes_atac, 
    embedding = "UMAP",
    quantCut = c(0.01, 0.95),
    size=0.3,
    imputeWeights = NULL
)


p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(plotList = p, 
    name = "1_Plot-UMAP-Marker-Genes-LSIumap-TopMarkersAND_atac_Markers-Imputation.pdf", 
    ArchRProj = paracetP2, 
    addDOC = TRUE, width = 5, height = 5)


markerGenes  <- c('SOX2','IRX2','RAX','LIX1','PAX6','HESX1','HES5','SIX3','PHC1','TUBB4A','NRXN1','CDH7','ROBO1','NCAM1','INA','NEFL','NEFM','TUBB3','FN1','CDH2','NES','VIM','ACTG1','EPHA1','CDH3','RELN','SEMA4A','SEMA4D','TH')

p <- plotEmbedding(
    ArchRProj = paracetP2,
    reducedDims = "IterativeLSI",
    colorBy = "GeneScoreMatrix", 
    name = markerGenes, 
    embedding = "UMAP",
    quantCut = c(0.01, 0.95),
    size=0.3,
    imputeWeights = NULL
)

p$POU5F1

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(plotList = p, 
    name = "1_Plot-UMAP-Marker-Genes-LSIumap-TopMarkersAND_rna_Markers-Imputation.pdf", 
    ArchRProj = paracetP2, 
    addDOC = TRUE, width = 5, height = 5)

Transcription_Factors <- c('ARX','ASCL1','DLX1','DLX2','DLX5','DLX6','EMX2','FOXD3','FOXG1','FOXH1','HES1','HES5','HES6','HESX1','ID1','ID2','INSM1','IRX2','ISL1','MYC','NANOG','NEUROD1','NEUROD4','NEUROG1','NHLH1','NKX2-1','NR6A1','OTX2','PAX6','POU2F2','POU3F1','POU5F1','PRRX1','RAX','REST','SIX3','SIX6','SOX11','SOX21','SOX2','SOX3','TLX3','ZIC2','LHX2','NFIB','POU3F3','SOX5','ZIC4','ZEB1','NR2F2','ARX','ASCL1','DLX1','DLX2','DLX5','DLX6','EMX2','FOXD3','FOXG1','FOXH1','HES1','HES5','HES6','HESX1','ID1','ID2','INSM1','IRX2','ISL1','MYC','NANOG','NEUROD1','NEUROD4','NEUROG1','NHLH1','NKX2-1','NR6A1','OTX2','PAX6','POU2F2','POU3F1','POU5F1','PRRX1','RAX','REST','SIX3','SIX6','SOX11','SOX21','SOX2','SOX3','TLX3','ZIC2','LHX2','NFIB','POU3F3','SOX5','ZIC4','ZEB1','NR2F2')

p <- plotEmbedding(
    ArchRProj = paracetP2,
    reducedDims = "IterativeLSI",
    colorBy = "GeneScoreMatrix", 
    name = Transcription_Factors, 
    embedding = "UMAP",
    quantCut = c(0.01, 0.95),
    size=0.3,
    imputeWeights = NULL
)

p$POU5F1

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(plotList = p, 
    name = "1_Plot-UMAP-Marker-Genes-LSIumap-TopMarkersAND_Transcriptionfactors-Imputation.pdf", 
    ArchRProj = paracetP2, 
    addDOC = TRUE, width = 5, height = 5)

```
#plotembedding genescore
```{r marker gene computation with ,include=FALSE} 
#make UMAP plots of marker genes

paracetP2 <- addImputeWeights(paracetP2)

markerGenes_atac  <- c('LOC105369438','LOC101927668','SCGB3A2','L1TD1','LNCPRESS2','RYR2','ELL2','ZFP42','TRG-AS1','LINC00678','RFTN2','ASTN2','PVT1','GPC5','UST-AS1','LNCPRESS2','TPRG1','RAB17','HDAC2-AS2','TRG-AS1','C3','SH2D3A','CLDN9','KLK9','PPP1R14A','SMIM24','KLK8','ARHGAP45','SPINT2','DPRX','VENTX','KLK8','IRX4','ARHGAP45','LINC01962','FAM83H','KLK9','MMP23A','TMEM102','CDX2','MIR2355','PTPRN2','GSE1','LINC01315','CCDC33','MAP6','MSI2','INSM1','MIR7158','EPHA4','FZD8','RBM5-AS1','NSF','MIR770','LOC101927557','MIR2392','PRDM16','PRTG','MAP1B','NSFP1','MSI2','FAM204A','MIR99AHG','CDH2','PCDH9','PLEKHG1','ZFHX4','SLIT2','GLI3','MAP1B','GLIS3','FIGN','FIGN','AATF','LOC101927557','AKAP6','MIR770','MBIP','MSI2','PRTG','NPAS3','PRDM16','SOX6')

p <- plotEmbedding(
    ArchRProj = paracetP2,
    reducedDims = "IterativeLSI",
    colorBy = "GeneScoreMatrix", 
    name = markerGenes_atac, 
    embedding = "UMAP",
    size=0.3,
    imputeWeights = getImputeWeights(paracetP2)
)


p$SERPINB2

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
       axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(plotList = p, 
    name = "1_Plot-UMAP-ATAC_Marker-Genes-MAGIC-TopMarkersANDatacMarkers-Imputation.pdf", 
    ArchRProj = paracetP2, 
    addDOC = TRUE, width = 5, height = 5)



markerGenes  <- c("POU5F1", "NANOG", "TDGF1", "GAL", "PODXL", "ID1", "MYC", "PAX6", "DNMT3B", "LIN28A", "FOXD3-AS1", "CD24", "EPCAM", "DPPA4", "FZD2", "NR6A1", "SOX21", "ZIC2", "SOX11", "HESX1", "MYLK", "IGFBP5", "FZD5", "FGF8", "SOX2", "RAX", "LIX1", "FEZF2", "POU3F1", "POU3F2", "NKX2-1", "ASCL1", "NEUROG1", "NEUROD1", "ISL1", "BSX", "HMX2", "OTP", "GSX1", "SF1", "GNRH1", "GRIA2", "L1CAM", "NHLH1", "ELAVL4", "SIX3", "SIX6", "OTX2", "TH", "INSM1", "ARX", "DBX1", "DCX", "DLL1", "DLX1", "DLX2", "DLX5", "DLX6", "GAD1", "GRIA1", "INA", "KIF19", "NEFL", "NEFM", "NEUROD4", "NSG2", "POU2F2", "RELN", "SCG2", "SIM1", "SIM2", "SNAP25", "SP9", "SST", "STMN2", "SYP", "SYT4", "TAC1", "TAGLN3", "TLX3", "NTRK1", "NTRK3", "PCSK2", "RET", "DLL3", "GAD2", "GAP43", "REST", "SOX3", "TUBB3", "HES1", "BMP4", "KLF15", "CDK1", "TYMS", "ANLN", "HES6", "LMO1", "ELAVL3", "PRRX1", "OTX2", "PCNA", "SPARCL1", "MAP6", "KRT18", "DDIT4", "PAMR1", "FOXH1", "FABP7", "RBFOX3", "SOX6", "CDH7", "CDH12", "PHC1", "PHC2", "YAF2", "CDH1", "CDH2", "GNG2", "EMX2", "CDK6", "TAGLN", "GNG8", "KLF11", "LHX9", "DCT", "VEPH1", "GNG3", "GNG11", "TRIM71", "SOX3", "KRT19", "NSG1", "JHY")
p <- plotEmbedding(
    ArchRProj = paracetP2,
    reducedDims = "IterativeLSI",
    colorBy = "GeneScoreMatrix", 
    name = markerGenes, 
    embedding = "UMAP",
    size=0.3,
    imputeWeights = getImputeWeights(paracetP2)
)


p$SERPINB2

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
       axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(plotList = p, 
    name = "1_Plot-UMAP-RNA_Marker-Genes-MAGIC-TopMarkersANDSCRNAMarkers-Imputation.pdf", 
    ArchRProj = paracetP2, 
    addDOC = TRUE, width = 5, height = 5)



Transcription_Factors <- c('ARX','ASCL1','DLX1','DLX2','DLX5','DLX6','EMX2','FOXD3','FOXG1','FOXH1','HES1','HES5','HES6','HESX1','ID1','ID2','INSM1','IRX2','ISL1','MYC','NANOG','NEUROD1','NEUROD4','NEUROG1','NHLH1','NKX2-1','NR6A1','OTX2','PAX6','POU2F2','POU3F1','POU5F1','PRRX1','RAX','REST','SIX3','SIX6','SOX11','SOX21','SOX2','SOX3','TLX3','ZIC2','LHX2','NFIB','POU3F3','SOX5','ZIC4','ZEB1','NR2F2','ARX','ASCL1','DLX1','DLX2','DLX5','DLX6','EMX2','FOXD3','FOXG1','FOXH1','HES1','HES5','HES6','HESX1','ID1','ID2','INSM1','IRX2','ISL1','MYC','NANOG','NEUROD1','NEUROD4','NEUROG1','NHLH1','NKX2-1','NR6A1','OTX2','PAX6','POU2F2','POU3F1','POU5F1','PRRX1','RAX','REST','SIX3','SIX6','SOX11','SOX21','SOX2','SOX3','TLX3','ZIC2','LHX2','NFIB','POU3F3','SOX5','ZIC4','ZEB1','NR2F2')


p <- plotEmbedding(
    ArchRProj = paracetP2,
    reducedDims = "IterativeLSI",
    colorBy = "GeneScoreMatrix", 
    name = Transcription_Factors, 
    embedding = "UMAP",
    size=0.3,
    imputeWeights = getImputeWeights(paracetP2)
)


p$SERPINB2

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
       axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(plotList = p, 
    name = "1_Plot-UMAP-RNA_Marker-Genes-MAGIC-TopMarkersANDTRANSCRIOTIONFACTORS-Imputation.pdf", 
    ArchRProj = paracetP2, 
    addDOC = TRUE, width = 5, height = 5)

```

#plotbrowser

```{r Track blotting with ArchR Browser,include=FALSE} 




markerGenes  <- c('LOC105369438','LOC101927668','SCGB3A2','L1TD1','LNCPRESS2','RYR2','ELL2','ZFP42','TRG-AS1','LINC00678','RFTN2','ASTN2','PVT1','GPC5','UST-AS1','LNCPRESS2','TPRG1','RAB17','HDAC2-AS2','TRG-AS1','C3','SH2D3A','CLDN9','KLK9','PPP1R14A','SMIM24','KLK8','ARHGAP45','SPINT2','DPRX','VENTX','KLK8','IRX4','ARHGAP45','LINC01962','FAM83H','KLK9','MMP23A','TMEM102','CDX2','MIR2355','PTPRN2','GSE1','LINC01315','CCDC33','MAP6','MSI2','INSM1','MIR7158','EPHA4','FZD8','RBM5-AS1','NSF','MIR770','LOC101927557','MIR2392','PRDM16','PRTG','MAP1B','NSFP1','MSI2','FAM204A','MIR99AHG','CDH2','PCDH9','PLEKHG1','ZFHX4','SLIT2','GLI3','MAP1B','GLIS3','FIGN','FIGN','AATF','LOC101927557','AKAP6','MIR770','MBIP','MSI2','PRTG','NPAS3','PRDM16','SOX6')

p <- plotBrowserTrack(
    ArchRProj = paracetP2, 
    groupBy = "Clusters", 
    geneSymbol = markerGenes, 
    upstream = 50000,
    downstream = 50000,
    pal=Atac_cluster_palette
)
p

grid::grid.newpage()
plotPDF(plotList = p, 
    name = "1_Plot-Tracks-Marker-Genes-Cluster-specific-markers-from-ATAC.pdf", 
   ArchRProj = paracetP2, 
    addDOC = TRUE, width = 5, height = 5)
markerGenes  <- c('SOX2','IRX2','RAX','LIX1','PAX6','HESX1','HES5','LRRC75A','FOXH1','CDH1','TDGF1','DNMT3B','EPCAM','PODXL','ID1','POU5F1','GAL','FOXG1','DPPA4','NR6A1','CD24','LRRC75A','ZIC2','FGF8','DLX6','DLX5','BMP4','GRIA1','SIX6','NKX2-1','NEUROG1','DLL1','ASCL1','NHLH1','NEUROD1','FEZF2','L1CAM','INA','GRIA2','GNRH1','INSM1','ELAVL4','ISL1','TH','DCX','TAGLN3','SCG2','ELAVL3','PCSK2','MAP6','HES6','DLL3','TLX3','NEUROD4','SYP','STMN2','POU2F2','TUBB3','SNAP25','NSG2','GAP43','NEFM','NEFL','RET','NTRK3','RELN','NTRK1','DLX2','DLX1','GAD1','ARX','SOX3','SYT4','SPARCL1','OTX2','LMO1','FABP7','DCT','VEPH1','PRRX1','CDK6','GNG11','HES1','TYMS','PCNA','CDK1','DLX6-AS1','TUBB4B','SELENOP','ID2','CRABP1','APOE','LDHA','FZD5','KRT18','FGF17','ASCL1','REST','GNRH1','GAD1','GAD2','ARX','CDKN1C','TAGLN','NANOG','KIF15','SIX3','PHC1','TUBB4A','NRXN1','CDH7','ROBO1','NCAM1','INA','NEFL','NEFM','TUBB3','FN1','CDH2','NES','VIM','ACTG1','EPHA1','CDH3','RELN','SEMA4A','SEMA4D','TH')
markerGenes<-c('"EMX2","SOX11","TPM1","TUBA1B","RRM2","PCLAF","TTYH1","TYMS","APOE,ATF4","EP300","HDAC9","OTX2","DLX1","GNG8ASCL1","ASCL2","NTRK1","NKX2-1","CDKN1C","ID4","PAX6","DLK1","DIO3","IPW","MEG3","DDC","MAOA","PDLIM1","TTR","PCDH7,FZD5","ID3","ID2","DUT","CRABP1","DUSP4","DLK1","FOXG1","HES5","HES4,TPM1,H1FK,TPM2,NEAT1,RBP1,ELAVL3","ELAVL4","GNAS","STMN2","INA,TAGLN3,ISL1,INSM1","TUBB2A","TPM2,TUBA1A","TBX1","EOMES","NEUROD1","NEUROG1","DNMT1","DNMT3A","GADD45A","KDM6A","KDM6B","EZH2","CHD2","JARID2","RBBP4","TET2","TET3","FABP7","BEX1","STMN4","HAT1","SUZ12"')
p <- plotBrowserTrack(
    ArchRProj = paracetP2, 
    groupBy = "Clusters", 
    geneSymbol = markerGenes, 
    upstream = 50000,
    downstream = 50000,
    #minCells = 1500,
    pal=Atac_cluster_palette 
)
grid::grid.newpage()
plotPDF(plotList = p, 
    name = "CellNUMBER15001_Plot-Tracks-Marker-Genes-Cluster-specific-markers-from-scRNAsox1.pdf", 
   ArchRProj = paracetP2, 
    addDOC = TRUE, width = 5, height = 5)


Transcription_Factors <- c('ARX','ASCL1','DLX1','DLX2','DLX5','DLX6','EMX2','FOXD3','FOXG1','FOXH1','HES1','HES5','HES6','HESX1','ID1','ID2','INSM1','IRX2','ISL1','MYC','NANOG','NEUROD1','NEUROD4','NEUROG1','NHLH1','NKX2-1','NR6A1','OTX2','PAX6','POU2F2','POU3F1','POU5F1','PRRX1','RAX','REST','SIX3','SIX6','SOX11','SOX21','SOX2','SOX3','TLX3','ZIC2','LHX2','NFIB','POU3F3','SOX5','ZIC4','ZEB1','NR2F2','ARX','ASCL1','DLX1','DLX2','DLX5','DLX6','EMX2','FOXD3','FOXG1','FOXH1','HES1','HES5','HES6','HESX1','ID1','ID2','INSM1','IRX2','ISL1','MYC','NANOG','NEUROD1','NEUROD4','NEUROG1','NHLH1','NKX2-1','NR6A1','OTX2','PAX6','POU2F2','POU3F1','POU5F1','PRRX1','RAX','REST','SIX3','SIX6','SOX11','SOX21','SOX2','SOX3','TLX3','ZIC2','LHX2','NFIB','POU3F3','SOX5','ZIC4','ZEB1','NR2F2')


p <- plotBrowserTrack(
    ArchRProj = paracetP2, 
    groupBy = "Clusters", 
    geneSymbol = Transcription_Factors, 
    upstream = 50000,
    downstream = 50000,
    pal=Atac_cluster_palette 
)
grid::grid.newpage()
plotPDF(plotList = p, 
    name = "1_Plot-Tracks-Marker-Genes-Cluster-specific-markers_Transcription_factors.pdf", 
   ArchRProj = paracetP2, 
    addDOC = TRUE, width = 5, height = 5)

```


#scRNASeq 

```{r RNASEQ integration unconstrained} 

#Load scRNA-Seq dataset
#paracetP3 <- "ARCHR_PART1_FINAL_before_integration_RNA.rds"

addArchRThreads(threads=1)

library(SingleCellExperiment)
seRNA <- readRDS("~/RDSFiles/NeuroDiffParacetFinal_EndofPipe_Final_NeuroDiffParacet.qcFilt_releveled.rds")
seRNA
# Assign new IDs based on scCatch annotation results - Using Seurat Markers
#renumbered clusters from 0to13
library(ggplot2)
library(tidyverse)
library(ggrepel)
library(ggthemes)
library(dplyr)

DimPlot(seRNA, reduction = "umap", label = TRUE,group.by = "ident", pt.size = 0.5,cols = c('P1' = '#D37295', 'P2' = '#FFC0CB', 'P3' = '#8175AA', 'P6' = '#FFC966','P9' = '#8CC2CA','P5' = '#B15928', 'P8' = '#59A14F', 'P10' = '#C0D6E4', 'P7' = '#FFD700', 'P4' = '#A3ECB9', 'P11' = '#CCCCCC','P12' = '#F28E2B','P13' = '#B60A1C'))
# 
dev.copy(pdf,paste0("up_13.FINAL_original_LABELS_FILT.UMAP_final.pdf"), width=8, height=6, paper='special')
dev.off()                                                                                          
DimPlot(seRNA, reduction = "umap", label = TRUE,group.by = "orig.ident")
seRNA_1 = Seurat::as.SingleCellExperiment(seRNA)
#table(colData(seRNA)$BioClassification)
seRNA_1$ident
#The metadata contains a column called Seurat_clusters or SNN_res_0.5 which contains the cell type
colnames(colData(seRNA_1))
table(colData(seRNA_1)$ident)
  #integrate scATAC-seq with scRNA-seq
paracetP2<- addGeneIntegrationMatrix(
  ArchRProj = paracetP2, 
  useMatrix = "GeneScoreMatrix",
  matrixName = "GeneIntegrationMatrix",
  reducedDims = "IterativeLSI",
  seRNA = seRNA_1,
  addToArrow = FALSE,
  groupRNA = "ident",
  nameCell = "predictedCell_Un",
  nameGroup = "predictedGroup_Un",
  nameScore = "predictedScore_Un"
)

```
#preclust confusion
```{r Confusion matrix for assignment of clusters } 
###Confusion matrix for assigning clusters_CONSTRAINED INTEGRATION
set.seed(1)
cM <- as.matrix(confusionMatrix(paracetP2$Clusters, paracetP2$predictedGroup_Un))
preClust1 <- colnames(cM)[apply(cM, 1 , which.max)]
preClust1
cbind(preClust1, rownames(cM)) 
#Assignments
unique(unique(paracetP2$predictedGroup_Un))
preClust1
```
#confusionmatrix
```{r  string-based representation of clusters for downstream integration} 
#we will create a confusionMatrix that looks at the intersection of Clusters and predictedGroup_Un which contains the cell types as identified by scRNA-seq.

c1 <- paste0(c(paste0("P5"),"P3"), collapse="|")
c1

cNon1<- paste0(c(paste0("P9"),"P10","P13"), collapse="|")
cNon1



```
#Grouping CLusters
```{r  Assign scATAC to these categories for downstream integration} 
#

clust1 <- rownames(cM)[grep(c1, preClust1)]
clust1

clustNon1<- rownames(cM)[grep(cNon1, preClust1)]
clustNon1


#RNA get cells in these categories
rna1  <- colnames(seRNA_1)[grep(c1, colData(seRNA_1)$ident)]
head(rna1)

rnaNon1 <- colnames(seRNA_1)[grep(cNon1, colData(seRNA_1)$ident)]
head(rnaNon1)

```
#ADD GENE integration
```{r marker ID for ATAC} 

Sys.setenv('R_MAX_VSIZE'=1200000)
set.seed(1)


groupList <- SimpleList(
    cluster1 = SimpleList(
        ATAC = paracetP2$cellNames[paracetP2$Clusters %in% clust1],
        RNA = rna1
    ),
     NonFN1 = SimpleList(
        ATAC = paracetP2$cellNames[paracetP2$Clusters %in% clustNon1],
        RNA = rnaNon1
     )
)

paracetP2 <- addGeneIntegrationMatrix(
    ArchRProj = paracetP2, 
    useMatrix = "GeneScoreMatrix",
    matrixName = "GeneIntegrationMatrix",
    reducedDims = "IterativeLSI",
    seRNA = seRNA_1,
    addToArrow = FALSE, 
    groupList = groupList,
    groupRNA = "ident",
    nameCell = "predictedCell_Co",
    nameGroup = "predictedGroup_Co",
    nameScore = "predictedScore_Co"
)
```

#PlotUMAP
```{r Plot UMAP of Unconstrained p1 and constrained p2 integration ident}
ArchRPalettes$custom <- c('P3' = '#8175AA', 'P6' = '#FFC966','P9' = '#8CC2CA','P5' = '#B15928', 'P8' = '#59A14F', 'P10' = '#C0D6E4', 'P4' = '#A3ECB9', 'P12' = '#F28E2B','P13' = '#B60A1C')
 unconstrained_palette<- ArchRPalettes$custom
p1 <- plotEmbedding(
    paracetP2,
    embedding = "UMAP",
    reducedDims = "IterativeLSI",
    colorBy = "cellColData", 
    name = "predictedGroup_Un", 
    pal = unconstrained_palette,
    rastr = FALSE,
    size=0.3,
   
)
p1


p2 <- plotEmbedding(
    paracetP2, 
    reducedDims = "IterativeLSI",
    colorBy = "cellColData", 
    name = "predictedGroup_Co", 
    pal = constrained_palette,
    rastr = FALSE,
    size=0.3
)
p2
plotPDF(p1,p2, name = "2_Plot-UMAP-unconstrained-constrained-RNA-Integration_pal_constrained_palette_Co3_rmNonSignSignatures.pdf", ArchRProj = paracetP2, addDOC = TRUE, width = 5, height = 5)
saveArchRProject(ArchRProj = paracetP2, outputDirectory = "Save-paracetP2", load = FALSE)
```

<!-- #################################### -->
<!--      End of Save DIFF_P_2
<!--    Pseudo-scRNA-seq profile for scATACseq cell
<!-- #################################### -->



#ADD GENEINTEGRATION
```{r can re-run the integration with addToArrow true_will_rewrite_arrowfiles}

Sys.setenv('R_MAX_VSIZE'=1200000)
set.seed(1)

paracetP3 <- addGeneIntegrationMatrix(
    ArchRProj = paracetP2, 
    useMatrix = "GeneScoreMatrix",
    matrixName = "GeneIntegrationMatrix",
    reducedDims = "IterativeLSI",
    seRNA = seRNA_1,
    addToArrow = TRUE,
    force= TRUE,
    groupList = groupList,
    groupRNA = "ident",
    nameCell = "predictedCell",
    nameGroup = "predictedGroup",
    nameScore = "predictedScore"
)
getAvailableMatrices(paracetP2)

paracetP2 <- addImputeWeights(paracetP2)



```


```{r  UMAP plots overlayed with the gene expression p1 and gene score p2 values, include=FALSE} 
library(hexbin)
markerGenes  <- c("POU5F1", "NANOG", "GAL", "PODXL", "ID1", "MYC", "PAX6", "DNMT3B")


p1 <- plotEmbedding(
    ArchRProj = paracetP3, 
    colorBy = "GeneIntegrationMatrix", 
    name = markerGenes, 
    continuousSet = "horizonExtra",
    embedding = "UMAP",
    imputeWeights = getImputeWeights(paracetP2
    )
)
p1
p2 <- plotEmbedding(
    ArchRProj = paracetP3, 
    colorBy = "GeneScoreMatrix", 
    continuousSet = "horizonExtra",
    name = markerGenes, 
    embedding = "UMAP",
    imputeWeights = getImputeWeights(paracetP2)
)

p1c <- lapply(p1, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})

p2c <- lapply(p2, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})

do.call(cowplot::plot_grid, c(list(ncol = 3), p1c))
do.call(cowplot::plot_grid, c(list(ncol = 3), p2c))

plotPDF(plotList = p1, 
    name = "2_Plot-UMAP-Marker-Feature-Genes-RNA-W-Imputation_rna_markers_constrained.pdf", 
    ArchRProj = paracetP3, 
    addDOC = TRUE, width = 5, height = 5)


markerGenes <-c('SCGB3A2','L1TD1','RYR2','ELL2','PRDM16','SOX6')

p1 <- plotEmbedding(
    ArchRProj = paracetP3, 
    colorBy = "GeneIntegrationMatrix", 
    name = markerGenes, 
    continuousSet = "horizonExtra",
    embedding = "UMAP",
    imputeWeights = getImputeWeights(paracetP2
                            )
)


p2 <- plotEmbedding(
    ArchRProj = paracetP3, 
    colorBy = "GeneScoreMatrix", 
    continuousSet = "horizonExtra",
    name = markerGenes, 
    embedding = "UMAP",
    imputeWeights = getImputeWeights(paracetP2)
)

p1c <- lapply(p1, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})

p2c <- lapply(p2, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})

do.call(cowplot::plot_grid, c(list(ncol = 3), p1c))
do.call(cowplot::plot_grid, c(list(ncol = 3), p2c))

plotPDF(plotList = p1, 
    name = "2_Plot-UMAP-Marker-Feature-Genes-RNA-W-Imputation_ATAC_markers_constrained.pdf", 
    ArchRProj = paracetP3, 
    addDOC = TRUE, width = 5, height = 5)

```
[1,] "P5"      "C10"
 [2,] "P3"      "C7" 
 [3,] "P5"      "C11"
 [4,] "P4"      "C9" 
 [5,] "P9"      "C12"
 [6,] "P5"      "C1" 
 [7,] "P5"      "C8" 
 [8,] "P13"     "C3" 
 [9,] "P5"      "C14"
[10,] "P4"      "C5" 
[11,] "P4"      "C6" 
[12,] "P10"     "C4" 
[13,] "P4"      "C15"
[14,] "P10"     "C13"
[15,] "P3"      "C2" 

```{r Labeling scATAC-seq clusters with scRNA-seq information} 
cM <- confusionMatrix(paracetP2$Clusters, paracetP3$predictedGroup_Co)
labelOld <- rownames(cM)
labelOld

labelNew <- colnames(cM)[apply(cM, 1, which.max)]
labelNew

remapClust <- c(
  "P3" = "P3",
  "P5" = "P5", 
  "P9" = "P9", 
  "P10" = "P10", 
  "P13" = "P13"
 
)

remapClust <- remapClust[names(remapClust) %in% labelNew]

remapClust
labelNew2 <- mapLabels(labelNew, oldLabels = names(remapClust), newLabels = remapClust)
labelNew2

paracetP3$Clusters2 <- mapLabels(paracetP3$Clusters, newLabels = labelNew2, oldLabels = labelOld)

p1 <- plotEmbedding(
      paracetP3, 
      colorBy = "cellColData", 
      name = "Clusters2", 
      pal = constrained_palette
        )

p1

plotPDF(p1, name = "2_Plot-UMAP-constrained-final-names-Clusters.pdf", ArchRProj = paracetP3, addDOC = TRUE, width = 5, height = 5)

saveArchRProject(ArchRProj = paracetP3, outputDirectory = "Save-paracetP3", load = FALSE)

```
<!-- #################################### -->
<!--    END OF SAVE-PROJECT3        
<!--        Making Pseudo-bulk Replicates -->
<!-- #################################### -->



```{r making pseudo bulk replicates for downstream analysis} 
#peak matrix file

paracetP4 <- addGroupCoverages(ArchRProj = paracetP3, groupBy = "Clusters2")
```

```{r calling peaks with MACS2} 

Sys.setenv('R_MAX_VSIZE'=1200000)
set.seed(1)
addArchRThreads(threads = 1)
#INstall macs2 using pip
#pathToMacs2 <- findMacs2()
#library(ArchR)

#Use path where Macs2 is installed # pip install macs2
#pathToMacs2 <- findMacs2("/Users/ankushs/MACS2/bin/macs2")
#if block does not work go to terminal
#step1_go to terminal
#step2_ cd /Users/ankushs/
#step3_ source MACS2/bin/activate
#we made virtual environment on python 3.7
# have to install macs2 on system and give path 
paracetP4 <- addReproduciblePeakSet(
    ArchRProj = paracetP4, 
    groupBy = "Clusters2", 
    pathToMacs2 = "/Users/ankushs/MACS2/bin/macs2"
)
getPeakSet(paracetP4)



```

```{r peakmatrix} 
saveArchRProject(ArchRProj = paracetP4, outputDirectory = "Save-paracetP4", load = FALSE)
```
<!-- #################################### -->
<!--        END OF SAVE PROJECT4          -->
<!--     MARKER, MOTIFS TRAJECTORIES      -->
<!-- #################################### -->
```{r ID marker peaks} 

#library(ArchR)
paracetP5 <- addPeakMatrix(paracetP4)

getAvailableMatrices(paracetP5)
table(paracetP5$Clusters2)

markersPeaks <- getMarkerFeatures(
    ArchRProj = paracetP5, 
    useMatrix = "PeakMatrix", 
    groupBy = "Clusters2",
    bias = c("TSSEnrichment", "log10(nFrags)"),
    testMethod = "wilcoxon"
)
markersPeaks

markerList <- getMarkers(markersPeaks, cutOff = "FDR <= 0.01 & Log2FC >= 0.1")
markerList
markerList$P5
markerList$P3

markerList$P9
markerList$P10
markerList$P13


heatmapPeaks <-markerHeatmap(
  seMarker = markersPeaks, 
  cutOff = "FDR <= 0.1 & Log2FC >= 0.5",
  transpose = TRUE
)

#draw(heatmapPeaks, heatmap_legend_side = "bot", annotation_legend_side = "bot")

plotPDF(heatmapPeaks, name = "4_Peak-Marker-Heatmap", width = 8, height = 6, ArchRProj = paracetP5, addDOC = TRUE)

pma <- plotMarkers(seMarker = markersPeaks, name = "P5", cutOff = "FDR <= 0.1 & Log2FC >= 1", plotAs = "MA")
pma

pv <- plotMarkers(seMarker = markersPeaks, name = "P5", cutOff = "FDR <= 0.1 & Log2FC <= -1", plotAs = "Volcano")
pv
plotPDF(pma, pv, name = "4_R2-Markers-MA-Volcano", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)
dev.off()

```

```{r ID marker peaks} 
#R0 R2 R7 R9 R12
#not working
heatmapPeaks <- markerHeatmap(
  seMarker = markersPeaks, 
  cutOff = "FDR <= 0.1 & Log2FC >= 0.1",
  transpose = TRUE
)
draw(heatmapPeaks, heatmap_legend_side = "bot", annotation_legend_side = "bot")

plotPDF(heatmapPeaks, name = "3_Peak-Marker-Heatmap_0.1_0.5", width = 8, height = 6, ArchRProj = paracetP5, addDOC = TRUE)
dev.off()
dev.set(dev.next())
```

```{r Marker Peak MA and Volcano Plots} 

pma <- plotMarkers(seMarker = markersPeaks, name = "P5", cutOff = "FDR <= 0.1 & Log2FC >= 1", plotAs = "MA")
pma
pv <- plotMarkers(seMarker = markersPeaks, name = "P5", cutOff = "FDR <= 0.1 & Log2FC >= 1", plotAs = "Volcano")
pv
plotPDF(pma, pv, name = "4_P5-Markers-MA-volcano_0.1_1_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

pma <- plotMarkers(seMarker = markersPeaks, name = "P9", cutOff = "FDR <= 0.1 & Log2FC >= 1", plotAs = "MA")
pma
pv <- plotMarkers(seMarker = markersPeaks, name = "P9", cutOff = "FDR <= 0.1 & Log2FC >= 1", plotAs = "Volcano")
pv
plotPDF(pma, pv, name = "4_P9-Markers-MA-volcano_0.1_1_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)


pma <- plotMarkers(seMarker = markersPeaks, name = "P10", cutOff = "FDR <= 0.1 & Log2FC >= 1", plotAs = "MA")
pma
pv <- plotMarkers(seMarker = markersPeaks, name = "P10", cutOff = "FDR <= 0.1 & Log2FC >= 1", plotAs = "Volcano")
pv
plotPDF(pma, pv, name = "4_P10-Markers-MA-volcano_0.1_1_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)


pma <- plotMarkers(seMarker = markersPeaks, name = "P13", cutOff = "FDR <= 0.1 & Log2FC >= 1", plotAs = "MA")
pma
pv <- plotMarkers(seMarker = markersPeaks, name = "P13", cutOff = "FDR <= 0.1 & Log2FC >= 1", plotAs = "Volcano")
pv
plotPDF(pma, pv, name = "4_P13-Markers-MA-volcano_0.1_1_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

```

#plotbrowsertrackafter constraints
```{r Marker Peaks of choice in Browser Tracks,include=FALSE} 
markerGenes <- c("EMX2","SOX11","TPM1","TUBA1B","RRM2","PCLAF","TTYH1","TYMS","APOE,ATF4","EP300","HDAC9","OTX2","DLX1","GNG8ASCL1","ASCL2","NTRK1","NKX2-1","CDKN1C","ID4","PAX6","DLK1","DIO3","IPW","MEG3","DDC","MAOA","PDLIM1","TTR","PCDH7,FZD5","ID3","ID2","DUT","CRABP1","DUSP4","DLK1","FOXG1","HES5","HES4,TPM1,H1FK,TPM2,NEAT1,RBP1,ELAVL3","ELAVL4","GNAS","STMN2","INA,TAGLN3,ISL1,INSM1","TUBB2A","TPM2,TUBA1A","TBX1","EOMES","NEUROD1","NEUROG1","DNMT1","DNMT3A","GADD45A","KDM6A","KDM6B","EZH2","CHD2","JARID2","RBBP4","TET2","TET3","FABP7","BEX1","STMN4","HAT1","SUZ12")
markerGenes <- c("ELAVL4")
 
p <- plotBrowserTrack(
    ArchRProj = paracetP5, 
    groupBy = "Clusters2", 
    geneSymbol = markerGenes, 
    upstream = 25000,
    downstream =75000 ,
    maxCells = 5000,
   pal = constrained_palette
)

grid::grid.newpage()
grid::grid.draw(p$ELAVL4)


plotPDF(plotList = p, 
    name = "ELAVL4_1_maxcellsequals5000_Plot-Tracks-afterconstrainingclusters.pdf", 
    ArchRProj = paracetP5, 
    addDOC = TRUE, width = 5, height = 5)



p <- plotBrowserTrack(
    ArchRProj = paracetP5, 
    groupBy = "Sample", 
    geneSymbol = markerGenes, 
    upstream = 10000,
    downstream = 50000,
    maxCells = 5000,
   pal = sample_id_palette
)

grid::grid.newpage()
#grid::grid.draw(p$CD14)


plotPDF(plotList = p, 
    name = "ELAVL4_1_maxcellsequals5000_SAMPLES_Plot-Tracks-afterconstrainingclusters.pdf", 
    ArchRProj = paracetP5, 
    addDOC = TRUE, width = 5, height = 5)


```





```{r Pairwise Testing Between Groups, include=FALSE}
#This performs a differential test between the two provided groups. In all of these differential tests, the peaks #that are higher in the group passed to useGroups will have positive fold change values while the peaks that are #higher in the group passed to bgdGroups will have negative fold change values.

 markerTest <- getMarkerFeatures(
   ArchRProj = paracetP5, 
   useMatrix = "PeakMatrix",
   groupBy = "Clusters2",
   testMethod = "wilcoxon",
   bias = c("TSSEnrichment", "log10(nFrags)"),
   useGroups = "P5",
   bgdGroups = "P13"
 )
 
 pma <- plotMarkers(seMarker = markerTest, name = "P5", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "MA")
 pma
 pv <- plotMarkers(seMarker = markerTest, name = "P5", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.", plotAs = "Volcano")
 pv
 plotPDF(pma, pv, name = "4_P5-vs-P13-Markers-MA-Volcano_0.5_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)
 
 
 
 markerTest <- getMarkerFeatures(
   ArchRProj = paracetP5, 
   useMatrix = "PeakMatrix",
   groupBy = "Clusters2",
   testMethod = "wilcoxon",
   bias = c("TSSEnrichment", "log10(nFrags)"),
   useGroups = "P5",
   bgdGroups = "P10"
 )
 
 pma <- plotMarkers(seMarker = markerTest, name = "P5", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "MA")
 pma
 pv <- plotMarkers(seMarker = markerTest, name = "P5", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.", plotAs = "Volcano")
 pv
 plotPDF(pma, pv, name = "4_P5-vs-P10-Markers-MA-Volcano_0.5_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)
 
 markerTest <- getMarkerFeatures(
   ArchRProj = paracetP5, 
   useMatrix = "PeakMatrix",
   groupBy = "Clusters2",
   testMethod = "wilcoxon",
   bias = c("TSSEnrichment", "log10(nFrags)"),
   useGroups = "P5",
   bgdGroups = "P9"
 )
 
 pma <- plotMarkers(seMarker = markerTest, name = "P5", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "MA")
 pma
 pv <- plotMarkers(seMarker = markerTest, name = "P5", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "Volcano")
 pv
 plotPDF(pma, pv, name = "4_P5-vs-P9-Markers-MA-Volcano_0.5_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

 markerTest <- getMarkerFeatures(
   ArchRProj = paracetP5, 
   useMatrix = "PeakMatrix",
   groupBy = "Clusters2",
   testMethod = "wilcoxon",
   bias = c("TSSEnrichment", "log10(nFrags)"),
   useGroups = "P5",
   bgdGroups = "P3"
 )
 
 pma <- plotMarkers(seMarker = markerTest, name = "P5", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "MA")
 pma
 pv <- plotMarkers(seMarker = markerTest, name = "P3", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.2", plotAs = "Volcano")
 pv
 plotPDF(pma, pv, name = "P5-vs-P3-Markers-MA-Volcano_0.5_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

 markerTest <- getMarkerFeatures(
   ArchRProj = paracetP5, 
   useMatrix = "PeakMatrix",
   groupBy = "Clusters2",
   testMethod = "wilcoxon",
   bias = c("TSSEnrichment", "log10(nFrags)"),
   useGroups = "P3",
   bgdGroups = "P10"
 )
 
 pma <- plotMarkers(seMarker = markerTest, name = "P3", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "MA")
 pma
 pv <- plotMarkers(seMarker = markerTest, name = "P10", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "Volcano")
 pv
 plotPDF(pma, pv, name = "4_P3-vs-P10-Markers-MA-Volcano_0.5_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)
 
 markerTest <- getMarkerFeatures(
   ArchRProj = paracetP5, 
   useMatrix = "PeakMatrix",
   groupBy = "Clusters2",
   testMethod = "wilcoxon",
   bias = c("TSSEnrichment", "log10(nFrags)"),
   useGroups = "P3",
   bgdGroups = "P9"
 )
 
 pma <- plotMarkers(seMarker = markerTest, name = "P3", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "MA")
 pma
 pv <- plotMarkers(seMarker = markerTest, name = "P9", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.", plotAs = "Volcano")
 pv
 plotPDF(pma, pv, name = "4_P3-vs-P9-Markers-MA-Volcano_0.5_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)
 
 
 R4
 
 markerTest <- getMarkerFeatures(
   ArchRProj = paracetP5, 
   useMatrix = "PeakMatrix",
   groupBy = "Clusters2",
   testMethod = "wilcoxon",
   bias = c("TSSEnrichment", "log10(nFrags)"),
   useGroups = "P3",
   bgdGroups = "P13"
 )
 
 pma <- plotMarkers(seMarker = markerTest, name = "P3", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "MA")
 pma
 pv <- plotMarkers(seMarker = markerTest, name = "P13", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.", plotAs = "Volcano")
 pv
 plotPDF(pma, pv, name = "4_P3-vs-P13-Markers-MA-Volcano_0.5_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)
 
 

 
 
 
 
 markerTest <- getMarkerFeatures(
   ArchRProj = paracetP5, 
   useMatrix = "PeakMatrix",
   groupBy = "Clusters2",
   testMethod = "wilcoxon",
   bias = c("TSSEnrichment", "log10(nFrags)"),
   useGroups = "P9",
   bgdGroups = "P10"
 )
 
 pma <- plotMarkers(seMarker = markerTest, name = "P9", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "MA")
 pma
 pv <- plotMarkers(seMarker = markerTest, name = "P10", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "Volcano")
 pv
 plotPDF(pma, pv, name = "P9-vs-P10-Markers-MA-Volcano_0.5_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)



 pma <- plotMarkers(seMarker = markerTest, name = "P9", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "MA")
 pma
 pv <- plotMarkers(seMarker = markerTest, name = "P10", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "Volcano")
 pv
 plotPDF(pma, pv, name = "P4-vs-P13-Markers-MA-Volcano_0.5_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)
 
 
 
 
 markerTest <- getMarkerFeatures(
   ArchRProj = paracetP5, 
   useMatrix = "PeakMatrix",
   groupBy = "Clusters2",
   testMethod = "wilcoxon",
   bias = c("TSSEnrichment", "log10(nFrags)"),
   useGroups = "P10",
   bgdGroups = "P13"
 )
 
 pma <- plotMarkers(seMarker = markerTest, name = "P10", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "MA")
 pma
 pv <- plotMarkers(seMarker = markerTest, name = "P13", cutOff = "FDR <= 0.1 & abs(Log2FC) >= 0.5", plotAs = "Volcano")
 pv
 plotPDF(pma, pv, name = "P10-vs-P13-Markers-MA-Volcano_0.5_", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

```



#JASPARMOTIF

```{r JASPAR2020ChromVAR Deviatons Enrichment with ArchR,include=FALSE}
###
library(JASPAR2020)
#pairwise between the compared Clusters2 above
markerTest <- getMarkerFeatures(
  ArchRProj = paracetP5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "P5",
  bgdGroups = "P13"
)
#devtools::install_github("GreenleafLab/chromVARmotifs")
library(chromVARmotifs)
paracetP5 <- addMotifAnnotations(ArchRProj = paracetP5, motifSet = "JASPAR2020", name = "Motif_JASPAR", force = TRUE)

motifsUp <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif_JASPAR",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.1"
  )
motifsUp
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp


motifsDo <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif_JASPAR",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.1"
  )
motifsDo

df <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggDo <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo
plotPDF(ggUp, ggDo, name = "4_P5-vs-P13-Markers-Motifs-Enriched_0_1_JASPAR", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

p<-TFBSTools::Matrix(getPeakAnnotation( paracetP5, "Motif_JASPAR")$motifs)
library(seqLogo)
r <- makePWM(p$PAX6_7)
seqLogo(r, ic.scale=FALSE)

markerTest <- getMarkerFeatures(
  ArchRProj = paracetP5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "P5",
  bgdGroups = "P10"
)

motifsUp <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif_JASPAR",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.25"
  )
motifsUp
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

motifsDo <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif_JASPAR",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.1"
  )
motifsDo

df <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggDo <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo
plotPDF(ggUp, ggDo, name = "4_P5-vs-P10-Markers-Motifs-Enriched_0_1_JASPAR", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)


p<-TFBSTools::Matrix(getPeakAnnotation( paracetP5, "Motif_JASPAR")$motifs)
library(seqLogo)
r <- makePWM(p$CREM_6)
seqLogo(r, ic.scale=FALSE)

markerTest <- getMarkerFeatures(
  ArchRProj = paracetP5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "P5",
  bgdGroups = "P9"
)

motifsUp <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif_JASPAR",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.25"
  )
motifsUp
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

motifsDo <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif_JASPAR",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.1"
  )
motifsDo

df <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggDo <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo
plotPDF(ggUp, ggDo, name = "4_P5-vs-P9-Markers-Motifs-Enriched_0_1_JASPAR", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)


p<-TFBSTools::Matrix(getPeakAnnotation( paracetP5, "Motif_JASPAR")$motifs)
library(seqLogo)
r <- makePWM(p$CREM_6)
seqLogo(r, ic.scale=FALSE)


markerTest <- getMarkerFeatures(
  ArchRProj = paracetP5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "P3",
  bgdGroups = "P13"
)

motifsUp <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif_JASPAR",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.25"
  )
motifsUp
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

motifsDo <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif_JASPAR",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.1"
  )
motifsDo

df <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggDo <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo
plotPDF(ggUp, ggDo, name = "4_P3-vs-P13-Markers-Motifs-Enriched_0_1_JASPAR", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)


p<-TFBSTools::Matrix(getPeakAnnotation( paracetP5, "Motif_JASPAR")$motifs)
library(seqLogo)
r <- makePWM(p$SOX2_617)
seqLogo(r, ic.scale=FALSE)


```

#CISBP Motif enrichment

```{r CISBP Motif Enrichment in Differential Peaks, results='hide', include=FALSE} 
#pairwice between the compared Clusters2 above
library(chromVARmotifs)

markerTest<- getMarkerFeatures(
  ArchRProj = paracetP5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "P5",
  bgdGroups = "P13"
)

paracetP5 <- addMotifAnnotations(ArchRProj = paracetP5, motifSet = "cisbp", name = "Motif", force = TRUE)

motifsUp <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.1"
  )
motifsUp
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

motifsDo <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.1"
  )
motifsDo

df <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggDo <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo
plotPDF(ggUp, ggDo, name="4_P5-vs-P13-Markers-Motifs-Enriched_0_1_cisbp", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

###


markerTest <- getMarkerFeatures(
  ArchRProj = paracetP5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "P5",
  bgdGroups = "P10"
)


paracetP5 <- addMotifAnnotations(ArchRProj = paracetP5, motifSet = "cisbp", name = "Motif", force = TRUE)

motifsUp <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.1"
  )
motifsUp
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

motifsDo <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.1"
  )
motifsDo

df <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggDo <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo
plotPDF(ggUp, ggDo, name = "4_P5-vs-P10-Markers-Motifs-Enriched_0_1_cisbp", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)
##
markerTest <- getMarkerFeatures(
  ArchRProj = paracetP5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "P5",
  bgdGroups = "P9"
)


paracetP5 <- addMotifAnnotations(ArchRProj = paracetP5, motifSet = "cisbp", name = "Motif", force = TRUE)

motifsUp <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.1"
  )
motifsUp
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

motifsDo <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.1"
  )
motifsDo

df <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggDo <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo
plotPDF(ggUp, ggDo, name = "4_P5-vs-P9-Markers-Motifs-Enriched_0_1_cisbp", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

###

markerTest <- getMarkerFeatures(
  ArchRProj = paracetP5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "P5",
  bgdGroups = "P3"
)


paracetP5 <- addMotifAnnotations(ArchRProj = paracetP5, motifSet = "cisbp", name = "Motif", force = TRUE)

motifsUp <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.1"
  )
motifsUp
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

motifsDo <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.1"
  )
motifsDo

df <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggDo <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo
plotPDF(ggUp, ggDo, name = "4_P5-vs-P3-Markers-Motifs-Enriched_0_1_cisbp", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)



##
markerTest <- getMarkerFeatures(
  ArchRProj = paracetP5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "P3",
  bgdGroups = "P13"
)


paracetP5 <- addMotifAnnotations(ArchRProj = paracetP5, motifSet = "cisbp", name = "Motif", force = TRUE)

motifsUp <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.1"
  )
motifsUp
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

motifsDo <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.1"
  )
motifsDo

df <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggDo <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo
plotPDF(ggUp, ggDo, name = "4_p3-vs-p13-Markers-Motifs-Enriched_0_1_cisbp", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)





markerTest <- getMarkerFeatures(
  ArchRProj = paracetP5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "P3",
  bgdGroups = "P10"
)


paracetP5 <- addMotifAnnotations(ArchRProj = paracetP5, motifSet = "cisbp", name = "Motif", force = TRUE)

motifsUp <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.1"
  )
motifsUp
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

motifsDo <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.1"
  )
motifsDo

df <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggDo <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo
plotPDF(ggUp, ggDo, name = "4_p3-vs-p10-Markers-Motifs-Enriched_0_1_cisbp", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)



markerTest <- getMarkerFeatures(
  ArchRProj = paracetP5, 
  useMatrix = "PeakMatrix",
  groupBy = "Clusters2",
  testMethod = "wilcoxon",
  bias = c("TSSEnrichment", "log10(nFrags)"),
  useGroups = "P3",
  bgdGroups = "P9"
)


paracetP5 <- addMotifAnnotations(ArchRProj = paracetP5, motifSet = "cisbp", name = "Motif", force = TRUE)

motifsUp <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.1"
  )
motifsUp
df <- data.frame(TF = rownames(motifsUp), mlog10Padj = assay(motifsUp)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggUp <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(P-adj) Motif Enrichment") + 
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggUp

motifsDo <- peakAnnoEnrichment(
    seMarker = markerTest,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC <= -0.1"
  )
motifsDo

df <- data.frame(TF = rownames(motifsDo), mlog10Padj = assay(motifsDo)[,1])
df <- df[order(df$mlog10Padj, decreasing = TRUE),]
df$rank <- seq_len(nrow(df))
head(df)

ggDo <- ggplot(df, aes(rank, mlog10Padj, color = mlog10Padj)) + 
  geom_point(size = 1) +
  ggrepel::geom_label_repel(
        data = df[rev(seq_len(30)), ], aes(x = rank, y = mlog10Padj, label = TF), 
        size = 1.5,
        nudge_x = 2,
        color = "black"
  ) + theme_ArchR() + 
  ylab("-log10(FDR) Motif Enrichment") +
  xlab("Rank Sorted TFs Enriched") +
  scale_color_gradientn(colors = paletteContinuous(set = "comet"))

ggDo
plotPDF(ggUp, ggDo, name = "4_p3-vs-p9-Markers-Motifs-Enriched_0_1_cisbp", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)




rm(seRNA)
rm(seRNA_1)
```
#Enrich MOtif
```{r ID Motif Enrichment in Marker Peaks} 
#START FROM here
library(magick)
enrichMotifs <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = paracetP5,
    peakAnnotation = "Motif",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.1"
  )
enrichMotifs
heatmapEM <- plotEnrichHeatmap(enrichMotifs, n = 7, transpose = TRUE, pal = constrained_palette)
ComplexHeatmap::draw(heatmapEM, heatmap_legend_side = "bot", annotation_legend_side = "bot")
plotPDF(heatmapEM, name = "4_Motifs-Enriched-Marker-peaks-0.1_mycolor-Heatmap_CISBP", width = 6, height = 6, ArchRProj = paracetP5, addDOC = TRUE, rastr=FALSE)

```
#ADDARCHANNOTATIONFUNCTION
```{r ADD ARCHR function}
addArchRAnnotations <- function(
 ArchRProj = NULL,
  db = "ArchR",
  collection = "EncodeTFBS",
  name = collection,
  force = FALSE,
  logFile = createLogFile("addArchRAnnotations")
){

 ArchR:::.validInput(input = ArchRProj, name = "ArchRProj", valid = c("ArchRProj"))
  ArchR:::.validInput(input = db, name = "db", valid = c("character"))
  ArchR:::.validInput(input = collection, name = "collection", valid = c("character"))
  ArchR:::.validInput(input = name, name = "name", valid = c("character"))
  ArchR:::.validInput(input = force, name = "force", valid = c("boolean"))
  ArchR:::.validInput(input = logFile, name = "logFile", valid = c("character"))

  tstart <- Sys.time()
  ArchR:::.startLogging(logFile = logFile)
  ArchR:::.logThis(mget(names(formals()),sys.frame(sys.nframe())), "addArchRAnnotations Input-Parameters", logFile = logFile)

  if(name %in% names(ArchRProj@peakAnnotation)){
    if(force){
      message("peakAnnotation name already exists! Overriding.")
    }else{
      stop("peakAnnotation name already exists! set force = TRUE to override!")
    }
  }

  genome <- tolower(validBSgenome(getGenome(ArchRProj))@metadata$genome)

  annoPath <- file.path(find.package("ArchR", NULL, quiet = TRUE), "data", "Annotations")
  dir.create(annoPath, showWarnings = FALSE)

  if(tolower(db) == "lola"){

    if(genome == "hg19"){
      url <- "https://jeffgranja.s3.amazonaws.com/ArchR/Annotations/LOLA-Hg19-v1.Anno"
    }else if(genome == "hg38"){
      url <- "https://jeffgranja.s3.amazonaws.com/ArchR/Annotations/LOLA-Hg38-v1.Anno"
    }else if(genome == "mm9"){
      url <- "https://jeffgranja.s3.amazonaws.com/ArchR/Annotations/LOLA-Mm9-v1.Anno"
    }else if(genome == "mm10"){
      url <- "https://jeffgranja.s3.amazonaws.com/ArchR/Annotations/LOLA-Mm10-v1.Anno"
    }

    #Download
    if(!file.exists(file.path(annoPath, basename(url)))){
      message("Annotation ", basename(url)," does not exist! Downloading..")
      download.file(
        url = url,
        destfile = file.path(annoPath, basename(url)),
        quiet = FALSE
      )
    }
    AnnoFile <- file.path(annoPath, basename(url))

  }else if(tolower(db) == "archr"){

    if(genome == "hg19"){
      url <- "https://jeffgranja.s3.amazonaws.com/ArchR/Annotations/ArchR-Hg19-v1.Anno"
    }else if(genome == "hg38"){
      url <- "https://jeffgranja.s3.amazonaws.com/ArchR/Annotations/ArchR-Hg38-v1.Anno"
    }else if(genome == "mm9"){
      stop("ArchR mm9 annotations not yet supported! Try LOLA for now!")
      url <- "https://jeffgranja.s3.amazonaws.com/ArchR/Annotations/ArchR-Mm9-v1.Anno"
    }else if(genome == "mm10"){
      stop("ArchR mm10 annotations not yet supported! Try LOLA for now!")
      url <- "https://jeffgranja.s3.amazonaws.com/ArchR/Annotations/ArchR-Mm10-v1.Anno"
    }

    #Download
    if(!file.exists(file.path(annoPath, basename(url)))){
      message("Annotation ", basename(url)," does not exist! Downloading..")
      download.file(
        url = url,
        destfile = file.path(annoPath, basename(url)),
        quiet = FALSE
      )
   }
    AnnoFile <- file.path(annoPath, basename(url))

  }else if(tolower(db) %ni% c("archr", "lola")){

    if(!file.exists(db)){
      stop("Database path does not exist! Please supply valid database for Annotations!")
    }
    AnnoFile <- db

  }else{

    stop("Please supply valid database for Annotations!")

}

 # Check AnnoFile is ArchRAnno
 if (h5read(AnnoFile, "Class") != "ArchRAnno") {
   stop("Not Valid ArchRAnno!")
  }

  #Check if Collection is Valid
 h5d <- h5ls(AnnoFile)
  collections <- h5d[h5d$group == "/" & h5d$otype == "H5I_GROUP",]$name
  if(any(tolower(collections) == tolower(collection))){
    collection <- collections[tolower(collections) %in% tolower(collection)]
  }else{
    stop("Not Valid Collection in ArchRAnno Database!")
  }

  #############################################################
  # Peak Overlap Matrix
  #############################################################
  peakSet <- getPeakSet(ArchRProj)
  chr <- paste0(unique(seqnames(peakSet)))

  ArchR:::.logMessage("Annotating Chromosomes", verbose = TRUE, logFile = logFile)
  regionMat <- lapply(seq_along(chr), function(i){
    ArchR:::.logMessage(paste0("\tAnnotating Chr: ", chr[i]), verbose = TRUE, logFile = logFile)
    regions <- ArchR:::.getRegionsFromAnno(AnnoFile = AnnoFile, Group = collection, chr = chr[i])
    o <- DataFrame(findOverlaps(query = peakSet, subject = regions, ignore.strand = TRUE))
    o$ID <- as.integer(mcols(regions)[o$subjectHits, "ID"])
    o$subjectHits <- NULL
    o
  }) %>% Reduce("rbind", .)
  ArchR:::.logThis(regionMat, "regionMat-Overlaps", logFile=logFile)

  ArchR:::.logThis(range(regionMat[, 1]), "regionMat-Overlaps-1", logFile=logFile)
  ArchR:::.logThis(range(regionMat[, 2]), "regionMat-Overlaps-2", logFile=logFile)
  ArchR:::.logThis(peakSet, "regionMat-Overlaps-1-Peaks", logFile=logFile)
  ArchR:::.logThis(unique(regionMat$ID), "regionMat-Overlaps-2-IDs", logFile=logFile)
  regionMat <- Matrix::sparseMatrix(
    i = regionMat[, 1],
    j = regionMat[, 2],
    x = rep(TRUE, nrow(regionMat)),
    dims = c(length(peakSet), max(regionMat$ID))
  )
 ArchR:::.logThis(regionMat, "regionMat", logFile=logFile)

  #Get Region Metadata
  regionMetadata <- DataFrame(h5read(AnnoFile, paste0(collection,"/Info")))
  rownames(regionMetadata) <- regionMetadata$name
  colnames(regionMat) <- rownames(regionMetadata)
  regionMetadata$name <- NULL

  #Matches Summarized Experiment
  regionMat <- SummarizedExperiment::SummarizedExperiment(
    assays=SimpleList(matches = regionMat),
    rowRanges = peakSet,
    colData = regionMetadata
  )
  ArchR:::.logThis(regionMat, "regionSE", logFile=logFile)

  dir.create(file.path(getOutputDirectory(ArchRProj), "Annotations"), showWarnings=FALSE)
  saveMatches <- file.path(getOutputDirectory(ArchRProj), "Annotations", paste0(name,"-Matches-In-Peaks.rds"))

  out <- SimpleList(
    regionMatches = regionMat,
    regionPositions = "None",
    date = Sys.Date()
  )

  ArchRProj@peakAnnotation[[name]]$Name <- name
  ArchRProj@peakAnnotation[[name]]$Positions <- "None"
  ArchRProj@peakAnnotation[[name]]$Matches <- saveMatches

 ArchR:::.safeSaveRDS(out, file.path(getOutputDirectory(ArchRProj),  "Annotations",
 paste0(name,"-In-Peaks-Summary.rds")), compress = FALSE)
  ArchR:::.safeSaveRDS(out$regionMatches, saveMatches, compress = FALSE)

  ArchR:::.endLogging(logFile = logFile)

  return(ArchRProj)

 }

```

#Encode AddArchRAnnotation
```{r Encode TF Binding Sites}

paracetP5 <- addArchRAnnotations(ArchRProj = paracetP5, collection = "EncodeTFBS",force = TRUE)

enrichEncode <- peakAnnoEnrichment(
seMarker = markersPeaks,
ArchRProj = paracetP5,
peakAnnotation = "EncodeTFBS",
cutOff = "FDR <= 0.1 & Log2FC >= 0.5"
 )
enrichEncode
heatmapEncode <- plotEnrichHeatmap(enrichEncode, n = 7, transpose = TRUE,pal = constrained_palette)
ComplexHeatmap::draw(heatmapEncode, heatmap_legend_side = "bot", annotation_legend_side = "bot")
plotPDF(heatmapEncode, name = "4_EncodeTFBS-Enriched-Marker-Heatmap", width = 8, height = 6, ArchRProj = paracetP5, addDOC = TRUE,rastr=FALSE)
        
```
#Peak annoenrichmentENCODE TFBS 
```{r Encode TF Binding Sites}
enrichEncode <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = paracetP5,
  cutOff = "FDR <= 0.1 & Log2FC >= 0.1"
 )

enrichEncode
heatmapEncode <- plotEnrichHeatmap(enrichEncode, n = 7, transpose = TRUE, pal = constrained_palette)
#ComplexHeatmap::draw(heatmapEncode, heatmap_legend_side = "bot", annotation_legend_side = "bot")
plotPDF(heatmapEncode, name = "4_EncodeTFBS-Enriched-Marker-Heatmap_MyChoice_logFC0.1", width = 8, height = 6, ArchRProj = paracetP5, addDOC = TRUE)
heatmapEncode
#saveRDS(paracetP3, file = "ARCHR_PART4_FINAL_after_MACS2_peak_calling.rds")

```

#CODEX
```{r Codex TFBS}

#paracetP5 <- addArchRAnnotations(ArchRProj = paracetP5, collection = "Encodetfbs")
paracetP5 <- addArchRAnnotations(ArchRProj = paracetP5, collection = "Codex")
## Annotating Chr: 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 X

enrichCodex <- peakAnnoEnrichment(
    seMarker = markersPeaks,
    ArchRProj = paracetP5,
    peakAnnotation = "Codex",
    cutOff = "FDR <= 0.1 & Log2FC >= 0.1"
  )


enrichCodex
heatmapCodex <- plotEnrichHeatmap(enrichCodex, n = 7, transpose = TRUE, pal=constrained_palette)
ComplexHeatmap::draw(heatmapCodex, heatmap_legend_side = "bot", annotation_legend_side = "bot")
plotPDF(heatmapCodex, name = "4_Codex-Enriched-Marker-Heatmap_0.1_", width = 8, height = 6, ArchRProj = paracetP5, addDOC = TRUE,rastr=FALSE)

```


#Cisbp Custom deviation

```{r ChromVAR Deviatons Enrichment with ArchR,include=FALSE}



#predicting enrichment of TF activity on a per-cell basis from sparse chromatin accessibility data

#make sure we have added motif annotations
if("Motif" %ni% names(paracetP5@peakAnnotation)){
    paracetP5 <- addMotifAnnotations(ArchRProj = paracetP5, motifSet = "cisbp", name = "Motif")
}

# add a set of background peaks which are used in computing deviations
paracetP5 <- addBgdPeaks(paracetP5, force = TRUE)

#compute per-cell deviations accross all of our motif annotations
library("ArchR")
paracetP5 <- addDeviationsMatrix(
  ArchRProj = paracetP5, 
  peakAnnotation = "Motif",
  force = TRUE
)

plotVarDev <- getVarDeviations(paracetP5, name = "MotifMatrix", plot = TRUE)

plotPDF(plotVarDev, name = "4_Plot_cisbp_Variable-Motif-Deviation-Scores", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

motifs <- c("POU5F1", "NANOG", "TDGF1", "GAL", "PODXL", "ID1", "MYC", "PAX6", "DNMT3B", "LIN28A", "FOXD3-AS1", "CD24", "EPCAM", "DPPA4", "FZD2", "NR6A1", "SOX21", "ZIC2", "SOX11", "HESX1", "MYLK", "IGFBP5", "FZD5", "FGF8", "SOX2", "RAX", "LIX1", "FEZF2", "POU3F1", "POU3F2", "NKX2-1", "ASCL1", "NEUROG1", "NEUROD1", "ISL1", "BSX", "HMX2", "OTP", "GSX1", "SF1", "GNRH1", "GRIA2", "L1CAM", "NHLH1", "ELAVL4", "SIX3", "SIX6", "OTX2", "TH", "INSM1", "ARX", "DBX1", "DCX", "DLL1", "DLX1", "DLX2", "DLX5", "DLX6", "GAD1", "GRIA1", "INA", "KIF19", "NEFL", "NEFM", "NEUROD4", "NSG2", "POU2F2", "RELN", "SCG2", "SIM1", "SIM2", "SNAP25", "SP9", "SST", "STMN2", "SYP", "SYT4", "TAC1", "TAGLN3", "TLX3", "NTRK1", "NTRK3", "PCSK2", "RET", "DLL3", "GAD2", "GAP43", "REST", "SOX3", "TUBB3", "HES1", "BMP4", "KLF15", "CDK1", "TYMS", "ANLN", "HES6", "LMO1", "ELAVL3", "PRRX1", "OTX2", "PCNA", "SPARCL1", "MAP6", "KRT18", "DDIT4", "PAMR1", "FOXH1", "FABP7", "RBFOX3", "SOX6", "CDH7", "CDH12", "PHC1", "PHC2", "YAF2", "CDH1", "CDH2", "GNG2", "EMX2", "CDK6", "TAGLN", "GNG8", "KLF11", "LHX9", "DCT", "VEPH1", "GNG3", "GNG11", "TRIM71", "SOX3", "KRT19", "NSG1", "JHY")
markerMotifs <- getFeatures(paracetP5, select = paste(motifs, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs

#here we can remove TF binding motifs that are not of interest
#markerMotifs <- grep("z:", markerMotifs, value = TRUE)
#markerMotifs <- markerMotifs[markerMotifs %ni% "z:SREBF1_22"]
#markerMotifs

#we can plot the distribution of chromVAR deviation scores for each cluster. Notice that we supply the impute weights that we calculated previously during our gene score analyses.
p <- plotGroups(ArchRProj = paracetP5, 
  groupBy = "Clusters2", 
  colorBy = "MotifMatrix", 
  name = markerMotifs,
  pal=constrained_palette,
  imputeWeights = getImputeWeights(paracetP5)
)

#we can plot this
p2 <- lapply(seq_along(p), function(x){
  if(x != 1){
    p[[x]] + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6) +
    theme(plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")) +
    theme(
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        axis.title.y=element_blank()
    ) + ylab("")
  }
  else{
    p[[x]] + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6) +
    theme(plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")) +
    theme(
        axis.ticks.y=element_blank(),
        axis.title.y=element_blank()
    ) + ylab("")
  }
})
do.call(cowplot::plot_grid, c(list(nrow = 1, rel_widths = c(2, rep(1, length(p2) - 1))),p2))

plotPDF(p, name = "4_Plot_cisbp_chromvar-Groups-Deviations-w-Imputation", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)


#Plot onto our UMAPS
p <- plotEmbedding(
    ArchRProj = paracetP5, 
    colorBy = "MotifMatrix", 
    #reducedDim("IterativeLSI"),
    name = sort(markerMotifs), 
    embedding = "UMAP",
    size=0.3,
    imputeWeights = getImputeWeights(paracetP5)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(p, name = "4_UMAP-cisbp_chromvar-Motif_matrix-Groups-Deviations-w-Imputation", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

# TF deviation z-scores compare to the inferred gene expression via gene scores of the corresponding TF genes, we can overlay the gene scores for each of these TFs on the UMAP embedding

markerRNA <- getFeatures(paracetP5, select = paste(motifs, collapse="|"), useMatrix = "GeneScoreMatrix")
markerRNA <- markerRNA[markerRNA %ni% c("POU5F1", "NANOG", "TDGF1", "GAL", "PODXL", "ID1", "MYC", "PAX6", "DNMT3B", "LIN28A", "FOXD3-AS1", "CD24", "EPCAM", "DPPA4", "FZD2", "NR6A1", "SOX21", "ZIC2", "SOX11", "HESX1", "MYLK", "IGFBP5", "FZD5", "FGF8", "SOX2", "RAX", "LIX1", "FEZF2", "POU3F1", "POU3F2", "NKX2-1", "ASCL1", "NEUROG1", "NEUROD1", "ISL1", "BSX", "HMX2", "OTP", "GSX1", "SF1", "GNRH1", "GRIA2", "L1CAM", "NHLH1", "ELAVL4", "SIX3", "SIX6", "OTX2", "TH", "INSM1", "ARX", "DBX1", "DCX", "DLL1", "DLX1", "DLX2", "DLX5", "DLX6", "GAD1", "GRIA1", "INA", "KIF19", "NEFL", "NEFM", "NEUROD4", "NSG2", "POU2F2", "RELN", "SCG2", "SIM1", "SIM2", "SNAP25", "SP9", "SST", "STMN2", "SYP", "SYT4", "TAC1", "TAGLN3", "TLX3", "NTRK1", "NTRK3", "PCSK2", "RET", "DLL3", "GAD2", "GAP43", "REST", "SOX3", "TUBB3", "HES1", "BMP4", "KLF15", "CDK1", "TYMS", "ANLN", "HES6", "LMO1", "ELAVL3", "PRRX1", "OTX2", "PCNA", "SPARCL1", "MAP6", "KRT18", "DDIT4", "PAMR1", "FOXH1", "FABP7", "RBFOX3", "SOX6", "CDH7", "CDH12", "PHC1", "PHC2", "YAF2", "CDH1", "CDH2", "GNG2", "EMX2", "CDK6", "TAGLN", "GNG8", "KLF11", "LHX9", "DCT", "VEPH1", "GNG3", "GNG11", "TRIM71", "SOX3", "KRT19", "NSG1", "JHY")]
markerRNA

p <- plotEmbedding(
    ArchRProj = paracetP5, 
    colorBy = "GeneScoreMatrix", 
    name = sort(markerRNA), 
    embedding = "UMAP",
    size=0.3,
    imputeWeights = getImputeWeights(paracetP5)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(p, name = "4_UMAP-cisbp_chromvar-geneScores-Plot-Groups-Deviations-w-Imputation", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

#we can plot the linked gene expression for each of these TFs on the UMAP embedding

markerRNA <- getFeatures(paracetP5, select = paste(motifs, collapse="|"), useMatrix = "GeneIntegrationMatrix")
markerRNA <- markerRNA[markerRNA %ni% c("POU5F1", "NANOG", "TDGF1", "GAL", "PODXL", "ID1", "MYC", "PAX6", "DNMT3B", "LIN28A", "FOXD3-AS1", "CD24", "EPCAM", "DPPA4", "FZD2", "NR6A1", "SOX21", "ZIC2", "SOX11", "HESX1", "MYLK", "IGFBP5", "FZD5", "FGF8", "SOX2", "RAX", "LIX1", "FEZF2", "POU3F1", "POU3F2", "NKX2-1", "ASCL1", "NEUROG1", "NEUROD1", "ISL1", "BSX", "HMX2", "OTP", "GSX1", "SF1", "GNRH1", "GRIA2", "L1CAM", "NHLH1", "ELAVL4", "SIX3", "SIX6", "OTX2", "TH", "INSM1", "ARX", "DBX1", "DCX", "DLL1", "DLX1", "DLX2", "DLX5", "DLX6", "GAD1", "GRIA1", "INA", "KIF19", "NEFL", "NEFM", "NEUROD4", "NSG2", "POU2F2", "RELN", "SCG2", "SIM1", "SIM2", "SNAP25", "SP9", "SST", "STMN2", "SYP", "SYT4", "TAC1", "TAGLN3", "TLX3", "NTRK1", "NTRK3", "PCSK2", "RET", "DLL3", "GAD2", "GAP43", "REST", "SOX3", "TUBB3", "HES1", "BMP4", "KLF15", "CDK1", "TYMS", "ANLN", "HES6", "LMO1", "ELAVL3", "PRRX1", "OTX2", "PCNA", "SPARCL1", "MAP6", "KRT18", "DDIT4", "PAMR1", "FOXH1", "FABP7", "RBFOX3", "SOX6", "CDH7", "CDH12", "PHC1", "PHC2", "YAF2", "CDH1", "CDH2", "GNG2", "EMX2", "CDK6", "TAGLN", "GNG8", "KLF11", "LHX9", "DCT", "VEPH1", "GNG3", "GNG11", "TRIM71", "SOX3", "KRT19", "NSG1", "JHY")]
markerRNA

p <- plotEmbedding(
    ArchRProj = paracetP5, 
    colorBy = "GeneIntegrationMatrix", 
    name = sort(markerRNA), 
    embedding = "UMAP",
    size=0.3,
    imputeWeights = getImputeWeights(paracetP5)
)


p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(p, name = "4_UMAP-cisbp_chromvar_LinkedGeneintegrationExpression-Plot-Groups-Deviations-w-Imputation", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

#saveRDS(paracetP5, file = "ARCHR_Diff_P4_FINAL_after_MACS2_peak_calling.rds")

```

#Jaspar Custom deviations
```{r ArchR and Custom Deviations,include=FALSE}
#predicting enrichment of TF activity on a per-cell basis from sparse chromatin accessibility data

#make sure we have added motif annotations
if("Motif" %ni% names(paracetP5@peakAnnotation)){
    paracetP5 <- addMotifAnnotations(ArchRProj = paracetP5, motifSet = "JASPAR2020", name = "Motif")
}

# add a set of background peaks which are used in computing deviations
paracetP5 <- addBgdPeaks(paracetP5, force = TRUE)

#compute per-cell deviations across all of our motif annotations

paracetP5 <- addDeviationsMatrix(
  ArchRProj = paracetP5, 
  peakAnnotation = "Motif",
  force = TRUE
)

plotVarDev <- getVarDeviations(paracetP5, name = "MotifMatrix", plot = TRUE)

plotPDF(plotVarDev, name = "4_JASPAR_2020_Variable-Motif-Deviation-Scores", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

motifs <- c("POU5F1", "NANOG", "TDGF1", "GAL", "PODXL", "ID1", "MYC", "PAX6", "DNMT3B", "LIN28A", "FOXD3-AS1", "CD24", "EPCAM", "DPPA4", "FZD2", "NR6A1", "SOX21", "ZIC2", "SOX11", "HESX1", "MYLK", "IGFBP5", "FZD5", "FGF8", "SOX2", "RAX", "LIX1", "FEZF2", "POU3F1", "POU3F2", "NKX2-1", "ASCL1", "NEUROG1", "NEUROD1", "ISL1", "BSX", "HMX2", "OTP", "GSX1", "SF1", "GNRH1", "GRIA2", "L1CAM", "NHLH1", "ELAVL4", "SIX3", "SIX6", "OTX2", "TH", "INSM1", "ARX", "DBX1", "DCX", "DLL1", "DLX1", "DLX2", "DLX5", "DLX6", "GAD1", "GRIA1", "INA", "KIF19", "NEFL", "NEFM", "NEUROD4", "NSG2", "POU2F2", "RELN", "SCG2", "SIM1", "SIM2", "SNAP25", "SP9", "SST", "STMN2", "SYP", "SYT4", "TAC1", "TAGLN3", "TLX3", "NTRK1", "NTRK3", "PCSK2", "RET", "DLL3", "GAD2", "GAP43", "REST", "SOX3", "TUBB3", "HES1", "BMP4", "KLF15", "CDK1", "TYMS", "ANLN", "HES6", "LMO1", "ELAVL3", "PRRX1", "OTX2", "PCNA", "SPARCL1", "MAP6", "KRT18", "DDIT4", "PAMR1", "FOXH1", "FABP7", "RBFOX3", "SOX6", "CDH7", "CDH12", "PHC1", "PHC2", "YAF2", "CDH1", "CDH2", "GNG2", "EMX2", "CDK6", "TAGLN", "GNG8", "KLF11", "LHX9", "DCT", "VEPH1", "GNG3", "GNG11", "TRIM71", "SOX3", "KRT19", "NSG1", "JHY")
markerMotifs <- getFeatures(paracetP5, select = paste(motifs, collapse="|"), useMatrix = "MotifMatrix")
markerMotifs

#here we can remove TF binding motifs that are not of interest
#markerMotifs <- grep("z:", markerMotifs, value = TRUE)
#markerMotifs <- markerMotifs[markerMotifs %ni% "z:SREBF1_22"]
#markerMotifs

#we can plot the distribution of chromVAR deviation scores for each cluster. Notice that we supply the impute weights that we calculated previously during our gene score analyses.
p <- plotGroups(ArchRProj = paracetP5, 
  groupBy = "Clusters2", 
  colorBy = "MotifMatrix", 
  name = markerMotifs,
  pal=constrained_palette,
  imputeWeights = getImputeWeights(paracetP5)
)






#we can plot this
p2 <- lapply(seq_along(p), function(x){
  if(x != 1){
    p[[x]] + guides(color=FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6) +
    theme(plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")) +
    theme(
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank(),
        axis.title.y=element_blank()
    ) + ylab("")
  }
  else{
    p[[x]] + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6) +
    theme(plot.margin = unit(c(0.1, 0.1, 0.1, 0.1), "cm")) +
    theme(
        axis.ticks.y=element_blank(),
        axis.title.y=element_blank()
    ) + ylab("")
  }
})
do.call(cowplot::plot_grid, c(list(nrow = 1, rel_widths = c(2, rep(1, length(p2) - 1))),p2))

plotPDF(p, name = "4_Plot_JASPAR2020_chromvar-Groups-_Motif_Matrix_Deviations-w-Imputation", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

#Plot onto our UMAPS
p <- plotEmbedding(
    ArchRProj = paracetP5, 
    colorBy = "MotifMatrix", 
    #reducedDim("IterativeLSI"),
    name = sort(markerMotifs), 
    embedding = "UMAP",
    
   
    imputeWeights = getImputeWeights(paracetP5)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(p, name = "4_UMAP-JASPAR_2020_chromvar-Plot-Groups-_Motif_Matrix_Deviations-w-Imputation", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

# TF deviation z-scores compare to the inferred gene expression via gene scores of the corresponding TF genes, we can overlay the gene scores for each of these TFs on the UMAP embedding

markerRNA <- getFeatures(paracetP5, select = paste(motifs, collapse="|"), useMatrix = "GeneScoreMatrix")
markerRNA <- markerRNA[markerRNA %ni% c("POU5F1", "NANOG", "TDGF1", "GAL", "PODXL", "ID1", "MYC", "PAX6", "DNMT3B", "LIN28A", "FOXD3-AS1", "CD24", "EPCAM", "DPPA4", "FZD2", "NR6A1", "SOX21", "ZIC2", "SOX11", "HESX1", "MYLK", "IGFBP5", "FZD5", "FGF8", "SOX2", "RAX", "LIX1", "FEZF2", "POU3F1", "POU3F2", "NKX2-1", "ASCL1", "NEUROG1", "NEUROD1", "ISL1", "BSX", "HMX2", "OTP", "GSX1", "SF1", "GNRH1", "GRIA2", "L1CAM", "NHLH1", "ELAVL4", "SIX3", "SIX6", "OTX2", "TH", "INSM1", "ARX", "DBX1", "DCX", "DLL1", "DLX1", "DLX2", "DLX5", "DLX6", "GAD1", "GRIA1", "INA", "KIF19", "NEFL", "NEFM", "NEUROD4", "NSG2", "POU2F2", "RELN", "SCG2", "SIM1", "SIM2", "SNAP25", "SP9", "SST", "STMN2", "SYP", "SYT4", "TAC1", "TAGLN3", "TLX3", "NTRK1", "NTRK3", "PCSK2", "RET", "DLL3", "GAD2", "GAP43", "REST", "SOX3", "TUBB3", "HES1", "BMP4", "KLF15", "CDK1", "TYMS", "ANLN", "HES6", "LMO1", "ELAVL3", "PRRX1", "OTX2", "PCNA", "SPARCL1", "MAP6", "KRT18", "DDIT4", "PAMR1", "FOXH1", "FABP7", "RBFOX3", "SOX6", "CDH7", "CDH12", "PHC1", "PHC2", "YAF2", "CDH1", "CDH2", "GNG2", "EMX2", "CDK6", "TAGLN", "GNG8", "KLF11", "LHX9", "DCT", "VEPH1", "GNG3", "GNG11", "TRIM71", "SOX3", "KRT19", "NSG1", "JHY")]
markerRNA

p <- plotEmbedding(
    ArchRProj = paracetP5, 
    colorBy = "GeneScoreMatrix", 
    name = sort(markerRNA), 
    embedding = "UMAP",
   
    imputeWeights = getImputeWeights(paracetP5)
)

p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(p, name = "4_UMAP-JASPAR2020_chromvar-geneScores-Plot-Groups-Deviations-w-Imputation", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)

#we can plot the linked gene expression for each of these TFs on the UMAP embedding

markerRNA <- getFeatures(paracetP5, select = paste(motifs, collapse="|"), useMatrix = "GeneIntegrationMatrix")
markerRNA <- markerRNA[markerRNA %ni% c("POU5F1", "NANOG", "TDGF1", "GAL", "PODXL", "ID1", "MYC", "PAX6", "DNMT3B", "LIN28A", "FOXD3-AS1", "CD24", "EPCAM", "DPPA4", "FZD2", "NR6A1", "SOX21", "ZIC2", "SOX11", "HESX1", "MYLK", "IGFBP5", "FZD5", "FGF8", "SOX2", "RAX", "LIX1", "FEZF2", "POU3F1", "POU3F2", "NKX2-1", "ASCL1", "NEUROG1", "NEUROD1", "ISL1", "BSX", "HMX2", "OTP", "GSX1", "SF1", "GNRH1", "GRIA2", "L1CAM", "NHLH1", "ELAVL4", "SIX3", "SIX6", "OTX2", "TH", "INSM1", "ARX", "DBX1", "DCX", "DLL1", "DLX1", "DLX2", "DLX5", "DLX6", "GAD1", "GRIA1", "INA", "KIF19", "NEFL", "NEFM", "NEUROD4", "NSG2", "POU2F2", "RELN", "SCG2", "SIM1", "SIM2", "SNAP25", "SP9", "SST", "STMN2", "SYP", "SYT4", "TAC1", "TAGLN3", "TLX3", "NTRK1", "NTRK3", "PCSK2", "RET", "DLL3", "GAD2", "GAP43", "REST", "SOX3", "TUBB3", "HES1", "BMP4", "KLF15", "CDK1", "TYMS", "ANLN", "HES6", "LMO1", "ELAVL3", "PRRX1", "OTX2", "PCNA", "SPARCL1", "MAP6", "KRT18", "DDIT4", "PAMR1", "FOXH1", "FABP7", "RBFOX3", "SOX6", "CDH7", "CDH12", "PHC1", "PHC2", "YAF2", "CDH1", "CDH2", "GNG2", "EMX2", "CDK6", "TAGLN", "GNG8", "KLF11", "LHX9", "DCT", "VEPH1", "GNG3", "GNG11", "TRIM71", "SOX3", "KRT19", "NSG1", "JHY")]
markerRNA

p <- plotEmbedding(
    ArchRProj = paracetP5, 
    colorBy = "GeneIntegrationMatrix", 
    name = sort(markerRNA), 
    embedding = "UMAP",
    imputeWeights = getImputeWeights(paracetP5)
)


p2 <- lapply(p, function(x){
    x + guides(color = FALSE, fill = FALSE) + 
    theme_ArchR(baseSize = 6.5) +
    theme(plot.margin = unit(c(0, 0, 0, 0), "cm")) +
    theme(
        axis.text.x=element_blank(), 
        axis.ticks.x=element_blank(), 
        axis.text.y=element_blank(), 
        axis.ticks.y=element_blank()
    )
})
do.call(cowplot::plot_grid, c(list(ncol = 3),p2))

plotPDF(p, name = "4_UMAP-JASPAR2020_chromvar_LinkedGeneExpression-Plot-Groups-Deviations-w-Imputation", width = 5, height = 5, ArchRProj = paracetP5, addDOC = TRUE)
```
#footprinting
```{r  Footprinting with ArchR, include=FALSE} 
#paracetP5 <- addGroupCoverages(ArchRProj = paracetP5, groupBy = "Clusters2", force = TRUE)
motifPositions <- getPositions(paracetP5)
motifPositions
###markers RNA
motifs <-  c("NEUROD1", "NEUROG1", "NFE2L3", "SOX9", "ZFHX2", "STAT2", "MAFK", "STAT5B", "REL", "JDP2", "NHLH2", "HMGA1", "ONECUT2", "TCF7L2", "HMX1", "ONECUT1", "SP4", "ISL2", "FOSL1", "RARB", "STAT1", "ONECUT3", "LHX4", "RREB1", "PBX1", "NHLH1", "CIC", "MAFB", "LHX1", "PKNOX1", "LHX5", "TEAD3", "NR2F2", "EMX2", "RHOXF1", "HNF1B", "UNCX", "ATOH7", "NFE2L1", "OTX2", "BPTF", "RELA", "SP1", "IKZF1 NEUROD4", "ALX1", "PKNOX2", "KLF4", "SNAI3","SNAI1", "LHX6", "DLX5", "OLIG2", "ATF7", "LHX5", "ISL1", "LHX8", "NFYC","IRF3", "MEIS3", "ZNF219", "FUBP1","TCF15", "E2F8", "OTP", "ZNF263", "ZNF75A", "FOSL2", "HMX2", "BCL6", "KLF5", "DLX1", "TFDP2", "MAFG-DT", "HMG20B", "KLF11", "ZNF384", "SMAD9", "ARX", "POU3F2", "LMO2", "E2F7", "NFATC4", "YY2", "CREB3", "ATF4", "MAFA-AS1", "E2F6", "ZFHX3", "ZNF75D", "BARX1", "NFKB1", "SMAD5", "TFDP1", "TOPORS", "UBP1", "NR2E1", "E2F4", "ARID3B", "TAL2", "DLX6", "TCF7", "KLF3", "RFX4", "TBR1", "PLAG1", "TCF7L1", "POU3F3", "EOMES", "LHX2", "NR2F1", "IRF7", "LHX9", "RUNX1", "ELF1", "IRF2", "TEAD4", "RUNX2", "CREB5", "ARID5A")
motifs <- c("SP3","PIM3")




markerMotifs <- unlist(lapply(motifs, function(x) grep(x, names(motifPositions), value = TRUE)))
markerMotifs <- markerMotifs[markerMotifs %ni% "SREBF1_22"]
markerMotifs


seFoot <- getFootprints(
  ArchRProj = paracetP5, 
  positions = motifPositions[markerMotifs], 
  groupBy = "Clusters2",
)
#ARCHR Palettes

#Normalization of Footprints for Tn5 Bias

#Subtracting the Tn5 Bias

plotFootprints(
  seFoot = seFoot1,
  ArchRProj = paracetP5, 
  normMethod = "Subtract",
  plotName = "SP3_6_Footprints-Subtract-Bias-S_Z",
  addDOC = TRUE,
  smoothWindow = 5,
  height = 6,
  width = 4,
  pal=constrained_palette
)

plotFootprints(
  seFoot = seFoot1,
  ArchRProj = paracetP5, 
  normMethod = "Divide",
  plotName = "SP3_6_Footprints-Divide-Bias",
  addDOC = TRUE,
  smoothWindow = 5,
  height = 6,
  width = 4,
  pal=constrained_palette
)




seTSS1 <- getFootprints(
  ArchRProj = paracetP5, 
  positions = GRangesList(TSS = getTSS(paracetP5)), 
  groupBy = "Clusters2",
  flank = 2000
)

plotFootprints(
  seFoot = seTSS1,
  ArchRProj = paracetP5, 
  normMethod = "None",
  plotName = "SP3_TSS-No-Normalization",
  addDOC = TRUE,
  flank = 2000,
  flankNorm = 100,
  height = 6,
  width = 4,
  pal=constrained_palette
)

```

#Add Coaccessibility

```{r Co-accessibility with ArchR,include=FALSE} 


paracetP5 <- addCoAccessibility(
    ArchRProj = paracetP5,
    reducedDims = "IterativeLSI"
)

cA <- getCoAccessibility(
    ArchRProj = paracetP5,
    corCutOff = 0.5,
    resolution = 1,
    returnLoops = FALSE
)

cA

metadata(cA)[[1]]

cA <- getCoAccessibility(
    ArchRProj = paracetP5,
    corCutOff = 0.5,
    resolution = 1,
    returnLoops = TRUE
)

cA[[1]]

cA <- getCoAccessibility(
    ArchRProj = paracetP5,
    corCutOff = 0.5,
    resolution = 1000,
    returnLoops = TRUE
)

cA[[1]]

cA <- getCoAccessibility(
    ArchRProj = paracetP5,
    corCutOff = 0.5,
    resolution = 10000,
    returnLoops = TRUE
)

cA[[1]]

```

#Plot_coaccessibility browsertrack
```{r Plotting browser tracks of Co-accessibility,include=FALSE} 

RNAmarkerGenes1  <- c("POU5F1", "NANOG", "TDGF1", "GAL", "PODXL", "ID1", "MYC", "PAX6", "DNMT3B", "LIN28A", "FOXD3-AS1", "CD24", "EPCAM", "DPPA4", "FZD2", "NR6A1", "SOX21", "ZIC2", "SOX11", "HESX1", "MYLK", "IGFBP5", "FZD5", "FGF8", "SOX2", "RAX", "LIX1", "FEZF2", "POU3F1", "POU3F2", "NKX2-1", "ASCL1", "NEUROG1", "NEUROD1", "ISL1", "BSX", "HMX2", "OTP", "GSX1", "SF1", "GNRH1", "GRIA2", "L1CAM", "NHLH1", "ELAVL4", "SIX3", "SIX6", "OTX2", "TH", "INSM1", "ARX", "DBX1", "DCX", "DLL1", "DLX1", "DLX2", "DLX5", "DLX6", "GAD1", "GRIA1", "INA", "KIF19", "NEFL", "NEFM", "NEUROD4", "NSG2", "POU2F2", "RELN", "SCG2", "SIM1", "SIM2", "SNAP25", "SP9", "SST", "STMN2", "SYP", "SYT4", "TAC1", "TAGLN3", "TLX3", "NTRK1", "NTRK3", "PCSK2", "RET", "DLL3", "GAD2", "GAP43", "REST", "SOX3", "TUBB3", "HES1", "BMP4", "KLF15", "CDK1", "TYMS", "ANLN", "HES6", "LMO1", "ELAVL3", "PRRX1", "OTX2", "PCNA", "SPARCL1", "MAP6", "KRT18", "DDIT4", "PAMR1", "FOXH1", "FABP7", "RBFOX3", "SOX6", "CDH7", "CDH12", "PHC1", "PHC2", "YAF2", "CDH1", "CDH2", "GNG2", "EMX2", "CDK6", "TAGLN", "GNG8", "KLF11", "LHX9", "DCT", "VEPH1", "GNG3", "GNG11", "TRIM71", "SOX3", "KRT19", "NSG1", "JHY")

set.seed(1)
addArchRThreads(threads = 1)

p <- plotBrowserTrack(
    ArchRProj = paracetP5, 
    groupBy = "Clusters2", 
    geneSymbol = RNAmarkerGenes1, 
    upstream = 50000,
    downstream = 50000,
    pal=constrained_palette,
    loops = getCoAccessibility(paracetP5)
)

grid::grid.newpage()
grid::grid.draw(p$CD14)

plotPDF(plotList = p, 
    name = "5_Plot-Tracks-RNASEQ_LIST-Genes-0.5-CoAccessibility.pdf", 
    ArchRProj = paracetP5, 
    addDOC = TRUE, width = 5, height = 5)



ATACmarkerGenes1  <- c("POU5F1", "NANOG", "TDGF1", "GAL", "PODXL", "ID1", "MYC", "PAX6", "DNMT3B", "LIN28A", "FOXD3-AS1", "CD24", "EPCAM", "DPPA4", "FZD2", "NR6A1", "SOX21", "ZIC2", "SOX11", "HESX1", "MYLK", "IGFBP5", "FZD5", "FGF8", "SOX2", "RAX", "LIX1", "FEZF2", "POU3F1", "POU3F2", "NKX2-1", "ASCL1", "NEUROG1", "NEUROD1", "ISL1", "BSX", "HMX2", "OTP", "GSX1", "SF1", "GNRH1", "GRIA2", "L1CAM", "NHLH1", "ELAVL4", "SIX3", "SIX6", "OTX2", "TH", "INSM1", "ARX", "DBX1", "DCX", "DLL1", "DLX1", "DLX2", "DLX5", "DLX6", "GAD1", "GRIA1", "INA", "KIF19", "NEFL", "NEFM", "NEUROD4", "NSG2", "POU2F2", "RELN", "SCG2", "SIM1", "SIM2", "SNAP25", "SP9", "SST", "STMN2", "SYP", "SYT4", "TAC1", "TAGLN3", "TLX3", "NTRK1", "NTRK3", "PCSK2", "RET", "DLL3", "GAD2", "GAP43", "REST", "SOX3", "TUBB3", "HES1", "BMP4", "KLF15", "CDK1", "TYMS", "ANLN", "HES6", "LMO1", "ELAVL3", "PRRX1", "OTX2", "PCNA", "SPARCL1", "MAP6", "KRT18", "DDIT4", "PAMR1", "FOXH1", "FABP7", "RBFOX3", "SOX6", "CDH7", "CDH12", "PHC1", "PHC2", "YAF2", "CDH1", "CDH2", "GNG2", "EMX2", "CDK6", "TAGLN", "GNG8", "KLF11", "LHX9", "DCT", "VEPH1", "GNG3", "GNG11", "TRIM71", "SOX3", "KRT19", "NSG1", "JHY")

set.seed(1)
addArchRThreads(threads = 1)

p <- plotBrowserTrack(
    ArchRProj = paracetP5, 
    groupBy = "Clusters2", 
    geneSymbol = ATACmarkerGenes1, 
    upstream = 50000,
    downstream = 50000,
    pal=constrained_palette,
    loops = getCoAccessibility(paracetP5)
)

grid::grid.newpage()
grid::grid.draw(p$CD14)

plotPDF(plotList = p, 
    name = "5_Plot-Tracks-ATACSEQ_LIST-Genes-0.5-CoAccessibility.pdf", 
    ArchRProj = paracetP5, 
    addDOC = TRUE, width = 5, height = 5)

```
#Coaccessibility
```{r contd.1 Plotting browser tracks of Co-accessibility for MOTIFS,include=FALSE}

 motifs <- c("Bach1::Mafk","HOXA2","HOXA4","HOXA5","HOXA6","HOXA7","HOXA9","HOXB13","HOXB2","HOXB3","HOXB4","HOXB5","HOXB6","HOXB7","HOXB8","HOXB9","HOXC10","HOXC11", "HOXC12", "HOXC13", "HOXC4", "HOXC8", "HOXC9", "HOXD10", "HOXD11", "HOXD12", "HOXD13", "HOXD3", "HOXD4", "HOXD8", "HOXD9", "Hsf", "HSF1", "HSF1", "HSF2",  "HSF4", "HSFA6B", "HSFB2A", "HSFB2")



set.seed(1)
addArchRThreads(threads = 1)

p <- plotBrowserTrack(
    ArchRProj = paracetP5, 
    groupBy = "Clusters2", 
    geneSymbol = motifs, 
    upstream = 50000,
    downstream = 50000,
    pal=constrained_palette,
    loops = getCoAccessibility(paracetP5)
)

grid::grid.newpage()
grid::grid.draw(p$CD14)

plotPDF(plotList = p, 
    name = "5_Plot-Tracks-motifs_LIST-Genes-0.5-CoAccessibility.pdf", 
    ArchRProj = paracetP5, 
    addDOC = TRUE, width = 5, height = 5)
```

#Peak2GeneLinkage
```{r Peak2GeneLinkage with ArchR} 

set.seed(1)


paracetP5 <- addPeak2GeneLinks(
    ArchRProj = paracetP5,
    reducedDims = "IterativeLSI",
    logFile = createLogFile("addPeak2GeneLinks")
)
metadata(paracetP5@peakSet)$Peak2GeneLink
markerGenes1  <- c("POU5F1","NANOG")
p2g <- getPeak2GeneLinks(
    ArchRProj = paracetP5,
    corCutOff = 0.4,
    resolution = 1,
    returnLoops = TRUE
   
)

p2g[[1]]
p2g <- getPeak2GeneLinks(
    ArchRProj = paracetP5,
    corCutOff = 0.1,
    resolution = 10000,
    returnLoops = TRUE
    
)
p2g[[1]]
write.csv(p2g[[1]],"getPeak2GeneLinks_withcorCutOff0.1.csv",sep="\t")

```


```{r Peak2GeneLinkage with ArchR} 

set.seed(1)
paracetP5 <- addPeak2GeneLinks(
    ArchRProj = paracetP5,
    reducedDims = "IterativeLSI",
    useMatrix = "GeneIntegrationMatrix"
    )
p2g <- getPeak2GeneLinks(
    ArchRProj = paracetP5,
    corCutOff = 0.4,
    resolution = 1,
    returnLoops = FALSE
)
p2g
metadata(p2g)
metadata(p2g)$seATAC


# Making Loop object from peak2gene link object p2g when return loop = FALSE

p2g$geneName <- mcols(metadata(p2g)$geneSet)$name[p2g$idxRNA]
tssgenes<- as.data.frame(metadata(p2g)$geneSet)
colnames(tssgenes)[2] <- "StartTSS"
colnames(tssgenes)[3] <- "EndTSS"
p2g$peakName <- (metadata(p2g)$peakSet %>% {paste0(seqnames(.), "_", start(.), "_", end(.))})[p2g$idxATAC]
temp1=as.data.frame(p2g)
temp  <- strsplit(p2g$peakName, "_")
temp <- matrix(unlist(temp), ncol=3, byrow=TRUE)
temp <- as.data.frame(temp)
temp1$chr <- temp$V1
temp1$start <- temp$V2
temp1$end <- temp$V3
mergedp2g <- merge(temp1,tssgenes ,by.x="geneName", by.y= "name")
mergedp2g$peakCentre=as.numeric(mergedp2g$start)+250
mergedp2g$looplength=as.numeric(mergedp2g$StartTSS)-as.numeric(mergedp2g$peakCentre)

write.table(mergedp2g,"_PARACET_peak2genelinks_corr_0.45_res-1_loopobject.txt")

#getpeak2genelinks 
p2g <- getPeak2GeneLinks(
    ArchRProj = paracetP5,
    corCutOff = 0.45,
    resolution = 1,
    returnLoops = FALSE
    
)
library(GenomicRanges)

p2g
markerGenes1  <- c("POU5F1","NANOG")
```

#ALL_PEAK2GeneLinks data
```{r dataformat include=FALSE} 
peak2genelinkmat <- plotPeak2GeneHeatmap(ArchRProj = paracetP5, groupBy = "Clusters2",
seed = 1,
  palATAC = paletteContinuous("coolwarm"),
  palRNA = paletteContinuous("blueYellow"),
  corCutOff = 0.45,
  FDRCutOff = 1e-04,
returnMatrices = TRUE,
nPlot = 100000
)
yy <- as.data.frame(peak2genelinkmat@listData[["Peak2GeneLinks"]])
xx1 <- as.data.frame(peak2genelinkmat@listData[["ATAC"]]@listData[["kmeansId"]])
yy$kmeansATAC <- xx1$`peak2genelinkmat@listData[["ATAC"]]@listData[["kmeansId"]]`
xx2 <- as.data.frame(peak2genelinkmat@listData[["RNA"]]@listData[["kmeansId"]])
yy$kmeansRNA <- xx2$`peak2genelinkmat@listData[["RNA"]]@listData[["kmeansId"]]`
yy$peak <- str_replace(yy$peak,":","_")
yy$peak <- str_replace(yy$peak,"-","_")
yy

temp_mat=as.data.frame(peak2genelinkmat@listData[["ATAC"]]$matrix)
temp_kmean=as.data.frame(peak2genelinkmat@listData[["ATAC"]]$kmeansId)
rownames(temp_kmean)=rownames(temp_mat)
which((rownames(temp_kmean)==rownames(yy))==TRUE)
unique(temp_kmean$`peak2genelinkmat@listData[["ATAC"]]$kmeansId`)
plotPDF(p, name = "5_Plot-Heatmap_peak2genelinks-scRNA_scATAC_blueyellow_purpleOrange_updated.pdf", ArchRProj = paracetP5, addDOC = TRUE, width = 6, height = 16)
temp_kmean
######extraction of data of peak2genelinks ######

yy$genepeak <- paste(yy$gene,yy$peak,sep=":")

mergedp2g$genepeak <- paste(mergedp2g$geneName,mergedp2g$peakName,sep=":")
alldat <- merge(mergedp2g,yy,by="genepeak")
write.table(alldat,"_PARACET_peak2genelinks_day0d20_alldatakmeans_corr_0.45.txt")


```



#peak2genelinkbrowseratac
```{r Plotting browser tracks with peak-to-gene links, include=FALSE} 

markerGenes2  <- c("POU5F1", "NANOG", "TDGF1", "GAL", "PODXL", "ID1", "MYC", "PAX6", "DNMT3B", "LIN28A", "FOXD3-AS1", "CD24", "EPCAM", "DPPA4", "FZD2", "NR6A1", "SOX21", "ZIC2", "SOX11", "HESX1", "MYLK", "IGFBP5", "FZD5", "FGF8", "SOX2", "RAX", "LIX1", "FEZF2", "POU3F1", "POU3F2", "NKX2-1", "ASCL1", "NEUROG1", "NEUROD1", "ISL1", "BSX", "HMX2", "OTP", "GSX1", "SF1", "GNRH1", "GRIA2", "L1CAM", "NHLH1", "ELAVL4", "SIX3", "SIX6", "OTX2", "TH", "INSM1")


#By default, max cells are 500 for plotting peaks, if you want to change default max cells  Then use  maxCells = 5000 for 5000 cells in plotbrowser track. Option only supported in ArchR dev version or ArchR 1.0.3
p <- plotBrowserTrack(
    ArchRProj = paracetP5, 
    groupBy = "Clusters2", 
    geneSymbol = markerGenes2, 
    upstream = 50000,
    downstream = 50000,
    pal=constrained_palette,
    loops = getPeak2GeneLinks(paracetP5)
)

grid::grid.newpage()
grid::grid.draw(p$POU5F1)

plotPDF(plotList = p, 
    name = "_5_Plot-Tracks-Marker-Genes-with-Peak2GeneLinks.pdf", 
    ArchRProj = paracetP5, 
    addDOC = TRUE, width = 5, height = 5)

#

p <- plotPeak2GeneHeatmap(ArchRProj = paracetP5, groupBy = "Clusters2",seed = 1,
  palATAC = paletteContinuous("coolwarm"),
  palRNA = paletteContinuous("blueYellow"),)
p

plotPDF(p, name = "5_Plot-Heatmap_peak2genelinks-scRNA_scATAC_blueyellow_purpleOrange.pdf", ArchRProj = paracetP5, addDOC = TRUE, width = 6, height = 16)


```

#Trajectory analysis
```{r Identification of Positive TF-Regulators,include=FALSE} 
#Step 1. Identify Deviant TF Motifs

seGroupMotif <- getGroupSE(ArchRProj = paracetP5, useMatrix = "MotifMatrix", groupBy = "Clusters2")

seGroupMotif

seZ <- seGroupMotif[rowData(seGroupMotif)$seqnames=="z",]

rowData(seZ)$maxDelta <- lapply(seq_len(ncol(seZ)), function(x){
  rowMaxs(assay(seZ) - assay(seZ)[,x])
}) %>% Reduce("cbind", .) %>% rowMaxs

#Step 2. Identify Correlated TF Motifs and TF Gene Score/Expression

corGSM_MM <- correlateMatrices(
    ArchRProj = paracetP5,
    useMatrix1 = "GeneScoreMatrix",
    useMatrix2 = "MotifMatrix",
    reducedDims = "IterativeLSI"
)

corGSM_MM

corGIM_MM <- correlateMatrices(
    ArchRProj = paracetP5,
    useMatrix1 = "GeneIntegrationMatrix",
    useMatrix2 = "MotifMatrix",
    reducedDims = "IterativeLSI"
)

corGIM_MM

corGSM_MM$maxDelta <- rowData(seZ)[match(corGSM_MM$MotifMatrix_name, rowData(seZ)$name), "maxDelta"]
corGIM_MM$maxDelta <- rowData(seZ)[match(corGIM_MM$MotifMatrix_name, rowData(seZ)$name), "maxDelta"]

corGSM_MM <- corGSM_MM[order(abs(corGSM_MM$cor), decreasing = TRUE), ]
corGSM_MM <- corGSM_MM[which(!duplicated(gsub("\\-.*","",corGSM_MM[,"MotifMatrix_name"]))), ]
corGSM_MM$TFRegulator <- "NO"
corGSM_MM$TFRegulator[which(corGSM_MM$cor > 0.5 & corGSM_MM$padj < 0.01 & corGSM_MM$maxDelta > quantile(corGSM_MM$maxDelta, 0.75))] <- "YES"
sort(corGSM_MM[corGSM_MM$TFRegulator=="YES",1])

#highligh TFs in a dot blot
p <- ggplot(data.frame(corGSM_MM), aes(cor, maxDelta, color = TFRegulator)) +
  geom_point() + 
  theme_ArchR() +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("NO"="darkgrey", "YES"="firebrick3")) +
  xlab("Correlation To Gene Score") +
  ylab("Max TF Motif Delta") +
  scale_y_continuous(
    expand = c(0,0), 
    limits = c(0, max(corGSM_MM$maxDelta)*1.05)
  )



plotPDF(plotList =p,name="5_correlationtogenesTFregulators_motifs.pdf", ArchRProj = paracetP5, addDOC = TRUE, width = 5, height = 5)
dev.off()
corGIM_MM <- corGIM_MM[order(abs(corGIM_MM$cor), decreasing = TRUE), ]
corGIM_MM <- corGIM_MM[which(!duplicated(gsub("\\-.*","",corGIM_MM[,"MotifMatrix_name"]))), ]
corGIM_MM$TFRegulator <- "NO"
corGIM_MM$TFRegulator[which(corGIM_MM$cor > 0.5 & corGIM_MM$padj < 0.01 & corGIM_MM$maxDelta > quantile(corGIM_MM$maxDelta, 0.75))] <- "YES"
sort(corGIM_MM[corGIM_MM$TFRegulator=="YES",1])

p <- ggplot(data.frame(corGIM_MM), aes(cor, maxDelta, color = TFRegulator)) +
  geom_point() + 
  theme_ArchR() +
  geom_vline(xintercept = 0, lty = "dashed") + 
  scale_color_manual(values = c("NO"="darkgrey", "YES"="firebrick3")) +
  xlab("Correlation To Gene Expression") +
  ylab("Max TF Motif Delta") +
  scale_y_continuous(
    expand = c(0,0), 
    limits = c(0, max(corGIM_MM$maxDelta)*1.05)
  )

p
plotPDF(plotList = p, 
    name = "5_Plot-positive_tf_regulator.pdf", 
    ArchRProj = paracetP5, 
    addDOC = TRUE, width = 5, height = 5)
p
write.csv(corGSM_MM, "5_positive_matrix_tf_regulator.csv", sep = "\t")

```
#clustermarkerlist
```{r ID marker peaks} 
#
markersPM <- getMarkerFeatures(
    ArchRProj = paracetP5, 
    useMatrix = "PeakMatrix", 
    groupBy = "Clusters2",
    bias = c("log10(nFrags)"),
    testMethod = "wilcoxon"
)
markerListPM <- getMarkers(markersPM, cutOff = "FDR <= 0.01 & Log2FC >= 0.25")
Cluster_marker_list_PM <- write.csv(markerListPM, "5_ATAC_cluster_marker_list_PM.csv", sep = "\t")

```
#Trajectory Analysis 
```{r ID marker peaks,include=FALSE} 

#paracetP5 <- loadArchRProject("Save-paracetP5/")

p6 <- plotEmbedding(ArchRProj = paracetP5, colorBy = "cellColData", name = "Clusters2", embedding = "UMAP" , pal=constrained_palette,)
#p7 <- plotEmbedding(ArchRProj = paracetP5, colorBy = "cellColData", name = "Clusters22", embedding = "UMAP")
p6

trajectory <- c("P3","P5","P9","P13","P10")
trajectory

paracetP5 <- addTrajectory(
    ArchRProj = paracetP5, 
    name = "Differentiation", 
    groupBy = "Clusters2",
    trajectory = trajectory, 
    embedding = "UMAP", 
    force = TRUE,
)
head(paracetP5$Differentiation[!is.na(paracetP5$Differentiation)])

p <- plotTrajectory(paracetP5, trajectory = "Differentiation", colorBy = "cellColData", name = "Differentiation",size=0.3)

p[[1]]
plotPDF(p, name = "Plot-Differentiation-Traj-UMAP_cell_Coldata.pdf", ArchRProj = paracetP5, addDOC = TRUE, width = 5, height = 5)


paracetP5 <- addImputeWeights(paracetP5)
p1 <- plotTrajectory(paracetP5, trajectory = "Differentiation", colorBy = "GeneScoreMatrix", name = "DLX1", continuousSet = "horizonExtra")
p1

p2 <- plotTrajectory(paracetP5, trajectory = "Differentiation", colorBy = "GeneIntegrationMatrix", name = "DLX1", continuousSet = "blueYellow")

ggAlignPlots(p1[[1]], p2[[1]], type = "h")


plotPDF(p1[[1]], p2[[1]], name = "Plot-Differentiation-Traj-UMAP_DLX1_gene_score_and_GeneIntegrationmatrix.pdf", ArchRProj = paracetP5, addDOC = TRUE, width = 5, height = 5)


trajMM  <- getTrajectory(ArchRProj = paracetP5, name = "Differentiation", useMatrix = "MotifMatrix", log2Norm = FALSE)
p1 <- plotTrajectoryHeatmap(trajMM, pal = paletteContinuous(set = "solarExtra"))
trajGSM <- getTrajectory(ArchRProj = paracetP5, name = "Differentiation", useMatrix = "GeneScoreMatrix", log2Norm = TRUE)
p2 <- plotTrajectoryHeatmap(trajGSM,  pal = paletteContinuous(set = "horizonExtra"))
p2
trajGIM <- getTrajectory(ArchRProj = paracetP5, name = "Differentiation", useMatrix = "GeneIntegrationMatrix", log2Norm = FALSE)

p3 <- plotTrajectoryHeatmap(trajGIM,  pal = paletteContinuous(set = "blueYellow"))

trajPM  <- getTrajectory(ArchRProj = paracetP5, name = "Differentiation", useMatrix = "PeakMatrix", log2Norm = TRUE)

p4 <- plotTrajectoryHeatmap(trajPM, pal = paletteContinuous(set = "solarExtra"))

plotPDF(p1, p2, p3, p4, name = "6_Plot-DIfferentiation-Traj-Heatmaps.pdf", ArchRProj = paracetP5, addDOC = TRUE, width = 6, height = 8)

corGSM_MM <- correlateTrajectories(trajGSM, trajMM)

corGSM_MM[[1]]

trajGSM2 <- trajGSM[corGSM_MM[[1]]$name1, ]
trajMM2 <- trajMM[corGSM_MM[[1]]$name2, ]


trajCombined <- trajGSM2

assay(trajCombined,withDimnames=FALSE) <- t(apply(assay(trajGSM2), 1, scale)) + t(apply(assay(trajMM2), 1, scale))

combinedMat <- plotTrajectoryHeatmap(trajCombined, returnMat = TRUE, varCutOff = 0)

rowOrder <- match(rownames(combinedMat), rownames(trajGSM2))

ht1 <- plotTrajectoryHeatmap(trajGSM2,  pal = paletteContinuous(set = "horizonExtra"),  varCutOff = 0, rowOrder = rowOrder)

ht2 <- plotTrajectoryHeatmap(trajMM2, pal = paletteContinuous(set = "solarExtra"), varCutOff = 0, rowOrder = rowOrder)
ht1 + ht2

plotPDF(ht1 + ht2, name = "6_Plot-DIfferentiation-Traj_combined_mat_GeneScore_Motif_Matrix_Heatmaps.pdf", ArchRProj = paracetP5, addDOC = TRUE, width = 6, height = 8)
corGIM_MM <- correlateTrajectories(trajGIM, trajMM)
corGIM_MM[[1]]

trajGIM2 <- trajGIM[corGIM_MM[[1]]$name1, ]
trajMM2 <- trajMM[corGIM_MM[[1]]$name2, ]

trajCombined <- trajGIM2
assay(trajCombined,withDimnames=FALSE) <- t(apply(assay(trajGIM2), 1, scale)) + t(apply(assay(trajMM2), 1, scale))

combinedMat <- plotTrajectoryHeatmap(trajCombined, returnMat = TRUE, varCutOff = 0)

rowOrder <- match(rownames(combinedMat), rownames(trajGIM2))
ht1 <- plotTrajectoryHeatmap(trajGIM2,  pal = paletteContinuous(set = "blueYellow"),  varCutOff = 0, rowOrder = rowOrder)
ht2 <- plotTrajectoryHeatmap(trajMM2, pal = paletteContinuous(set = "solarExtra"), varCutOff = 0, rowOrder = rowOrder)
ht1 + ht2
plotPDF(ht1 + ht2, name = "6_Plot-DIfferentiation-Traj_Combined_correlate_GeneScore_Motif_Matrix_Heatmaps.pdf", ArchRProj = paracetP5, addDOC = TRUE, width = 6, height = 8)


saveArchRProject(ArchRProj = paracetP5, outputDirectory = "Save-paracetP5", load = FALSE)
```


#END OF THE PIPELINE

